---
title: "AXH_DEseq-heatmaps_Pdam"
author: "Mike Connelly"
date: "9/21/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI_AntiXHeat-master")
```

### Create colony and condition annotation colors
```{r}
anno_colors <- list(
  Colony = c(HW1=colcolors[1], HW2=colcolors[2], WT1=colcolors[3], WT2=colcolors[4]),
  Treatment = c(control=condcolors_antixheat[1],  Heat=condcolors_antixheat[3], Antibiotics=condcolors_antixheat[2], Antibiotics.Heat=condcolors_antixheat[4]),
  DEG = c(Upregulated="red", Downregulated="blue"))
```
### Create sample heatmap cell color
```{r}
colonyclustercol <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
```
### Create gene expression cell color
```{r}
geneexpcolors <- colorRampPalette( rev(brewer.pal(9, "Spectral")) )(300)
```
### Create gene expression cell color
```{r}
# Function to plot color bar
color.bar <- function(lut, min, max=-min, nticks=11, ticks=seq(min, max, len=nticks), title='') {
    scale = (length(lut)-1)/(max-min)

    dev.new(width=1.75, height=5)
    plot(c(0,10), c(min,max), type='n', bty='n', xaxt='n', xlab='', yaxt='n', ylab='', main=title)
    axis(2, ticks, las=1)
    for (i in 1:(length(lut)-1)) {
    	y = (i-1)/scale + min
    	rect(0,y,10,y+1/scale, col=lut[i], border=NA)
    }	
}
```
```{r}
geneexpcolors <- colorRampPalette( rev(brewer.pal(9, "RdYlBu")) )(300)

geneexpcolors1 <- colorRampPalette(colors()[c(26,1,553)])(300)

# Here is a fancy color palette inspired by http://www.colbyimaging.com/wiki/statistics/color-bars
down <- rainbow(50, start=rgb2hsv(col2rgb('cyan'))[1], end=rgb2hsv(col2rgb('blue'))[1])
up <- rainbow(50, start=rgb2hsv(col2rgb('red'))[1], end=rgb2hsv(col2rgb('yellow'))[1])
cols <- c(rev(down), "white", rev(up))
mypalette <- colorRampPalette(cols)(400)
6/length(mypalette)

geneexpcolors2 <- mypalette

color.bar(down, -1, 1)
color.bar(up, -1, 1)
color.bar(mypalette, -3, 3)

color.bar(geneexpcolors, -1, 1)
color.bar(geneexpcolors1, -1, 1)
color.bar(geneexpcolors2, -1, 1)
```

## Regularized-log transformation 
```{r, include=FALSE}
      rld_antixheat <- rlogTransformation(dds_antixheat)
```
### Create heatmap annotation data frame
```{r}
anno_col <- as.data.frame(colData(rld_antixheat)[, c("Treatment","Colony")])
```

## Sample Clustering Heatmap
```{r}
sampleDists <- dist(t(assay(rld_antixheat)))
sampleDistMatrix <- as.matrix( sampleDists )
```

```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Heatmap_AXH_samples.pdf", width = 8.5, height = 7.5)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colonyclustercol,
         annotation_colors = anno_colors,
         annotation_col = anno_col,
         annotation_row = anno_col,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 8,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 4,
         treeheight_row = 50,
         cutree_rows = 4)
```

## Gene Clustering Heatmaps
### Create top 20 variable gene expression matrix 
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes20 <- head(order(rowVars(assay(rld_antixheat)), decreasing = TRUE), 20)
mat20  <- assay(rld_antixheat)[ topVarGenes20, ]
rownames(mat20) <- rowData(rld_antixheat)$IDGeneInfo[topVarGenes20]
```
```{r}
mat20  <- mat20 - rowMeans(mat20)
```
## 20 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Heatmap_AXH_top20.pdf", width = 11, height = 8.5)
pheatmap(mat20, 
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 20 Most Variable Gene Expression")
```

### Create top 50 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes50 <- head(order(rowVars(assay(rld_antixheat)), decreasing = TRUE), 50)
mat50  <- assay(rld_antixheat)[ topVarGenes50, ]
rownames(mat50) <- rowData(rld_antixheat)$IDGeneInfo[topVarGenes50]
```
```{r}
mat50  <- mat50 - rowMeans(mat50)
```
## 50 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Heatmap_AXH_top50.pdf", width = 11, height = 8.5)
pheatmap(mat50,
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 5,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 50 Most Variable Gene Expression")
```

### Create top 500 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes500 <- head(order(rowVars(assay(rld_antixheat)), decreasing = TRUE), 500)
mat500  <- assay(rld_antixheat)[ topVarGenes500, ]
rownames(mat500) <- rowData(rld_antixheat)$IDGeneInfo[topVarGenes500]
```
## 500 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Heatmap_AXH_top500.pdf", width = 11, height = 8.5)
pheatmap(mat500,          
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = FALSE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 500 Most Variable Gene Expression")
```

### Create Antibiotics vs. Control differentially expressed gene matrix 
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
rownames_ressig_anti_ctrl_up <- row.names(ressig_anti_ctrl_up)
mat_ressig_anti_ctrl_up  <- assay(rld_antixheat)[rownames_ressig_anti_ctrl_up, ]
row.names(mat_ressig_anti_ctrl_up) <- ressig_anti_ctrl_up$IDGeneInfo
```
```{r}
mat_ressig_anti_ctrl_up  <- mat_ressig_anti_ctrl_up - rowMeans(mat_ressig_anti_ctrl_up)
```
## Antibiotics vs. Control up DEG Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXheat-master/DESeqresults_Pdam/figures/Heatmap_Anti_upDEG.pdf", width = 11, height = 8.5)
pheatmap(mat_ressig_anti_ctrl_up, col=geneexpcolors2,
         annotation_col = anno_col,
         annotation_row = NULL,
         show_rownames = TRUE,
         fontsize_row = 10,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         legend = FALSE,
         treeheight_col = 60,
         treeheight_row = 120)
```
