---
title: "AXH_DESeq_Pdam"
author: "Mike Connelly"
date: "12/9/2019"
output: html_document
---

RMarkdown document containing scripts for DESeq2 modelling of differential gene expression between fragments of four *Pocillopora* coral genotypes collected at Wanglitung and Houwan reefs and placed in three experimental treatment groups and once control group during EAPSI Taiwan, summer 2017.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, error = TRUE, include = TRUE, message = TRUE, warning = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_AxH/")
```

# Load packages, set plotting themes, colors, and shapes
Include all packages needed for the entire DESeq2 analysis and downstream visualizations
```{r packages, include=FALSE}
packages <- c("tidyverse", "RColorBrewer", "ggpubr", "ggrepel", "DESeq2", "apeglm", "pheatmap", "genefilter", "calibrate", "VennDiagram", "UpSetR")
lapply(packages, library, character.only = TRUE)
library("topGO")
library("GO.db")
library("Rgraphviz")
```
```{r}
theme_set(theme_bw())
condcolors_AxH <- c("blue", "cyan", "darkorange", "gold")
condcolors_anti <- c(condcolors_AxH[1:2])
condcolors_heat <- c(condcolors_AxH[1], condcolors_AxH[3])
condcolors_anti.heat <- c(condcolors_AxH[1], condcolors_AxH[4])
###
condfillcolors_AxH <- c("#99CCFF", "#FFCC99", "#99FFFF", "#FFFF66")
colshapes <- c(16, 17, 15, 18)
colcolors <- c("olivedrab3", "springgreen", "deepskyblue", "skyblue")
###
condcolors_null <- c(rep("black", 4))
colshapes_null <- c(rep(20, 8))
```
```{r}
DEGcolors <- c("red", "blue", "black")
```
```{r}
### Create colony and condition annotation colors
anno_colors <- list(
  Colony = c(HW1=colcolors[1], HW2=colcolors[2], WT1=colcolors[3], WT2=colcolors[4]),
  Treatment = c(control=condcolors_AxH[1],  Heat=condcolors_AxH[3], Antibiotics=condcolors_AxH[2], Antibiotics.Heat=condcolors_AxH[4]),
  DEG = c(Upregulated="red", Downregulated="blue"))
```
```{r}
### Create gene expression cell color
# Function to plot color bar
color.bar <- function(lut, min, max=-min, nticks=11, ticks=seq(min, max, len=nticks), title='') {
    scale = (length(lut)-1)/(max-min)

    dev.new(width=1.75, height=5)
    plot(c(0,10), c(min,max), type='n', bty='n', xaxt='n', xlab='', yaxt='n', ylab='', main=title)
    axis(2, ticks, las=1)
    for (i in 1:(length(lut)-1)) {
    	y = (i-1)/scale + min
    	rect(0,y,10,y+1/scale, col=lut[i], border=NA)
    }	
}
```
```{r}
### Create sample heatmap cell color
colonyclustercol <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
### Create gene expression cell color
geneexpcolors <- colorRampPalette( rev(brewer.pal(9, "Spectral")) )(300)
geneexpcolors <- colorRampPalette( rev(brewer.pal(9, "RdYlBu")) )(300)
geneexpcolors1 <- colorRampPalette(colors()[c(26,1,553)])(300)

# Here is a fancy color palette inspired by http://www.colbyimaging.com/wiki/statistics/color-bars
down <- rainbow(50, start=rgb2hsv(col2rgb('cyan'))[1], end=rgb2hsv(col2rgb('blue'))[1])
up <- rainbow(50, start=rgb2hsv(col2rgb('red'))[1], end=rgb2hsv(col2rgb('yellow'))[1])
cols <- c(rev(down), "white", rev(up))
mypalette <- colorRampPalette(cols)(400)
6/length(mypalette)

geneexpcolors2 <- mypalette

color.bar(down, -1, 1)
color.bar(up, -1, 1)
color.bar(mypalette, -3, 3)
color.bar(geneexpcolors, -1, 1)
color.bar(geneexpcolors1, -1, 1)
color.bar(geneexpcolors2, -1, 1)
```

# DESeq2 Analysis
### Import sample metadata and counts data
```{r, echo=TRUE}
samples <- read.table("./data/EAPSI_samples_AxH.txt", header = TRUE)
countdata <- read.delim("./outputs/STARcounts_Pdam/AxH_Pdam.counts", comment.char="#")
```
### Tidy counts data
* Set Gene ID's as row names  
* Remove first five columns (chr, start, end, strand, length)  
* Remove file prefixes and suffixes  
* Create condition factors and column data  
```{r, echo=FALSE}
row.names(countdata) <- countdata$Geneid
countdata <- countdata[ ,7:ncol(countdata)]
colnames(countdata) <- gsub("X.", "", colnames(countdata))
colnames(countdata) <- gsub("scratch.projects.transcriptomics.mikeconnelly.projects.EAPSI_Pocillopora_AxH.outputs.STARalign_Pdam.", "", colnames(countdata))
colnames(countdata) <- gsub("_PdamAligned.out.bam$", "", colnames(countdata))
```
### Import gene feature annotation information
```{r, include=FALSE}
genefeatures <- read.delim(file = "./data/pdam_genome_IDInfo.gff", header = F)
colnames(genefeatures) <- c("IDGeneInfo")
rownames(genefeatures) <- rownames(countdata)
```
```{r}
#Check gene feature annotation and countdata rowname order coherence
all(rownames(countdata) == rownames(genefeatures))
```
### Select which treatments to use in DESeq object construction
* anti: Control-Antibiotics  
* heat: Control-Heat  
* anti.heat: Control-Antibiotics.Heat  
* antixheat: Control-Antibiotics-Heat-Antibiotics.Heat  
```{r, echo=FALSE}
countdata_anti <- dplyr::select(countdata, matches("[HW][wt][12].[56]."))
countdata_heat <- dplyr::select(countdata, matches("[HW][wt][12].[46]."))
countdata_anti.heat <- dplyr::select(countdata, matches("[HW][wt][12].[16]."))
countdata_AxH <- dplyr::select(countdata, matches("[HW][wt][12].[1456]."))
###
samples_anti <- dplyr::filter(samples, Treatment == "control" | Treatment == "Antibiotics")
samples_heat <- dplyr::filter(samples, Treatment == "control" | Treatment == "Heat")
samples_anti.heat <- dplyr::filter(samples, Treatment == "control" | Treatment == "Antibiotics.Heat")
samples_AxH <- dplyr::filter(samples, Treatment == "control" | Treatment == "Antibiotics" | Treatment == "Heat" | Treatment == "Antibiotics.Heat")
```
```{r, echo=FALSE}
countdata_HW1 <- dplyr::select(countdata_AxH, matches("Hw1.*"))
countdata_HW2 <- dplyr::select(countdata_AxH, matches("Hw2.*"))
countdata_WT1 <- dplyr::select(countdata_AxH, matches("Wt1.*"))
countdata_WT2 <- dplyr::select(countdata_AxH, matches("Wt2.*"))
###
samples_HW1 <- dplyr::filter(samples_AxH, Colony == "HW1")# %>% select(-contains("W"), -contains("2"))
samples_HW2 <- dplyr::filter(samples_AxH, Colony == "HW2")# %>% select(-contains("W"), -contains("1"))
samples_WT1 <- dplyr::filter(samples_AxH, Colony == "WT1")# %>% select(-contains("H"), -contains("2"))
samples_WT2 <- dplyr::filter(samples_AxH, Colony == "WT2")# %>% select(-contains("H"), -contains("1"))
```
### Convert to matrices and create DESeq objects
```{r, include=FALSE}
countdata_anti <- as.matrix(countdata_anti)
countdata_heat <- as.matrix(countdata_heat)
countdata_anti.heat <- as.matrix(countdata_anti.heat)
countdata_AxH <- as.matrix(countdata_AxH)
###
countdata_HW1 <- as.matrix(countdata_HW1)
countdata_HW2 <- as.matrix(countdata_HW2)
countdata_WT1 <- as.matrix(countdata_WT1)
countdata_WT2 <- as.matrix(countdata_WT2)
```
```{r, message=FALSE}
dds_anti <- DESeqDataSetFromMatrix(countData=countdata_anti, colData=samples_anti, design= ~Batch + Colony + Treatment)
dds_anti$Treatment <- factor(dds_anti$Treatment, levels=c("control", "Antibiotics"))
relevel(dds_anti$Treatment, ref = "control")

dds_heat <- DESeqDataSetFromMatrix(countData=countdata_heat, colData=samples_heat, design= ~Batch + Colony + Treatment)
dds_heat$Treatment <- factor(dds_heat$Treatment, levels=c("control", "Heat"))
relevel(dds_heat$Treatment, ref = "control")

dds_anti.heat <- DESeqDataSetFromMatrix(countData=countdata_anti.heat, colData=samples_anti.heat, design= ~Batch + Colony + Treatment)
dds_anti.heat$Treatment <- factor(dds_anti.heat$Treatment, levels=c("control", "Antibiotics.Heat"))
relevel(dds_anti.heat$Treatment, ref = "control")

dds_AxH <- DESeqDataSetFromMatrix(countData=countdata_AxH, colData=samples_AxH, design= ~Batch + Colony + Treatment)
dds_AxH$Treatment <- factor(dds_AxH$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_AxH$Treatment, ref = "control")
```
```{r, message=FALSE}
dds_HW1 <- DESeqDataSetFromMatrix(countData=countdata_HW1, colData=samples_HW1, design= ~Treatment)
### Batch effect encompasses Treatment
dds_HW1$Treatment <- factor(dds_HW1$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_HW1$Treatment, ref = "control")

dds_HW2 <- DESeqDataSetFromMatrix(countData=countdata_HW2, colData=samples_HW2, design= ~Batch + Treatment)
dds_HW2$Treatment <- factor(dds_HW2$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_HW2$Treatment, ref = "control")

dds_WT1 <- DESeqDataSetFromMatrix(countData=countdata_WT1, colData=samples_WT1, design= ~Batch + Treatment)
dds_WT1$Treatment <- factor(dds_WT1$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_WT1$Treatment, ref = "control")

dds_WT2 <- DESeqDataSetFromMatrix(countData=countdata_WT2, colData=samples_WT2, design= ~Batch + Treatment)
dds_WT2$Treatment <- factor(dds_WT2$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_WT2$Treatment, ref = "control")
```
```{r, echo=FALSE}
#Check annotation and dds object rowname order coherence
all(rownames(dds_anti) == rownames(genefeatures))
all(rownames(dds_heat) == rownames(genefeatures))
all(rownames(dds_anti.heat) == rownames(genefeatures))
all(rownames(dds_AxH) == rownames(genefeatures))
###
all(rownames(dds_HW1) == rownames(genefeatures))
all(rownames(dds_HW2) == rownames(genefeatures))
all(rownames(dds_WT1) == rownames(genefeatures))
all(rownames(dds_WT2) == rownames(genefeatures))
```
```{r, echo=FALSE}
#Add gene feature annotation to DESeqDataSets
mcols(dds_anti) <- cbind(mcols(dds_anti), genefeatures)
mcols(dds_heat) <- cbind(mcols(dds_heat), genefeatures)
mcols(dds_anti.heat) <- cbind(mcols(dds_anti.heat), genefeatures)
mcols(dds_AxH) <- cbind(mcols(dds_AxH), genefeatures)
###
mcols(dds_HW1) <- cbind(mcols(dds_HW1), genefeatures)
mcols(dds_HW2) <- cbind(mcols(dds_HW2), genefeatures)
mcols(dds_WT1) <- cbind(mcols(dds_WT1), genefeatures)
mcols(dds_WT2) <- cbind(mcols(dds_WT2), genefeatures)
```

### Perform DESeq2 analyses
```{r, message=FALSE}
dds_anti <- DESeq(dds_anti)
dds_heat <- DESeq(dds_heat)
dds_anti.heat <- DESeq(dds_anti.heat)
dds_AxH <- DESeq(dds_AxH)
```
```{r, message=FALSE}
dds_HW1 <- DESeq(dds_HW1)
dds_HW2 <- DESeq(dds_HW2)
dds_WT1 <- DESeq(dds_WT1)
dds_WT2 <- DESeq(dds_WT2)
```

### Filter low counts
```{r}
keep <- rowSums(counts(dds_AxH), na.rm = TRUE) >= 10
dds_anti <- dds_anti[keep,]
dds_heat <- dds_heat[keep,]
dds_anti.heat <- dds_anti.heat[keep,]
dds_AxH <- dds_AxH[keep,]
```
```{r}
dds_HW1 <- dds_HW1[keep,]
dds_HW2 <- dds_HW2[keep,]
dds_WT1 <- dds_WT1[keep,]
dds_WT2 <- dds_WT2[keep,]
```

### Obtain DESeq results
```{r, echo=FALSE}
res_anti <- results(dds_anti)
res_heat <- results(dds_heat)
res_anti.heat <- results(dds_anti.heat)
res_AxH <- results(dds_AxH)
```
```{r, echo=FALSE}
res_HW1 <- results(dds_HW1)
res_HW2 <- results(dds_HW2)
res_WT1 <- results(dds_WT1)
res_WT2 <- results(dds_WT2)
```

### Check outputs of DESeq2 analyses to ensure quality
```{r, echo=FALSE}
# Antibiotics
resultsNames(dds_anti)
summary(res_anti)
res_anti$IDGeneInfo <- mcols(dds_anti)$IDGeneInfo
res_anti$IDGeneInfo <- as.character(res_anti$IDGeneInfo)
```
```{r}
plotDispEsts(dds_anti)
plotMA(res_anti, ylim = c(-20, 20))
```
```{r, echo=FALSE}
# Heat
resultsNames(dds_heat)
summary(res_heat)
res_heat$IDGeneInfo <- mcols(dds_heat)$IDGeneInfo
res_heat$IDGeneInfo <- as.character(res_heat$IDGeneInfo)
```
```{r}
plotDispEsts(dds_heat)
plotMA(res_heat, ylim = c(-20, 20))
```
```{r, echo=FALSE}
# Antibiotics + Heat
resultsNames(dds_anti.heat)
summary(res_anti.heat)
res_anti.heat$IDGeneInfo <- mcols(dds_anti.heat)$IDGeneInfo
res_anti.heat$IDGeneInfo <- as.character(res_anti.heat$IDGeneInfo)
```
```{r}
plotDispEsts(dds_anti.heat)
plotMA(res_anti.heat, ylim = c(-20, 20))
```
```{r, echo=FALSE}
# Antibiotics X Heat
resultsNames(dds_AxH)
summary(res_AxH)
res_AxH$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_AxH$IDGeneInfo <- as.character(res_AxH$IDGeneInfo)
```
```{r}
plotDispEsts(dds_AxH)
plotMA(res_AxH, ylim = c(-20, 20))
```

## Perform the variance-stabilizing transformation
```{r}
vsd <- vst(dds_AxH, blind = FALSE)
```
## Return normalized and variance-stabilized counts matrix
```{r}
norm_vst_countdata <- assay(vsd)
```
## Transpose
```{r}
datExpr <- t(norm_vst_countdata)
```
```{r}
rownames(datExpr)
ncol(datExpr)
```
This returns 22537 genes for network analysis
## Save normalized and variance-stabilized matrix as data frame for loadig in to WGCNA analysis script
```{r}
write.csv(as.data.frame(norm_vst_countdata), file = "./outputs/STARcounts_Pdam/AxH_Pdam.vst.counts")
```

# DESeq2 Results Contrasts
```{r}
resultsNames(dds_anti)
resultsNames(dds_heat)
resultsNames(dds_anti.heat)
resultsNames(dds_AxH)
```

##Overall model results contrasts between individual treatments and control samples
### Antibiotics vs. Control
```{r}
res_anti_ctrl <- results(dds_AxH, contrast = c("Treatment", "Antibiotics", "control"))
res_anti_ctrl$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_anti_ctrl$IDGeneInfo <- as.character(res_anti_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_anti_ctrl[order(res_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_anti_ctrl.csv")
ressig_anti_ctrl <- subset(res_anti_ctrl, padj<0.01)
ressig_anti_ctrl_up <- subset(res_anti_ctrl, padj<0.01 & log2FoldChange>0)
ressig_anti_ctrl_dn <- subset(res_anti_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_anti_ctrl[order(ressig_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_anti_ctrl.csv")
write.csv(as.data.frame(ressig_anti_ctrl_up[order(ressig_anti_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_anti_ctrl_up.csv")
write.csv(as.data.frame(ressig_anti_ctrl_dn[order(ressig_anti_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_anti_ctrl_dn.csv")
colnames(res_anti_ctrl)
summary(res_anti_ctrl)
```

### Heat vs. Control
```{r}
res_heat_ctrl <- results(dds_AxH, contrast = c("Treatment", "Heat", "control"))
res_heat_ctrl$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_heat_ctrl$IDGeneInfo <- as.character(res_heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_heat_ctrl[order(res_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_heat_ctrl.csv")
ressig_heat_ctrl <- subset(res_heat_ctrl, padj<0.01)
ressig_heat_ctrl_up <- subset(res_heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_heat_ctrl_dn <- subset(res_heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_heat_ctrl[order(ressig_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_heat_ctrl.csv")
write.csv(as.data.frame(ressig_heat_ctrl_up[order(ressig_heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_heat_ctrl_dn[order(ressig_heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_heat_ctrl_dn.csv")
colnames(res_heat_ctrl)
summary(res_heat_ctrl)
```

### Antibiotics.Heat vs. Control
```{r}
res_anti.heat_ctrl <- results(dds_AxH, contrast = c("Treatment", "Antibiotics.Heat", "control"))
res_anti.heat_ctrl$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_anti.heat_ctrl$IDGeneInfo <- as.character(res_anti.heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_anti.heat_ctrl[order(res_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_anti.heat_ctrl.csv")
ressig_anti.heat_ctrl <- subset(res_anti.heat_ctrl, padj<0.01)
ressig_anti.heat_ctrl_up <- subset(res_anti.heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_anti.heat_ctrl_dn <- subset(res_anti.heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_anti.heat_ctrl[order(ressig_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_anti.heat_ctrl.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_up[order(ressig_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_anti.heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_dn[order(ressig_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_anti.heat_ctrl_dn.csv")
colnames(res_anti.heat_ctrl)
summary(res_anti.heat_ctrl)
```

### Antibiotics.Heat vs. Antibiotics
```{r}
res_anti.heat_anti <- results(dds_AxH, contrast = c("Treatment", "Antibiotics.Heat", "Antibiotics"))
res_anti.heat_anti$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_anti.heat_anti$IDGeneInfo <- as.character(res_anti.heat_anti$IDGeneInfo)
###
write.csv(as.data.frame(res_anti.heat_anti[order(res_anti.heat_anti$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_anti.heat_anti.csv")
ressig_anti.heat_ctrl <- subset(res_anti.heat_anti, padj<0.01)
ressig_anti.heat_ctrl_up <- subset(res_anti.heat_anti, padj<0.01 & log2FoldChange>0)
ressig_anti.heat_ctrl_dn <- subset(res_anti.heat_anti, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_anti.heat_ctrl[order(ressig_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_anti.heat_anti.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_up[order(ressig_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_anti.heat_anti_up.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_dn[order(ressig_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_anti.heat_anti_dn.csv")
colnames(res_anti.heat_anti)
summary(res_anti.heat_anti)
```

### Antibiotics.Heat vs. Heat
```{r}
res_anti.heat_heat <- results(dds_AxH, contrast = c("Treatment", "Antibiotics.Heat", "Heat"))
res_anti.heat_heat$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_anti.heat_heat$IDGeneInfo <- as.character(res_anti.heat_heat$IDGeneInfo)
###
write.csv(as.data.frame(res_anti.heat_heat[order(res_anti.heat_heat$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_anti.heat_heat.csv")
ressig_anti.heat_ctrl <- subset(res_anti.heat_heat, padj<0.01)
ressig_anti.heat_ctrl_up <- subset(res_anti.heat_heat, padj<0.01 & log2FoldChange>0)
ressig_anti.heat_ctrl_dn <- subset(res_anti.heat_heat, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_anti.heat_ctrl[order(ressig_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_anti.heat_heat.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_up[order(ressig_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_anti.heat_heat_up.csv")
write.csv(as.data.frame(ressig_anti.heat_ctrl_dn[order(ressig_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_anti.heat_heat_dn.csv")
colnames(res_anti.heat_heat)
summary(res_anti.heat_heat)
```

## Results contrasts between different coral genotypes
### Houwan1 vs. Houwan2
```{r}
res_HW1_HW2 <- results(dds_AxH, contrast = c("Colony", "HW1", "HW2"))
res_HW1_HW2$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_HW1_HW2$IDGeneInfo <- as.character(res_HW1_HW2$IDGeneInfo)
write.csv(as.data.frame(res_HW1_HW2[order(res_HW1_HW2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_HW2.csv")
ressig_HW1_HW2 <- subset(res_HW1_HW2, padj<0.01)
ressig_HW1_HW2_up <- subset(res_HW1_HW2, padj<0.01 & log2FoldChange>0)
ressig_HW1_HW2_dn <- subset(res_HW1_HW2, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_HW2[order(ressig_HW1_HW2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_HW2.csv")
write.csv(as.data.frame(ressig_HW1_HW2_up[order(ressig_HW1_HW2_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_HW2_up.csv")
write.csv(as.data.frame(ressig_HW1_HW2_dn[order(ressig_HW1_HW2_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_HW2_dn.csv")
colnames(res_HW1_HW2)
summary(res_HW1_HW2)
```

### Houwan1 vs. Wanglitung1
```{r}
res_HW1_WT1 <- results(dds_AxH, contrast = c("Colony", "HW1", "WT1"))
res_HW1_WT1$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_HW1_WT1$IDGeneInfo <- as.character(res_HW1_WT1$IDGeneInfo)
write.csv(as.data.frame(res_HW1_WT1[order(res_HW1_WT1$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_WT1.csv")
ressig_HW1_WT1 <- subset(res_HW1_WT1, padj<0.01)
ressig_HW1_WT1_up <- subset(res_HW1_WT1, padj<0.01 & log2FoldChange>0)
ressig_HW1_WT1_dn <- subset(res_HW1_WT1, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_WT1[order(ressig_HW1_WT1$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_WT1.csv")
write.csv(as.data.frame(ressig_HW1_WT1_up[order(ressig_HW1_WT1_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_WT1_up.csv")
write.csv(as.data.frame(ressig_HW1_WT1_dn[order(ressig_HW1_WT1_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_WT1_dn.csv")
colnames(res_HW1_WT1)
summary(res_HW1_WT1)
```

### Houwan1 vs. Wanglitung2
```{r}
res_HW1_WT2 <- results(dds_AxH, contrast = c("Colony", "HW1", "WT2"))
res_HW1_WT2$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_HW1_WT2$IDGeneInfo <- as.character(res_HW1_WT2$IDGeneInfo)
write.csv(as.data.frame(res_HW1_WT2[order(res_HW1_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_WT2.csv")
ressig_HW1_WT2 <- subset(res_HW1_WT2, padj<0.01)
ressig_HW1_WT2_up <- subset(res_HW1_WT2, padj<0.01 & log2FoldChange>0)
ressig_HW1_WT2_dn <- subset(res_HW1_WT2, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_WT2[order(ressig_HW1_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_WT2.csv")
write.csv(as.data.frame(ressig_HW1_WT2_up[order(ressig_HW1_WT2_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_WT2_up.csv")
write.csv(as.data.frame(ressig_HW1_WT2_dn[order(ressig_HW1_WT2_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_WT2_dn.csv")
colnames(res_HW1_WT2)
summary(res_HW1_WT2)
```

### Houwan2 vs. Wanglitung1
```{r}
res_HW2_WT1 <- results(dds_AxH, contrast = c("Colony", "HW2", "WT1"))
res_HW2_WT1$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_HW2_WT1$IDGeneInfo <- as.character(res_HW2_WT1$IDGeneInfo)
write.csv(as.data.frame(res_HW2_WT1[order(res_HW2_WT1$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_WT1.csv")
ressig_HW2_WT1 <- subset(res_HW2_WT1, padj<0.01)
ressig_HW2_WT1_up <- subset(res_HW2_WT1, padj<0.01 & log2FoldChange>0)
ressig_HW2_WT1_dn <- subset(res_HW2_WT1, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_WT1[order(ressig_HW2_WT1$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_WT1.csv")
write.csv(as.data.frame(ressig_HW2_WT1_up[order(ressig_HW2_WT1_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_WT1_up.csv")
write.csv(as.data.frame(ressig_HW2_WT1_dn[order(ressig_HW2_WT1_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_WT1_dn.csv")
colnames(res_HW2_WT1)
summary(res_HW2_WT1)
```

### Houwan2 vs. Wanglitung2
```{r}
res_HW2_WT2 <- results(dds_AxH, contrast = c("Colony", "HW2", "WT2"))
res_HW2_WT2$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_HW2_WT2$IDGeneInfo <- as.character(res_HW2_WT2$IDGeneInfo)
write.csv(as.data.frame(res_HW2_WT2[order(res_HW2_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_WT2.csv")
ressig_HW2_WT2 <- subset(res_HW2_WT2, padj<0.01)
ressig_HW2_WT2_up <- subset(res_HW2_WT2, padj<0.01 & log2FoldChange>0)
ressig_HW2_WT2_dn <- subset(res_HW2_WT2, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_WT2[order(ressig_HW2_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_WT2.csv")
write.csv(as.data.frame(ressig_HW2_WT2_up[order(ressig_HW2_WT2_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_WT2_up.csv")
write.csv(as.data.frame(ressig_HW2_WT2_dn[order(ressig_HW2_WT2_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_WT2_dn.csv")
colnames(res_HW2_WT2)
summary(res_HW2_WT2)
```

### Wanglitung1 vs. Wanglitung2
```{r}
res_WT1_WT2 <- results(dds_AxH, contrast = c("Colony", "WT1", "WT2"))
res_WT1_WT2$IDGeneInfo <- mcols(dds_AxH)$IDGeneInfo
res_WT1_WT2$IDGeneInfo <- as.character(res_WT1_WT2$IDGeneInfo)
write.csv(as.data.frame(res_WT1_WT2[order(res_WT1_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_WT2.csv")
ressig_WT1_WT2 <- subset(res_WT1_WT2, padj<0.01)
ressig_WT1_WT2_up <- subset(res_WT1_WT2, padj<0.01 & log2FoldChange>0)
ressig_WT1_WT2_dn <- subset(res_WT1_WT2, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_WT2[order(ressig_WT1_WT2$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_WT2.csv")
write.csv(as.data.frame(ressig_WT1_WT2_up[order(ressig_WT1_WT2_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_WT2_up.csv")
write.csv(as.data.frame(ressig_WT1_WT2_dn[order(ressig_WT1_WT2_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_WT2_dn.csv")
colnames(res_WT1_WT2)
summary(res_WT1_WT2)
```

## Colony-specific treatment contrasts results
### Houwan1
#### Antibiotics vs. Control
```{r}
res_HW1_anti_ctrl <- results(dds_HW1, contrast = c("Treatment", "Antibiotics", "control"))
res_HW1_anti_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW1_anti_ctrl$IDGeneInfo <- as.character(res_HW1_anti_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW1_anti_ctrl[order(res_HW1_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_anti_ctrl.csv")
ressig_HW1_anti_ctrl <- subset(res_HW1_anti_ctrl, padj<0.01)
ressig_HW1_anti_ctrl_up <- subset(res_HW1_anti_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW1_anti_ctrl_dn <- subset(res_HW1_anti_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_anti_ctrl[order(ressig_HW1_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_anti_ctrl.csv")
write.csv(as.data.frame(ressig_HW1_anti_ctrl_up[order(ressig_HW1_anti_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_anti_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW1_anti_ctrl_dn[order(ressig_HW1_anti_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_anti_ctrl_dn.csv")
colnames(res_HW1_anti_ctrl)
summary(res_HW1_anti_ctrl)
```

#### Heat vs. Control
```{r}
res_HW1_heat_ctrl <- results(dds_HW1, contrast = c("Treatment", "Heat", "control"))
res_HW1_heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW1_heat_ctrl$IDGeneInfo <- as.character(res_HW1_heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW1_heat_ctrl[order(res_HW1_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_heat_ctrl.csv")
ressig_HW1_heat_ctrl <- subset(res_HW1_heat_ctrl, padj<0.01)
ressig_HW1_heat_ctrl_up <- subset(res_HW1_heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW1_heat_ctrl_dn <- subset(res_HW1_heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_heat_ctrl[order(ressig_HW1_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_heat_ctrl.csv")
write.csv(as.data.frame(ressig_HW1_heat_ctrl_up[order(ressig_HW1_heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW1_heat_ctrl_dn[order(ressig_HW1_heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_heat_ctrl_dn.csv")
colnames(res_HW1_heat_ctrl)
summary(res_HW1_heat_ctrl)
```

#### Antibiotics.Heat vs. Control
```{r}
res_HW1_anti.heat_ctrl <- results(dds_HW1, contrast = c("Treatment", "Antibiotics.Heat", "control"))
res_HW1_anti.heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW1_anti.heat_ctrl$IDGeneInfo <- as.character(res_HW1_anti.heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW1_anti.heat_ctrl[order(res_HW1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_anti.heat_ctrl.csv")
ressig_HW1_anti.heat_ctrl <- subset(res_HW1_anti.heat_ctrl, padj<0.01)
ressig_HW1_anti.heat_ctrl_up <- subset(res_HW1_anti.heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW1_anti.heat_ctrl_dn <- subset(res_HW1_anti.heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl[order(ressig_HW1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_anti.heat_ctrl.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_up[order(ressig_HW1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_anti.heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_dn[order(ressig_HW1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_anti.heat_ctrl_dn.csv")
colnames(res_HW1_anti.heat_ctrl)
summary(res_HW1_anti.heat_ctrl)
```

#### Antibiotics.Heat vs. Antibiotics
```{r}
res_HW1_anti.heat_anti <- results(dds_HW1, contrast = c("Treatment", "Antibiotics.Heat", "Antibiotics"))
res_HW1_anti.heat_anti$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW1_anti.heat_anti$IDGeneInfo <- as.character(res_HW1_anti.heat_anti$IDGeneInfo)
###
write.csv(as.data.frame(res_HW1_anti.heat_anti[order(res_HW1_anti.heat_anti$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_anti.heat_anti.csv")
ressig_HW1_anti.heat_ctrl <- subset(res_HW1_anti.heat_anti, padj<0.01)
ressig_HW1_anti.heat_ctrl_up <- subset(res_HW1_anti.heat_anti, padj<0.01 & log2FoldChange>0)
ressig_HW1_anti.heat_ctrl_dn <- subset(res_HW1_anti.heat_anti, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl[order(ressig_HW1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_anti.heat_anti.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_up[order(ressig_HW1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_anti.heat_anti_up.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_dn[order(ressig_HW1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_anti.heat_anti_dn.csv")
colnames(res_HW1_anti.heat_anti)
summary(res_HW1_anti.heat_anti)
```

#### Antibiotics.Heat vs. Heat
```{r}
res_HW1_anti.heat_heat <- results(dds_HW1, contrast = c("Treatment", "Antibiotics.Heat", "Heat"))
res_HW1_anti.heat_heat$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW1_anti.heat_heat$IDGeneInfo <- as.character(res_HW1_anti.heat_heat$IDGeneInfo)
###
write.csv(as.data.frame(res_HW1_anti.heat_heat[order(res_HW1_anti.heat_heat$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW1_anti.heat_heat.csv")
ressig_HW1_anti.heat_ctrl <- subset(res_HW1_anti.heat_heat, padj<0.01)
ressig_HW1_anti.heat_ctrl_up <- subset(res_HW1_anti.heat_heat, padj<0.01 & log2FoldChange>0)
ressig_HW1_anti.heat_ctrl_dn <- subset(res_HW1_anti.heat_heat, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl[order(ressig_HW1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW1_anti.heat_heat.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_up[order(ressig_HW1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW1_anti.heat_heat_up.csv")
write.csv(as.data.frame(ressig_HW1_anti.heat_ctrl_dn[order(ressig_HW1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW1_anti.heat_heat_dn.csv")
colnames(res_HW1_anti.heat_heat)
summary(res_HW1_anti.heat_heat)
```

### Houwan2
#### Antibiotics vs. Control
```{r}
res_HW2_anti_ctrl <- results(dds_HW2, contrast = c("Treatment", "Antibiotics", "control"))
res_HW2_anti_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW2_anti_ctrl$IDGeneInfo <- as.character(res_HW2_anti_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW2_anti_ctrl[order(res_HW2_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_anti_ctrl.csv")
ressig_HW2_anti_ctrl <- subset(res_HW2_anti_ctrl, padj<0.01)
ressig_HW2_anti_ctrl_up <- subset(res_HW2_anti_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW2_anti_ctrl_dn <- subset(res_HW2_anti_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_anti_ctrl[order(ressig_HW2_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_anti_ctrl.csv")
write.csv(as.data.frame(ressig_HW2_anti_ctrl_up[order(ressig_HW2_anti_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_anti_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW2_anti_ctrl_dn[order(ressig_HW2_anti_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_anti_ctrl_dn.csv")
colnames(res_HW2_anti_ctrl)
summary(res_HW2_anti_ctrl)
```

#### Heat vs. Control
```{r}
res_HW2_heat_ctrl <- results(dds_HW2, contrast = c("Treatment", "Heat", "control"))
res_HW2_heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW2_heat_ctrl$IDGeneInfo <- as.character(res_HW2_heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW2_heat_ctrl[order(res_HW2_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_heat_ctrl.csv")
ressig_HW2_heat_ctrl <- subset(res_HW2_heat_ctrl, padj<0.01)
ressig_HW2_heat_ctrl_up <- subset(res_HW2_heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW2_heat_ctrl_dn <- subset(res_HW2_heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_heat_ctrl[order(ressig_HW2_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_heat_ctrl.csv")
write.csv(as.data.frame(ressig_HW2_heat_ctrl_up[order(ressig_HW2_heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW2_heat_ctrl_dn[order(ressig_HW2_heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_heat_ctrl_dn.csv")
colnames(res_HW2_heat_ctrl)
summary(res_HW2_heat_ctrl)
```

#### Antibiotics.Heat vs. Control
```{r}
res_HW2_anti.heat_ctrl <- results(dds_HW2, contrast = c("Treatment", "Antibiotics.Heat", "control"))
res_HW2_anti.heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_HW2_anti.heat_ctrl$IDGeneInfo <- as.character(res_HW2_anti.heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_HW2_anti.heat_ctrl[order(res_HW2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_anti.heat_ctrl.csv")
ressig_HW2_anti.heat_ctrl <- subset(res_HW2_anti.heat_ctrl, padj<0.01)
ressig_HW2_anti.heat_ctrl_up <- subset(res_HW2_anti.heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_HW2_anti.heat_ctrl_dn <- subset(res_HW2_anti.heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl[order(ressig_HW2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_anti.heat_ctrl.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_up[order(ressig_HW2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_anti.heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_dn[order(ressig_HW2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_anti.heat_ctrl_dn.csv")
colnames(res_HW2_anti.heat_ctrl)
summary(res_HW2_anti.heat_ctrl)
```

#### Antibiotics.Heat vs. Antibiotics
```{r}
res_HW2_anti.heat_anti <- results(dds_HW2, contrast = c("Treatment", "Antibiotics.Heat", "Antibiotics"))
res_HW2_anti.heat_anti$IDGeneInfo <- mcols(dds_HW2)$IDGeneInfo
res_HW2_anti.heat_anti$IDGeneInfo <- as.character(res_HW2_anti.heat_anti$IDGeneInfo)
###
write.csv(as.data.frame(res_HW2_anti.heat_anti[order(res_HW2_anti.heat_anti$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_anti.heat_anti.csv")
ressig_HW2_anti.heat_ctrl <- subset(res_HW2_anti.heat_anti, padj<0.01)
ressig_HW2_anti.heat_ctrl_up <- subset(res_HW2_anti.heat_anti, padj<0.01 & log2FoldChange>0)
ressig_HW2_anti.heat_ctrl_dn <- subset(res_HW2_anti.heat_anti, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl[order(ressig_HW2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_anti.heat_anti.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_up[order(ressig_HW2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_anti.heat_anti_up.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_dn[order(ressig_HW2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_anti.heat_anti_dn.csv")
colnames(res_HW2_anti.heat_anti)
summary(res_HW2_anti.heat_anti)
```

#### Antibiotics.Heat vs. Heat
```{r}
res_HW2_anti.heat_heat <- results(dds_HW2, contrast = c("Treatment", "Antibiotics.Heat", "Heat"))
res_HW2_anti.heat_heat$IDGeneInfo <- mcols(dds_HW2)$IDGeneInfo
res_HW2_anti.heat_heat$IDGeneInfo <- as.character(res_HW2_anti.heat_heat$IDGeneInfo)
###
write.csv(as.data.frame(res_HW2_anti.heat_heat[order(res_HW2_anti.heat_heat$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_HW2_anti.heat_heat.csv")
ressig_HW2_anti.heat_ctrl <- subset(res_HW2_anti.heat_heat, padj<0.01)
ressig_HW2_anti.heat_ctrl_up <- subset(res_HW2_anti.heat_heat, padj<0.01 & log2FoldChange>0)
ressig_HW2_anti.heat_ctrl_dn <- subset(res_HW2_anti.heat_heat, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl[order(ressig_HW2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_HW2_anti.heat_heat.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_up[order(ressig_HW2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_HW2_anti.heat_heat_up.csv")
write.csv(as.data.frame(ressig_HW2_anti.heat_ctrl_dn[order(ressig_HW2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_HW2_anti.heat_heat_dn.csv")
colnames(res_HW2_anti.heat_heat)
summary(res_HW2_anti.heat_heat)
```

### Wanglitung1
#### Antibiotics vs. Control
```{r}
res_WT1_anti_ctrl <- results(dds_WT1, contrast = c("Treatment", "Antibiotics", "control"))
res_WT1_anti_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT1_anti_ctrl$IDGeneInfo <- as.character(res_WT1_anti_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT1_anti_ctrl[order(res_WT1_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_anti_ctrl.csv")
ressig_WT1_anti_ctrl <- subset(res_WT1_anti_ctrl, padj<0.01)
ressig_WT1_anti_ctrl_up <- subset(res_WT1_anti_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT1_anti_ctrl_dn <- subset(res_WT1_anti_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_anti_ctrl[order(ressig_WT1_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_anti_ctrl.csv")
write.csv(as.data.frame(ressig_WT1_anti_ctrl_up[order(ressig_WT1_anti_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_anti_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT1_anti_ctrl_dn[order(ressig_WT1_anti_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_anti_ctrl_dn.csv")
colnames(res_WT1_anti_ctrl)
summary(res_WT1_anti_ctrl)
```

#### Heat vs. Control
```{r}
res_WT1_heat_ctrl <- results(dds_WT1, contrast = c("Treatment", "Heat", "control"))
res_WT1_heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT1_heat_ctrl$IDGeneInfo <- as.character(res_WT1_heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT1_heat_ctrl[order(res_WT1_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_heat_ctrl.csv")
ressig_WT1_heat_ctrl <- subset(res_WT1_heat_ctrl, padj<0.01)
ressig_WT1_heat_ctrl_up <- subset(res_WT1_heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT1_heat_ctrl_dn <- subset(res_WT1_heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_heat_ctrl[order(ressig_WT1_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_heat_ctrl.csv")
write.csv(as.data.frame(ressig_WT1_heat_ctrl_up[order(ressig_WT1_heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT1_heat_ctrl_dn[order(ressig_WT1_heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_heat_ctrl_dn.csv")
colnames(res_WT1_heat_ctrl)
summary(res_WT1_heat_ctrl)
```

#### Antibiotics.Heat vs. Control
```{r}
res_WT1_anti.heat_ctrl <- results(dds_WT1, contrast = c("Treatment", "Antibiotics.Heat", "control"))
res_WT1_anti.heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT1_anti.heat_ctrl$IDGeneInfo <- as.character(res_WT1_anti.heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT1_anti.heat_ctrl[order(res_WT1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_anti.heat_ctrl.csv")
ressig_WT1_anti.heat_ctrl <- subset(res_WT1_anti.heat_ctrl, padj<0.01)
ressig_WT1_anti.heat_ctrl_up <- subset(res_WT1_anti.heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT1_anti.heat_ctrl_dn <- subset(res_WT1_anti.heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl[order(ressig_WT1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_anti.heat_ctrl.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_up[order(ressig_WT1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_anti.heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_dn[order(ressig_WT1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_anti.heat_ctrl_dn.csv")
colnames(res_WT1_anti.heat_ctrl)
summary(res_WT1_anti.heat_ctrl)
```

#### Antibiotics.Heat vs. Antibiotics
```{r}
res_WT1_anti.heat_anti <- results(dds_WT1, contrast = c("Treatment", "Antibiotics.Heat", "Antibiotics"))
res_WT1_anti.heat_anti$IDGeneInfo <- mcols(dds_WT1)$IDGeneInfo
res_WT1_anti.heat_anti$IDGeneInfo <- as.character(res_WT1_anti.heat_anti$IDGeneInfo)
###
write.csv(as.data.frame(res_WT1_anti.heat_anti[order(res_WT1_anti.heat_anti$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_anti.heat_anti.csv")
ressig_WT1_anti.heat_ctrl <- subset(res_WT1_anti.heat_anti, padj<0.01)
ressig_WT1_anti.heat_ctrl_up <- subset(res_WT1_anti.heat_anti, padj<0.01 & log2FoldChange>0)
ressig_WT1_anti.heat_ctrl_dn <- subset(res_WT1_anti.heat_anti, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl[order(ressig_WT1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_anti.heat_anti.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_up[order(ressig_WT1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_anti.heat_anti_up.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_dn[order(ressig_WT1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_anti.heat_anti_dn.csv")
colnames(res_WT1_anti.heat_anti)
summary(res_WT1_anti.heat_anti)
```

#### Antibiotics.Heat vs. Heat
```{r}
res_WT1_anti.heat_heat <- results(dds_WT1, contrast = c("Treatment", "Antibiotics.Heat", "Heat"))
res_WT1_anti.heat_heat$IDGeneInfo <- mcols(dds_WT1)$IDGeneInfo
res_WT1_anti.heat_heat$IDGeneInfo <- as.character(res_WT1_anti.heat_heat$IDGeneInfo)
###
write.csv(as.data.frame(res_WT1_anti.heat_heat[order(res_WT1_anti.heat_heat$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT1_anti.heat_heat.csv")
ressig_WT1_anti.heat_ctrl <- subset(res_WT1_anti.heat_heat, padj<0.01)
ressig_WT1_anti.heat_ctrl_up <- subset(res_WT1_anti.heat_heat, padj<0.01 & log2FoldChange>0)
ressig_WT1_anti.heat_ctrl_dn <- subset(res_WT1_anti.heat_heat, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl[order(ressig_WT1_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT1_anti.heat_heat.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_up[order(ressig_WT1_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT1_anti.heat_heat_up.csv")
write.csv(as.data.frame(ressig_WT1_anti.heat_ctrl_dn[order(ressig_WT1_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT1_anti.heat_heat_dn.csv")
colnames(res_WT1_anti.heat_heat)
summary(res_WT1_anti.heat_heat)
```

### Wanglitung2
#### Antibiotics vs. Control
```{r}
res_WT2_anti_ctrl <- results(dds_WT2, contrast = c("Treatment", "Antibiotics", "control"))
res_WT2_anti_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT2_anti_ctrl$IDGeneInfo <- as.character(res_WT2_anti_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT2_anti_ctrl[order(res_WT2_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT2_anti_ctrl.csv")
ressig_WT2_anti_ctrl <- subset(res_WT2_anti_ctrl, padj<0.01)
ressig_WT2_anti_ctrl_up <- subset(res_WT2_anti_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT2_anti_ctrl_dn <- subset(res_WT2_anti_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT2_anti_ctrl[order(ressig_WT2_anti_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT2_anti_ctrl.csv")
write.csv(as.data.frame(ressig_WT2_anti_ctrl_up[order(ressig_WT2_anti_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT2_anti_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT2_anti_ctrl_dn[order(ressig_WT2_anti_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT2_anti_ctrl_dn.csv")
colnames(res_WT2_anti_ctrl)
summary(res_WT2_anti_ctrl)
```

#### Heat vs. Control
```{r}
res_WT2_heat_ctrl <- results(dds_WT2, contrast = c("Treatment", "Heat", "control"))
res_WT2_heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT2_heat_ctrl$IDGeneInfo <- as.character(res_WT2_heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT2_heat_ctrl[order(res_WT2_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT2_heat_ctrl.csv")
ressig_WT2_heat_ctrl <- subset(res_WT2_heat_ctrl, padj<0.01)
ressig_WT2_heat_ctrl_up <- subset(res_WT2_heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT2_heat_ctrl_dn <- subset(res_WT2_heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT2_heat_ctrl[order(ressig_WT2_heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT2_heat_ctrl.csv")
write.csv(as.data.frame(ressig_WT2_heat_ctrl_up[order(ressig_WT2_heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT2_heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT2_heat_ctrl_dn[order(ressig_WT2_heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT2_heat_ctrl_dn.csv")
colnames(res_WT2_heat_ctrl)
summary(res_WT2_heat_ctrl)
```

#### Antibiotics.Heat vs. Control
```{r}
res_WT2_anti.heat_ctrl <- results(dds_WT2, contrast = c("Treatment", "Antibiotics.Heat", "control"))
res_WT2_anti.heat_ctrl$IDGeneInfo <- mcols(dds_HW1)$IDGeneInfo
res_WT2_anti.heat_ctrl$IDGeneInfo <- as.character(res_WT2_anti.heat_ctrl$IDGeneInfo)
###
write.csv(as.data.frame(res_WT2_anti.heat_ctrl[order(res_WT2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT2_anti.heat_ctrl.csv")
ressig_WT2_anti.heat_ctrl <- subset(res_WT2_anti.heat_ctrl, padj<0.01)
ressig_WT2_anti.heat_ctrl_up <- subset(res_WT2_anti.heat_ctrl, padj<0.01 & log2FoldChange>0)
ressig_WT2_anti.heat_ctrl_dn <- subset(res_WT2_anti.heat_ctrl, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl[order(ressig_WT2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT2_anti.heat_ctrl.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_up[order(ressig_WT2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT2_anti.heat_ctrl_up.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_dn[order(ressig_WT2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT2_anti.heat_ctrl_dn.csv")
colnames(res_WT2_anti.heat_ctrl)
summary(res_WT2_anti.heat_ctrl)
```

#### Antibiotics.Heat vs. Antibiotics
```{r}
res_WT2_anti.heat_anti <- results(dds_WT2, contrast = c("Treatment", "Antibiotics.Heat", "Antibiotics"))
res_WT2_anti.heat_anti$IDGeneInfo <- mcols(dds_WT2)$IDGeneInfo
res_WT2_anti.heat_anti$IDGeneInfo <- as.character(res_WT2_anti.heat_anti$IDGeneInfo)
###
write.csv(as.data.frame(res_WT2_anti.heat_anti[order(res_WT2_anti.heat_anti$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT2_anti.heat_anti.csv")
ressig_WT2_anti.heat_ctrl <- subset(res_WT2_anti.heat_anti, padj<0.01)
ressig_WT2_anti.heat_ctrl_up <- subset(res_WT2_anti.heat_anti, padj<0.01 & log2FoldChange>0)
ressig_WT2_anti.heat_ctrl_dn <- subset(res_WT2_anti.heat_anti, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl[order(ressig_WT2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT2_anti.heat_anti.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_up[order(ressig_WT2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT2_anti.heat_anti_up.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_dn[order(ressig_WT2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT2_anti.heat_anti_dn.csv")
colnames(res_WT2_anti.heat_anti)
summary(res_WT2_anti.heat_anti)
```

#### Antibiotics.Heat vs. Heat
```{r}
res_WT2_anti.heat_heat <- results(dds_WT2, contrast = c("Treatment", "Antibiotics.Heat", "Heat"))
res_WT2_anti.heat_heat$IDGeneInfo <- mcols(dds_WT2)$IDGeneInfo
res_WT2_anti.heat_heat$IDGeneInfo <- as.character(res_WT2_anti.heat_heat$IDGeneInfo)
###
write.csv(as.data.frame(res_WT2_anti.heat_heat[order(res_WT2_anti.heat_heat$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/all/res_WT2_anti.heat_heat.csv")
ressig_WT2_anti.heat_ctrl <- subset(res_WT2_anti.heat_heat, padj<0.01)
ressig_WT2_anti.heat_ctrl_up <- subset(res_WT2_anti.heat_heat, padj<0.01 & log2FoldChange>0)
ressig_WT2_anti.heat_ctrl_dn <- subset(res_WT2_anti.heat_heat, padj<0.01 & log2FoldChange<0)
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl[order(ressig_WT2_anti.heat_ctrl$padj, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/overall/res_WT2_anti.heat_heat.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_up[order(ressig_WT2_anti.heat_ctrl_up$log2FoldChange, decreasing = TRUE),]), file = "./outputs/DESeq-results/tables/up/res_WT2_anti.heat_heat_up.csv")
write.csv(as.data.frame(ressig_WT2_anti.heat_ctrl_dn[order(ressig_WT2_anti.heat_ctrl_dn$log2FoldChange, decreasing = FALSE),]), file = "./outputs/DESeq-results/tables/down/res_WT2_anti.heat_heat_dn.csv")
colnames(res_WT2_anti.heat_heat)
summary(res_WT2_anti.heat_heat)
```

# Volcano Plots
### Antibiotics vs. Control Differential Expression
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_anti_ctrl$threshold <- ifelse(res_anti_ctrl$padj < 0.01 & res_anti_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_anti_ctrl$padj < 0.01 & res_anti_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_anti_ctrl$log10padj <- -log10(res_anti_ctrl$padj)
dat_genes_anti_ctrl_ctrl <- data.frame(cbind(res_anti_ctrl$log2FoldChange, res_anti_ctrl$log10padj, res_anti_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_anti_ctrl_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_anti_ctrl_ctrl) <- res_anti_ctrl$IDGeneInfo
#dat_genes_anti_ctrl_ctrl <- dat_genes_anti_ctrl_ctrl[order(dat_genes_anti_ctrl_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_anti_ctrl_ctrl$log2FoldChange <- as.numeric(dat_genes_anti_ctrl_ctrl$log2FoldChange)
dat_genes_anti_ctrl_ctrl$log10padj <- as.numeric(dat_genes_anti_ctrl_ctrl$log10padj)
dat_genes_anti_ctrl_ctrl$threshold <- factor(dat_genes_anti_ctrl_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```
```{r, warning = FALSE}
pdf(file = "./outputs/DESeq-results/figures/Volcano_anti_ctrl-ctrl.pdf", height = 3, width = 4.25)
gVolcano_anti_ctrl_ctrl <- ggplot(data=dat_genes_anti_ctrl_ctrl, aes(log2FoldChange, log10padj))
gVolcano_anti_ctrl_ctrl + geom_point(aes(color = threshold), alpha=0.7, size=2) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-15,15), breaks = seq(-15,15,5)) + 
  ylim(c(0, 30)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_anti_ctrl_ctrl[1:15, ], aes(label = rownames(dat_genes_anti_ctrl_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#99FFFF", 0.5)),
        panel.grid.major = element_line(size = 0.1, linetype = "solid", color = "dark grey"),
        panel.grid.minor = element_blank()) +
  ggtitle("Antibiotics vs. Control")
```

### Heat vs. Control Differential Expression
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_heat_ctrl$threshold <- ifelse(res_heat_ctrl$padj < 0.01 & res_heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_heat_ctrl$padj < 0.01 & res_heat_ctrl$log2FoldChange < -0, "Downregulated", "NA"))
res_heat_ctrl$log10padj <- -log10(res_heat_ctrl$padj)
dat_genes_heat_ctrl_ctrl <- data.frame(cbind(res_heat_ctrl$log2FoldChange, res_heat_ctrl$log10padj, res_heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_heat_ctrl_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_heat_ctrl_ctrl) <- res_heat_ctrl$IDGeneInfo
#dat_genes_heat_ctrl_ctrl <- dat_genes_heat_ctrl_ctrl[order(dat_genes_heat_ctrl_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_heat_ctrl_ctrl$log2FoldChange <- as.numeric(dat_genes_heat_ctrl_ctrl$log2FoldChange)
dat_genes_heat_ctrl_ctrl$log10padj <- as.numeric(dat_genes_heat_ctrl_ctrl$log10padj)
dat_genes_heat_ctrl_ctrl$threshold <- factor(dat_genes_heat_ctrl_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```
```{r, warning = FALSE}
pdf(file = "./outputs/DESeq-results/figures/Volcano_heat_ctrl-ctrl.pdf", height = 3, width = 4.25)
gVolcano_heat_ctrl_ctrl <- ggplot(data=dat_genes_heat_ctrl_ctrl, aes(log2FoldChange, log10padj))
gVolcano_heat_ctrl_ctrl + geom_point(aes(color = threshold), alpha=0.7, size=2) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-15,15), breaks = seq(-15,15,5)) + 
  ylim(c(0, 30)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_heat_ctrl_ctrl[1:15, ], aes(label = rownames(dat_genes_heat_ctrl_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFCC99", 0.5)),
        panel.grid.major = element_line(size = 0.1, linetype = "solid", color = "dark grey"),
        panel.grid.minor = element_blank()) +
  ggtitle("Heat vs. Control")
```

### Antibiotics.Heat vs. Control Differential Expression
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_anti.heat_ctrl$threshold <- ifelse(res_anti.heat_ctrl$padj < 0.01 & res_anti.heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_anti.heat_ctrl$padj < 0.01 & res_anti.heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_anti.heat_ctrl$log10padj <- -log10(res_anti.heat_ctrl$padj)
dat_genes_anti.heat_ctrl_ctrl <- data.frame(cbind(res_anti.heat_ctrl$log2FoldChange, res_anti.heat_ctrl$log10padj, res_anti.heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_anti.heat_ctrl_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_anti.heat_ctrl_ctrl) <- res_anti.heat_ctrl$IDGeneInfo
#dat_genes_anti.heat_ctrl_ctrl <- dat_genes_anti.heat_ctrl_ctrl[order(dat_genes_anti.heat_ctrl_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_anti.heat_ctrl_ctrl$log2FoldChange <- as.numeric(dat_genes_anti.heat_ctrl_ctrl$log2FoldChange)
dat_genes_anti.heat_ctrl_ctrl$log10padj <- as.numeric(dat_genes_anti.heat_ctrl_ctrl$log10padj)
dat_genes_anti.heat_ctrl_ctrl$threshold <- factor(dat_genes_anti.heat_ctrl_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```
```{r, warning = FALSE}
pdf(file = "./outputs/DESeq-results/figures/Volcano_anti.heat_ctrl-ctrl.pdf", height = 3, width = 4.25)
gVolcano_anti.heat_ctrl_ctrl <- ggplot(data=dat_genes_anti.heat_ctrl_ctrl, aes(log2FoldChange, log10padj))
gVolcano_anti.heat_ctrl_ctrl + geom_point(aes(color = threshold), alpha=0.7, size=2) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-15,15), breaks = seq(-15,15,5)) + 
  ylim(c(0, 30)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_anti.heat_ctrl_ctrl[1:15, ], aes(label = rownames(dat_genes_anti.heat_ctrl_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFFF66", 0.5)),
        panel.grid.major = element_line(size = 0.1, linetype = "solid", color = "dark grey"),
        panel.grid.minor = element_blank()) +
  ggtitle("Antibiotics + Heat vs. Control")
```

### Antibiotics vs. Control Differential Expression for each colony
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_HW1_anti_ctrl$threshold <- ifelse(res_HW1_anti_ctrl$padj < 0.01 & res_HW1_anti_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW1_anti_ctrl$padj < 0.01 & res_HW1_anti_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW1_anti_ctrl$log10padj <- -log10(res_HW1_anti_ctrl$padj)
dat_genes_HW1_anti_ctrl <- data.frame(cbind(res_HW1_anti_ctrl$log2FoldChange, res_HW1_anti_ctrl$log10padj, res_HW1_anti_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW1_anti_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW1_anti_ctrl) <- res_HW1_anti_ctrl$IDGeneInfo
#dat_genes_HW1_anti_ctrl <- dat_genes_HW1_anti_ctrl[order(dat_genes_HW1_anti_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW1_anti_ctrl$log2FoldChange <- as.numeric(dat_genes_HW1_anti_ctrl$log2FoldChange)
dat_genes_HW1_anti_ctrl$log10padj <- as.numeric(dat_genes_HW1_anti_ctrl$log10padj)
dat_genes_HW1_anti_ctrl$threshold <- factor(dat_genes_HW1_anti_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_HW2_anti_ctrl$threshold <- ifelse(res_HW2_anti_ctrl$padj < 0.01 & res_HW2_anti_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW2_anti_ctrl$padj < 0.01 & res_HW2_anti_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW2_anti_ctrl$log10padj <- -log10(res_HW2_anti_ctrl$padj)
dat_genes_HW2_anti_ctrl <- data.frame(cbind(res_HW2_anti_ctrl$log2FoldChange, res_HW2_anti_ctrl$log10padj, res_HW2_anti_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW2_anti_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW2_anti_ctrl) <- res_HW2_anti_ctrl$IDGeneInfo
#dat_genes_HW2_anti_ctrl <- dat_genes_HW2_anti_ctrl[order(dat_genes_HW2_anti_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW2_anti_ctrl$log2FoldChange <- as.numeric(dat_genes_HW2_anti_ctrl$log2FoldChange)
dat_genes_HW2_anti_ctrl$log10padj <- as.numeric(dat_genes_HW2_anti_ctrl$log10padj)
dat_genes_HW2_anti_ctrl$threshold <- factor(dat_genes_HW2_anti_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT1_anti_ctrl$threshold <- ifelse(res_WT1_anti_ctrl$padj < 0.01 & res_WT1_anti_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT1_anti_ctrl$padj < 0.01 & res_WT1_anti_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT1_anti_ctrl$log10padj <- -log10(res_WT1_anti_ctrl$padj)
dat_genes_WT1_anti_ctrl <- data.frame(cbind(res_WT1_anti_ctrl$log2FoldChange, res_WT1_anti_ctrl$log10padj, res_WT1_anti_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT1_anti_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT1_anti_ctrl) <- res_WT1_anti_ctrl$IDGeneInfo
#dat_genes_WT1_anti_ctrl <- dat_genes_WT1_anti_ctrl[order(dat_genes_WT1_anti_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT1_anti_ctrl$log2FoldChange <- as.numeric(dat_genes_WT1_anti_ctrl$log2FoldChange)
dat_genes_WT1_anti_ctrl$log10padj <- as.numeric(dat_genes_WT1_anti_ctrl$log10padj)
dat_genes_WT1_anti_ctrl$threshold <- factor(dat_genes_WT1_anti_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT2_anti_ctrl$threshold <- ifelse(res_WT2_anti_ctrl$padj < 0.01 & res_WT2_anti_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT2_anti_ctrl$padj < 0.01 & res_WT2_anti_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT2_anti_ctrl$log10padj <- -log10(res_WT2_anti_ctrl$padj)
dat_genes_WT2_anti_ctrl <- data.frame(cbind(res_WT2_anti_ctrl$log2FoldChange, res_WT2_anti_ctrl$log10padj, res_WT2_anti_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT2_anti_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT2_anti_ctrl) <- res_WT2_anti_ctrl$IDGeneInfo
#dat_genes_WT2_anti_ctrl <- dat_genes_WT2_anti_ctrl[order(dat_genes_WT2_anti_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT2_anti_ctrl$log2FoldChange <- as.numeric(dat_genes_WT2_anti_ctrl$log2FoldChange)
dat_genes_WT2_anti_ctrl$log10padj <- as.numeric(dat_genes_WT2_anti_ctrl$log10padj)
dat_genes_WT2_anti_ctrl$threshold <- factor(dat_genes_WT2_anti_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```

```{r, warning = FALSE}
pdf(file = "./outputs/DESeq-results/figures/Volcano_anti_ctrl_colonies.pdf", height = 3, width = 4.25)
gVolcano_HW1_anti <- ggplot(data=dat_genes_HW1_anti_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW1_anti + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#99FFFF", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_HW2_anti <- ggplot(data=dat_genes_HW2_anti_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW2_anti + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#99FFFF", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT1_anti <- ggplot(data=dat_genes_WT1_anti_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT1_anti + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#99FFFF", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT2_anti <- ggplot(data=dat_genes_WT2_anti_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT2_anti + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#99FFFF", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))
```

### Heat vs. Control Differential Expression for each colony
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_HW1_heat_ctrl$threshold <- ifelse(res_HW1_heat_ctrl$padj < 0.01 & res_HW1_heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW1_heat_ctrl$padj < 0.01 & res_HW1_heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW1_heat_ctrl$log10padj <- -log10(res_HW1_heat_ctrl$padj)
dat_genes_HW1_heat_ctrl <- data.frame(cbind(res_HW1_heat_ctrl$log2FoldChange, res_HW1_heat_ctrl$log10padj, res_HW1_heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW1_heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW1_heat_ctrl) <- res_HW1_heat_ctrl$IDGeneInfo
#dat_genes_HW1_heat_ctrl <- dat_genes_HW1_heat_ctrl[order(dat_genes_HW1_heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW1_heat_ctrl$log2FoldChange <- as.numeric(dat_genes_HW1_heat_ctrl$log2FoldChange)
dat_genes_HW1_heat_ctrl$log10padj <- as.numeric(dat_genes_HW1_heat_ctrl$log10padj)
dat_genes_HW1_heat_ctrl$threshold <- factor(dat_genes_HW1_heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_HW2_heat_ctrl$threshold <- ifelse(res_HW2_heat_ctrl$padj < 0.01 & res_HW2_heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW2_heat_ctrl$padj < 0.01 & res_HW2_heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW2_heat_ctrl$log10padj <- -log10(res_HW2_heat_ctrl$padj)
dat_genes_HW2_heat_ctrl <- data.frame(cbind(res_HW2_heat_ctrl$log2FoldChange, res_HW2_heat_ctrl$log10padj, res_HW2_heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW2_heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW2_heat_ctrl) <- res_HW2_heat_ctrl$IDGeneInfo
#dat_genes_HW2_heat_ctrl <- dat_genes_HW2_heat_ctrl[order(dat_genes_HW2_heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW2_heat_ctrl$log2FoldChange <- as.numeric(dat_genes_HW2_heat_ctrl$log2FoldChange)
dat_genes_HW2_heat_ctrl$log10padj <- as.numeric(dat_genes_HW2_heat_ctrl$log10padj)
dat_genes_HW2_heat_ctrl$threshold <- factor(dat_genes_HW2_heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT1_heat_ctrl$threshold <- ifelse(res_WT1_heat_ctrl$padj < 0.01 & res_WT1_heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT1_heat_ctrl$padj < 0.01 & res_WT1_heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT1_heat_ctrl$log10padj <- -log10(res_WT1_heat_ctrl$padj)
dat_genes_WT1_heat_ctrl <- data.frame(cbind(res_WT1_heat_ctrl$log2FoldChange, res_WT1_heat_ctrl$log10padj, res_WT1_heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT1_heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT1_heat_ctrl) <- res_WT1_heat_ctrl$IDGeneInfo
#dat_genes_WT1_heat_ctrl <- dat_genes_WT1_heat_ctrl[order(dat_genes_WT1_heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT1_heat_ctrl$log2FoldChange <- as.numeric(dat_genes_WT1_heat_ctrl$log2FoldChange)
dat_genes_WT1_heat_ctrl$log10padj <- as.numeric(dat_genes_WT1_heat_ctrl$log10padj)
dat_genes_WT1_heat_ctrl$threshold <- factor(dat_genes_WT1_heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT2_heat_ctrl$threshold <- ifelse(res_WT2_heat_ctrl$padj < 0.01 & res_WT2_heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT2_heat_ctrl$padj < 0.01 & res_WT2_heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT2_heat_ctrl$log10padj <- -log10(res_WT2_heat_ctrl$padj)
dat_genes_WT2_heat_ctrl <- data.frame(cbind(res_WT2_heat_ctrl$log2FoldChange, res_WT2_heat_ctrl$log10padj, res_WT2_heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT2_heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT2_heat_ctrl) <- res_WT2_heat_ctrl$IDGeneInfo
#dat_genes_WT2_heat_ctrl <- dat_genes_WT2_heat_ctrl[order(dat_genes_WT2_heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT2_heat_ctrl$log2FoldChange <- as.numeric(dat_genes_WT2_heat_ctrl$log2FoldChange)
dat_genes_WT2_heat_ctrl$log10padj <- as.numeric(dat_genes_WT2_heat_ctrl$log10padj)
dat_genes_WT2_heat_ctrl$threshold <- factor(dat_genes_WT2_heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```

```{r, warning = FALSE}
DEGcolors <- c("red", "blue", "dark grey")
pdf(file = "./outputs/DESeq-results/figures/Volcano_heat_ctrl_colonies.pdf", height = 3, width = 4.25)
gVolcano_HW1_heat <- ggplot(data=dat_genes_HW1_heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW1_heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFCC99", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_HW2_heat <- ggplot(data=dat_genes_HW2_heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW2_heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFCC99", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT1_heat <- ggplot(data=dat_genes_WT1_heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT1_heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFCC99", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT2_heat <- ggplot(data=dat_genes_WT2_heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT2_heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFCC99", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))
```

### Antibiotics.Heat vs. Control Differential Expression for each colony
```{r}
##Highlight genes that have an absolute fold change > 2 and a padj < 0.01
res_HW1_anti.heat_ctrl$threshold <- ifelse(res_HW1_anti.heat_ctrl$padj < 0.01 & res_HW1_anti.heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW1_anti.heat_ctrl$padj < 0.01 & res_HW1_anti.heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW1_anti.heat_ctrl$log10padj <- -log10(res_HW1_anti.heat_ctrl$padj)
dat_genes_HW1_anti.heat_ctrl <- data.frame(cbind(res_HW1_anti.heat_ctrl$log2FoldChange, res_HW1_anti.heat_ctrl$log10padj, res_HW1_anti.heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW1_anti.heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW1_anti.heat_ctrl) <- res_HW1_anti.heat_ctrl$IDGeneInfo
#dat_genes_HW1_anti.heat_ctrl <- dat_genes_HW1_anti.heat_ctrl[order(dat_genes_HW1_anti.heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW1_anti.heat_ctrl$log2FoldChange <- as.numeric(dat_genes_HW1_anti.heat_ctrl$log2FoldChange)
dat_genes_HW1_anti.heat_ctrl$log10padj <- as.numeric(dat_genes_HW1_anti.heat_ctrl$log10padj)
dat_genes_HW1_anti.heat_ctrl$threshold <- factor(dat_genes_HW1_anti.heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_HW2_anti.heat_ctrl$threshold <- ifelse(res_HW2_anti.heat_ctrl$padj < 0.01 & res_HW2_anti.heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_HW2_anti.heat_ctrl$padj < 0.01 & res_HW2_anti.heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_HW2_anti.heat_ctrl$log10padj <- -log10(res_HW2_anti.heat_ctrl$padj)
dat_genes_HW2_anti.heat_ctrl <- data.frame(cbind(res_HW2_anti.heat_ctrl$log2FoldChange, res_HW2_anti.heat_ctrl$log10padj, res_HW2_anti.heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_HW2_anti.heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_HW2_anti.heat_ctrl) <- res_HW2_anti.heat_ctrl$IDGeneInfo
#dat_genes_HW2_anti.heat_ctrl <- dat_genes_HW2_anti.heat_ctrl[order(dat_genes_HW2_anti.heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_HW2_anti.heat_ctrl$log2FoldChange <- as.numeric(dat_genes_HW2_anti.heat_ctrl$log2FoldChange)
dat_genes_HW2_anti.heat_ctrl$log10padj <- as.numeric(dat_genes_HW2_anti.heat_ctrl$log10padj)
dat_genes_HW2_anti.heat_ctrl$threshold <- factor(dat_genes_HW2_anti.heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT1_anti.heat_ctrl$threshold <- ifelse(res_WT1_anti.heat_ctrl$padj < 0.01 & res_WT1_anti.heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT1_anti.heat_ctrl$padj < 0.01 & res_WT1_anti.heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT1_anti.heat_ctrl$log10padj <- -log10(res_WT1_anti.heat_ctrl$padj)
dat_genes_WT1_anti.heat_ctrl <- data.frame(cbind(res_WT1_anti.heat_ctrl$log2FoldChange, res_WT1_anti.heat_ctrl$log10padj, res_WT1_anti.heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT1_anti.heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT1_anti.heat_ctrl) <- res_WT1_anti.heat_ctrl$IDGeneInfo
#dat_genes_WT1_anti.heat_ctrl <- dat_genes_WT1_anti.heat_ctrl[order(dat_genes_WT1_anti.heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT1_anti.heat_ctrl$log2FoldChange <- as.numeric(dat_genes_WT1_anti.heat_ctrl$log2FoldChange)
dat_genes_WT1_anti.heat_ctrl$log10padj <- as.numeric(dat_genes_WT1_anti.heat_ctrl$log10padj)
dat_genes_WT1_anti.heat_ctrl$threshold <- factor(dat_genes_WT1_anti.heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

res_WT2_anti.heat_ctrl$threshold <- ifelse(res_WT2_anti.heat_ctrl$padj < 0.01 & res_WT2_anti.heat_ctrl$log2FoldChange > 0, "Upregulated", ifelse(res_WT2_anti.heat_ctrl$padj < 0.01 & res_WT2_anti.heat_ctrl$log2FoldChange < 0, "Downregulated", "NA"))
res_WT2_anti.heat_ctrl$log10padj <- -log10(res_WT2_anti.heat_ctrl$padj)
dat_genes_WT2_anti.heat_ctrl <- data.frame(cbind(res_WT2_anti.heat_ctrl$log2FoldChange, res_WT2_anti.heat_ctrl$log10padj, res_WT2_anti.heat_ctrl$threshold), stringsAsFactors = FALSE)
colnames(dat_genes_WT2_anti.heat_ctrl) <- c("log2FoldChange", "log10padj", "threshold")
row.names(dat_genes_WT2_anti.heat_ctrl) <- res_WT2_anti.heat_ctrl$IDGeneInfo
#dat_genes_WT2_anti.heat_ctrl <- dat_genes_WT2_anti.heat_ctrl[order(dat_genes_WT2_anti.heat_ctrl$log2FoldChange, decreasing = TRUE),]
dat_genes_WT2_anti.heat_ctrl$log2FoldChange <- as.numeric(dat_genes_WT2_anti.heat_ctrl$log2FoldChange)
dat_genes_WT2_anti.heat_ctrl$log10padj <- as.numeric(dat_genes_WT2_anti.heat_ctrl$log10padj)
dat_genes_WT2_anti.heat_ctrl$threshold <- factor(dat_genes_WT2_anti.heat_ctrl$threshold, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
```

```{r, warning = FALSE}
DEGcolors <- c("red", "blue", "dark grey")
pdf(file = "./outputs/DESeq-results/figures/Volcano_anti.heat_ctrl_colonies.pdf", height = 3, width = 4.25)
gVolcano_HW1_anti.heat <- ggplot(data=dat_genes_HW1_anti.heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW1_anti.heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFFF66", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_HW2_anti.heat <- ggplot(data=dat_genes_HW2_anti.heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_HW2_anti.heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFFF66", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT1_anti.heat <- ggplot(data=dat_genes_WT1_anti.heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT1_anti.heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFFF66", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))

gVolcano_WT2_anti.heat <- ggplot(data=dat_genes_WT2_anti.heat_ctrl, aes(log2FoldChange, log10padj))
gVolcano_WT2_anti.heat + geom_point(aes(color = threshold), alpha=1, size=1) +
  scale_color_manual(values = DEGcolors) +
  scale_x_continuous(limits = c(-10, 10), breaks = seq(-10,10,2)) + 
  ylim(c(0, 20)) +
  xlab("log2 fold change") +
  ylab("-log10 p-value") + 
  #geom_text_repel(data = dat_genes_LPS_ctrl[1:15, ], aes(label = rownames(dat_genes_LPS_ctrl[1:15, ])), color = "black", size = 2.5, box.padding = unit(0.35, "lines")) +
  theme_bw() + ggtitle("") +
  theme(legend.position = "none",
        panel.background = element_rect(fill = alpha("#FFFF66", 0.5)),
         axis.text = element_text(size = 10),
        axis.title = element_text(size = 10))
```

# Principal Components Analysis

## Regularized-log transformations
```{r, include=FALSE}
      rld_anti <- rlogTransformation(dds_anti, blind = TRUE)
      rld_heat <- rlogTransformation(dds_heat, blind = TRUE)
      rld_anti.heat <- rlogTransformation(dds_anti.heat, blind = TRUE)
      rld_AxH <- rlogTransformation(dds_AxH, blind = TRUE)
###
      rld_HW1 <- rlogTransformation(dds_HW1, blind = TRUE)
      rld_HW2 <- rlogTransformation(dds_HW2, blind = TRUE)
      rld_WT1 <- rlogTransformation(dds_WT1, blind = TRUE)
      rld_WT2 <- rlogTransformation(dds_WT2, blind = TRUE)
```
Set factor orders 
```{r}
samples_anti$Treatment <- factor(samples_anti$Treatment, levels = c("control", "Antibiotics"), ordered = TRUE)
samples_heat$Treatment <- factor(samples_heat$Treatment, levels = c("control", "Heat"), ordered = TRUE)
samples_anti.heat$Treatment <- factor(samples_anti.heat$Treatment, levels = c("control", "Antibiotics.Heat"), ordered = TRUE)
samples_AxH$Treatment <- factor(samples_AxH$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat"), ordered = TRUE)
```

```{r}
plotPCA.custom <-  function(object, intgroup="Treatment", ntop=500, returnData=FALSE, pcs = c(1,2))
{
 stopifnot(length(pcs) == 2)    ### added this to check number of PCs ####
  # calculate the variance for each gene
  rv <- rowVars(assay(object))
  # select the ntop genes by variance
  select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]
  # perform a PCA on the data in assay(x) for the selected genes
  pca <- prcomp(t(assay(object)[select,]))
  # the contribution to the total variance for each component
  percentVar <- pca$sdev^2 / sum( pca$sdev^2 )
  if (!all(intgroup %in% names(colData(object)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(object)[, intgroup, drop=FALSE])
  # add the intgroup factors together to create a new grouping factor
  group <- if (length(intgroup) > 1) {
    factor(apply( intgroup.df, 1, paste, collapse=" : "))
  } else {
    colData(object)[[intgroup]]
  }
  # assemble the data for the plot
  ########## Here we just use the pcs object passed by the end user ####
  d <- data.frame(PC1=pca$x[,pcs[1]], PC2=pca$x[,pcs[2]], group=group, intgroup.df, name=colnames(object))
  if (returnData) {
    attr(d, "percentVar") <- percentVar[1:2]
    return(d)
  }
}
```

```{r}
PCA.load <-  function(object, intgroup="Treatment", ntop=500, returnData=FALSE, pcs = c(1,2))
{
 stopifnot(length(pcs) == 2)    ### added this to check number of PCs ####
  # calculate the variance for each gene
  rv <- rowVars(assay(object))
  # select the ntop genes by variance
  select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]
  # perform a PCA on the data in assay(x) for the selected genes
  pca <- prcomp(t(assay(object)[select,]))
  # the contribution to the total variance for each component
  percentVar <- pca$sdev^2 / sum( pca$sdev^2 )
  if (!all(intgroup %in% names(colData(object)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(object)[, intgroup, drop=FALSE])
  # add the intgroup factors together to create a new grouping factor
  group <- if (length(intgroup) > 1) {
    factor(apply( intgroup.df, 1, paste, collapse=" : "))
  } else {
    colData(object)[[intgroup]]
  }
  
  # extract loadings
  rot <- pca$rotation
  arot <- abs(rot)
  relrot <- sweep(arot, 2, colSums(arot), "/")
  #row.names(relrot) <- rowData(object)$IDGeneInfo[select]
  return(relrot)
}
#https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-parameter
```

## PCA Plot
PCA ggplot2 functions
```{r, echo=FALSE}
ggPCA <- function(rld, samples, condcolors, ntop = 500,  pclab = c(1,2)) {
  
PCAtmtdata <- plotPCA.custom(rld, intgroup = c("Treatment", "Colony"), ntop = 500, returnData = TRUE,  pcs = c(pclab[1],pclab[2]))

PCAtmtpercentVar <- round(100* attr(PCAtmtdata, "percentVar"))

PCAplot <- ggplot(PCAtmtdata, aes(PCAtmtdata$PC1, PCAtmtdata$PC2, color=samples$Treatment, shape=samples$Colony)) +
  geom_point(size=4, show.legend = TRUE) + xlab(paste0( "PC", pclab[1], ": ", PCAtmtpercentVar[pclab[1]], "% variance")) + 
  ylab(paste0( "PC", pclab[2], ": ", PCAtmtpercentVar[pclab[2]], "% variance")) + 
  coord_fixed(1) + 
  ggtitle("Principal Component Analysis")

PCAplot + 
  scale_color_manual(values=condcolors, name="Treatment") + 
  scale_shape_manual(values=colshapes, name="Colony") +
  theme(legend.position = "right")
}
```

```{r, echo=FALSE}
ggPCA_O_geno <- function(rld, samples, condcolors, ntop = 500,  pclab = c(1,2)) {
  
PCAtmtdata <- plotPCA.custom(rld, intgroup = c("Treatment", "Colony"), ntop = 500, returnData = TRUE,  pcs = c(pclab[1],pclab[2]))

PCAtmtpercentVar <- round(100* attr(PCAtmtdata, "percentVar"))

PCAplot <- ggplot(PCAtmtdata, aes(PCAtmtdata$PC1, PCAtmtdata$PC2, color=samples$Treatment, shape=samples$Colony, group=samples$Colony)) +
    stat_ellipse(aes(fill = samples$Colony), geom = "polygon", type = "norm", alpha = 0.5) +
  geom_point(size=6, show.legend = TRUE) + xlab(paste0( "PC", pclab[1], ": ", PCAtmtpercentVar[pclab[1]], "% variance")) + 
  ylab(paste0( "PC", pclab[2], ": ", PCAtmtpercentVar[pclab[2]], "% variance")) + 
  ggtitle("Principal Component Analysis")

PCAplot + 
  scale_color_manual(values=condcolors, name="Treatment") +
  scale_fill_manual(values=colcolors, name="Colony") +
  scale_shape_manual(values=colshapes, name="Colony") +
  theme(legend.position = "right")
}
```

```{r}
#pdf(file = "./outputs/DESeq-results/figures/PCAs.pdf", height = 4, width = 4.25)
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(1,2))
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(1,3))
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(1,4))
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(2,3))
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(2,4))
ggPCA(rld_anti, samples_anti, condcolors_anti, pclab = c(3,4))
```
```{r}
#pdf(file = "./outputs/DESeq-results/figures/PCAs.pdf", height = 4, width = 4.25)
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(1,2))
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(1,3))
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(1,4))
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(2,3))
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(2,4))
ggPCA(rld_heat, samples_heat, condcolors_heat, pclab = c(3,4))
```
```{r}
#pdf(file = "./outputs/DESeq-results/figures/PCAs.pdf", height = 4, width = 4.25)
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(1,2))
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(1,3))
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(1,4))
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(2,3))
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(2,4))
ggPCA(rld_anti.heat, samples_anti.heat, condcolors_anti.heat, pclab = c(3,4))
```
```{r}
pdf(file = "./outputs/DESeq-results/figures/PCAs_AXH.pdf", height = 4, width = 8)
ggPCA_O_geno(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(1,2))
ggPCA(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(1,3))
ggPCA(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(1,4))
ggPCA(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(2,3))
ggPCA(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(2,4))
ggPCA(rld_AxH, samples_AxH, condcolors_AxH, pclab = c(3,4))
```
# Gene Expression Heatmaps
### Create heatmap annotation data frame
```{r}
anno_col <- as.data.frame(colData(rld_AxH)[, c("Treatment","Colony")])
```

## Sample Clustering Heatmap
```{r}
sampleDists <- dist(t(assay(rld_AxH)))
sampleDistMatrix <- as.matrix( sampleDists )
```

```{r}
pdf(file = "~/computing/scripts/EAPSI_AxH-master/DESeqresults_Pdam/figures/Heatmap_AXH_samples.pdf", width = 8.5, height = 7.5)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colonyclustercol,
         annotation_colors = anno_colors,
         annotation_col = anno_col,
         annotation_row = anno_col,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 8,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 4,
         treeheight_row = 50,
         cutree_rows = 4)
```

## Gene Clustering Heatmaps
### Create top 20 variable gene expression matrix 
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes20 <- head(order(rowVars(assay(rld_AxH)), decreasing = TRUE), 20)
mat20  <- assay(rld_AxH)[ topVarGenes20, ]
rownames(mat20) <- rowData(rld_AxH)$IDGeneInfo[topVarGenes20]
```
```{r}
mat20  <- mat20 - rowMeans(mat20)
```
## 20 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AxH-master/DESeqresults_Pdam/figures/Heatmap_AXH_top20.pdf", width = 11, height = 8.5)
pheatmap(mat20, 
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 20 Most Variable Gene Expression")
```

### Create top 50 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes50 <- head(order(rowVars(assay(rld_AxH)), decreasing = TRUE), 50)
mat50  <- assay(rld_AxH)[ topVarGenes50, ]
rownames(mat50) <- rowData(rld_AxH)$IDGeneInfo[topVarGenes50]
```
```{r}
mat50  <- mat50 - rowMeans(mat50)
```
## 50 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AxH-master/DESeqresults_Pdam/figures/Heatmap_AXH_top50.pdf", width = 11, height = 8.5)
pheatmap(mat50,
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 5,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 50 Most Variable Gene Expression")
```

### Create top 500 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes500 <- head(order(rowVars(assay(rld_AxH)), decreasing = TRUE), 500)
mat500  <- assay(rld_AxH)[ topVarGenes500, ]
rownames(mat500) <- rowData(rld_AxH)$IDGeneInfo[topVarGenes500]
```
## 500 Gene Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AxH-master/DESeqresults_Pdam/figures/Heatmap_AXH_top500.pdf", width = 11, height = 8.5)
pheatmap(mat500,          
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = FALSE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top 500 Most Variable Gene Expression")
```

### Create Antibiotics vs. Control differentially expressed gene matrix 
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
rownames_ressig_anti_ctrl_up <- row.names(ressig_anti_ctrl_up)
mat_ressig_anti_ctrl_up  <- assay(rld_AxH)[rownames_ressig_anti_ctrl_up, ]
row.names(mat_ressig_anti_ctrl_up) <- ressig_anti_ctrl_up$IDGeneInfo
```
```{r}
mat_ressig_anti_ctrl_up  <- mat_ressig_anti_ctrl_up - rowMeans(mat_ressig_anti_ctrl_up)
```
## Antibiotics vs. Control up DEG Heatmap
```{r}
pdf(file = "~/computing/scripts/EAPSI_AxH-master/DESeqresults_Pdam/figures/Heatmap_Anti_upDEG.pdf", width = 11, height = 8.5)
pheatmap(mat_ressig_anti_ctrl_up, col=geneexpcolors2,
         annotation_col = anno_col,
         annotation_row = NULL,
         show_rownames = TRUE,
         fontsize_row = 10,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         legend = FALSE,
         treeheight_col = 60,
         treeheight_row = 120)
```

# DEG Venn Diagrams and UpSet Plots

### Overall model treatment DEG venn intersects
```{r}
AH <- intersect(ressig_anti_ctrl$IDGeneInfo, ressig_heat_ctrl$IDGeneInfo)
AX <- intersect(ressig_anti_ctrl$IDGeneInfo, ressig_anti.heat_ctrl$IDGeneInfo)
HX <- intersect(ressig_heat_ctrl$IDGeneInfo, ressig_anti.heat_ctrl$IDGeneInfo)
AXH <- intersect(intersect(ressig_anti_ctrl$IDGeneInfo, ressig_heat_ctrl$IDGeneInfo), ressig_anti.heat_ctrl$IDGeneInfo)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH.pdf", height = 4, width = 4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

OverVenn <- draw.triple.venn(area1 = length(ressig_anti_ctrl$IDGeneInfo),
                             area2 = length(ressig_heat_ctrl$IDGeneInfo),
                             area3 = length(ressig_anti.heat_ctrl$IDGeneInfo),
                             n12 = length(AH),
                             n13 = length(AX),
                             n23 = length(HX),
                             n123 = length(AXH),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             alpha = c(rep(0.5, 3)),
                             cex = c(rep(2, 4), 3, rep(2, 2)),
                             fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             fontfamily = c(rep("Arial", 7)),
                             label.col = c(rep("black", 4), "black", rep("black", 2)),
                             print.mode = "raw",
                             sigdigs = 2,
                             euler.d = TRUE,
                             overrideTriple = "override",
                             scaled = FALSE)
grid.draw(OverVenn)
```

### Up
```{r}
AH_up <- intersect(ressig_anti_ctrl_up$IDGeneInfo, ressig_heat_ctrl_up$IDGeneInfo)
AX_up <- intersect(ressig_anti_ctrl_up$IDGeneInfo, ressig_anti.heat_ctrl_up$IDGeneInfo)
HX_up <- intersect(ressig_heat_ctrl_up$IDGeneInfo, ressig_anti.heat_ctrl_up$IDGeneInfo)
AXH_up <- intersect(intersect(ressig_anti_ctrl_up$IDGeneInfo, ressig_heat_ctrl_up$IDGeneInfo), ressig_anti.heat_ctrl_up$IDGeneInfo)
```
```{r}
#pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_up.pdf", height = 2.8, width = 2.8)
vp = viewport(height = unit(2.8, "in"),  width = unit(2.8, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

UpVenn <- draw.triple.venn(area1 = length(ressig_anti_ctrl_up$IDGeneInfo),
                           area2 = length(ressig_heat_ctrl_up$IDGeneInfo),
                           area3 = length(ressig_anti.heat_ctrl_up$IDGeneInfo),
                           n12 = length(AH_up),
                           n13 = length(AX_up),
                           n23 = length(HX_up),
                           n123 = length(AXH_up),
                           #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                           fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                           #lwd = rep(1,3),
                           #cat.cex = rep(1.5, 3),
                           #cat.fontfamily = rep("Arial", 3),
                           #cat.fontface = rep("bold", 3),
                           #alpha = c(rep(0.5, 3)),
                           #cex = c(rep(1, 4), 2, rep(1, 2)),
                           #fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                           #fontfamily = c(rep("Arial", 7)),
                           #label.col = c(rep("black", 4), "black", rep("black", 2)),
                           print.mode = "raw")
grid.draw(UpVenn)
```

### Down
```{r}
AH <- intersect(ressig_anti_ctrl_dn$IDGeneInfo, ressig_heat_ctrl_dn$IDGeneInfo)
AX <- intersect(ressig_anti_ctrl_dn$IDGeneInfo, ressig_anti.heat_ctrl_dn$IDGeneInfo)
HX <- intersect(ressig_heat_ctrl_dn$IDGeneInfo, ressig_anti.heat_ctrl_dn$IDGeneInfo)
AXH <- intersect(AH, HX)
```
```{r}
#pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_dn.pdf", height = 2.8, width = 2.8)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

DnVenn <- draw.triple.venn(area1 = length(ressig_anti_ctrl_dn$IDGeneInfo),
                             area2 = length(ressig_heat_ctrl_dn$IDGeneInfo),
                             area3 = length(ressig_anti.heat_ctrl_dn$IDGeneInfo),
                             n12 = length(AH),
                             n13 = length(AX),
                             n23 = length(HX),
                             n123 = length(AXH),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             #lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             #alpha = c(rep(0.5, 3)),
                             #cex = c(rep(1, 4), 2, rep(1, 2)),
                             #fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             #fontfamily = c(rep("Arial", 7)),
                             #label.col = c(rep("black", 4), "black", rep("black", 2)),
                             #print.mode = "raw",
                             #sigdigs = 2,
                             scaled = TRUE)
grid.draw(DnVenn)
```

## Colony-specific DEG venn intersects by treatment
### Antibiotics vs. Control
```{r}
sumDEGsHW <- union(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_HW2_anti_ctrl$IDGeneInfo)
sumDEGsWT <- union(ressig_WT1_anti_ctrl$IDGeneInfo, ressig_WT2_anti_ctrl$IDGeneInfo)
sumDEGs <- union(sumDEGsHW, sumDEGsWT)
length(sumDEGs)

##Pairwise
HW1_HW2 <- intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_HW2_anti_ctrl$IDGeneInfo)
HW1_WT1 <- intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_WT1_anti_ctrl$IDGeneInfo)
HW1_WT2 <- intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_WT2_anti_ctrl$IDGeneInfo)
HW2_WT1 <- intersect(ressig_HW2_anti_ctrl$IDGeneInfo, ressig_WT1_anti_ctrl$IDGeneInfo)
HW2_WT2 <- intersect(ressig_HW2_anti_ctrl$IDGeneInfo, ressig_WT2_anti_ctrl$IDGeneInfo)
WT1_WT2 <- intersect(ressig_WT1_anti_ctrl$IDGeneInfo, ressig_WT2_anti_ctrl$IDGeneInfo)
##Triple
HW1_HW2_WT1 <- intersect(HW1_HW2, ressig_WT1_anti_ctrl$IDGeneInfo)
HW1_HW2_WT2 <- intersect(HW1_HW2, ressig_WT2_anti_ctrl$IDGeneInfo)
HW1_WT1_WT2 <- intersect(WT1_WT2, ressig_HW1_anti_ctrl$IDGeneInfo)
HW2_WT1_WT2 <- intersect(WT1_WT2, ressig_HW2_anti_ctrl$IDGeneInfo)
##Quadruple
HW12_WT12 <- intersect(HW1_HW2, WT1_WT2)
```
```{r}
setdiff(HW1_WT2, HW12_WT12)
```
```{r}
setdiff(WT1_WT2, HW12_WT12)
```

```{r}
setdiff(ressig_WT2_LPS$IDGeneInfo, sumDEGsHW)
```
```{r}
pdf("~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/anti_colony_Venn.pdf", height = 2.4, width = 2.4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

#HW1, WT2, HW2, WT1
anti_ctrl_venn <- draw.quad.venn(area1 = length(ressig_HW1_anti_ctrl$IDGeneInfo),
               area2 = length(ressig_WT2_anti_ctrl$IDGeneInfo),
               area3 = length(ressig_HW2_anti_ctrl$IDGeneInfo),
               area4 = length(ressig_WT1_anti_ctrl$IDGeneInfo),
               n12 = length(HW1_WT2),
               n13 = length(HW1_HW2),
               n14 = length(HW1_WT1),
               n23 = length(HW2_WT2),
               n24 = length(WT1_WT2),
               n34 = length(HW2_WT1),
               n123 = length(HW1_HW2_WT2),
               n124 = length(HW1_WT1_WT2),
               n134 = length(HW1_HW2_WT1),
               n234 = length(HW2_WT1_WT2),
               n1234 = length(HW12_WT12),
               category = c("HW1", "WT2", "HW2", "WT1"),
               fill = c("#CCFFCC", "#CCCCFF", "#99FFCC", "#CCFFFF"),
               lwd = rep(1,4),
               cat.cex = rep(0.8, 4),
               cat.fontfamily = rep("Arial", 4),
               cat.fontface = rep("bold", 4),
               alpha = c(0.3,0.3,0.3,0.3),
               cex = c(rep(0.8,5), 1.2, rep(0.8, 9)),
               fontface = c(rep("plain", 5), "bold", rep("plain", 9)),
               fontfamily = c(rep("Arial", 15)),
               label.col = c(rep("black", 5), "black", rep("black", 9)),
               print.mode = "raw",
               sigdigs = 2,
               scaled = TRUE
)
grid.draw(anti_ctrl_venn)
```

## Heat vs. Control
```{r}
sumDEGsHW <- union(ressig_HW1_heat_ctrl$IDGeneInfo, ressig_HW2_heat_ctrl$IDGeneInfo)
sumDEGsWT <- union(ressig_WT1_heat_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo)
sumDEGs <- union(sumDEGsHW, sumDEGsWT)
length(sumDEGs)

##Pairwise
HW1_HW2 <- intersect(ressig_HW1_heat_ctrl$IDGeneInfo, ressig_HW2_heat_ctrl$IDGeneInfo)
HW1_WT1 <- intersect(ressig_HW1_heat_ctrl$IDGeneInfo, ressig_WT1_heat_ctrl$IDGeneInfo)
HW1_WT2 <- intersect(ressig_HW1_heat_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo)
HW2_WT1 <- intersect(ressig_HW2_heat_ctrl$IDGeneInfo, ressig_WT1_heat_ctrl$IDGeneInfo)
HW2_WT2 <- intersect(ressig_HW2_heat_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo)
WT1_WT2 <- intersect(ressig_WT1_heat_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo)
##Triple
HW1_HW2_WT1 <- intersect(HW1_HW2, ressig_WT1_heat_ctrl$IDGeneInfo)
HW1_HW2_WT2 <- intersect(HW1_HW2, ressig_WT2_heat_ctrl$IDGeneInfo)
HW1_WT1_WT2 <- intersect(WT1_WT2, ressig_HW1_heat_ctrl$IDGeneInfo)
HW2_WT1_WT2 <- intersect(WT1_WT2, ressig_HW2_heat_ctrl$IDGeneInfo)
##Quadruple
HW12_WT12 <- intersect(HW1_HW2, WT1_WT2)
```
```{r}
setdiff(HW1_WT2, HW12_WT12)
```
```{r}
setdiff(WT1_WT2, HW12_WT12)
```

```{r}
pdf("~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/heat_colony_Venn.pdf", height = 2.4, width = 2.4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

#HW1, WT2, HW2, WT1
heat_ctrl_venn <- draw.quad.venn(area1 = length(ressig_HW1_heat_ctrl$IDGeneInfo),
               area2 = length(ressig_WT2_heat_ctrl$IDGeneInfo),
               area3 = length(ressig_HW2_heat_ctrl$IDGeneInfo),
               area4 = length(ressig_WT1_heat_ctrl$IDGeneInfo),
               n12 = length(HW1_WT2),
               n13 = length(HW1_HW2),
               n14 = length(HW1_WT1),
               n23 = length(HW2_WT2),
               n24 = length(WT1_WT2),
               n34 = length(HW2_WT1),
               n123 = length(HW1_HW2_WT2),
               n124 = length(HW1_WT1_WT2),
               n134 = length(HW1_HW2_WT1),
               n234 = length(HW2_WT1_WT2),
               n1234 = length(HW12_WT12),
               category = c("HW1", "WT2", "HW2", "WT1"),
               fill = c("#CCFFCC", "#CCCCFF", "#99FFCC", "#CCFFFF"),
               lwd = rep(1,4),
               cat.cex = rep(0.8, 4),
               cat.fontfamily = rep("Arial", 4),
               cat.fontface = rep("bold", 4),
               alpha = c(0.3,0.3,0.3,0.3),
               cex = c(rep(0.8,5), 1.2, rep(0.8, 9)),
               fontface = c(rep("plain", 5), "bold", rep("plain", 9)),
               fontfamily = c(rep("Arial", 15)),
               label.col = c(rep("black", 5), "black", rep("black", 9)),
               print.mode = "raw",
               sigdigs = 2,
               scaled = TRUE
)
grid.draw(heat_ctrl_venn)
```

## Antibiotics.Heat vs. Control
```{r}
sumDEGsHW <- union(ressig_HW1_anti.heat_ctrl$IDGeneInfo, ressig_HW2_anti.heat_ctrl$IDGeneInfo)
sumDEGsWT <- union(ressig_WT1_anti.heat_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
sumDEGs <- union(sumDEGsHW, sumDEGsWT)
length(sumDEGs)

##Pairwise
HW1_HW2 <- intersect(ressig_HW1_anti.heat_ctrl$IDGeneInfo, ressig_HW2_anti.heat_ctrl$IDGeneInfo)
HW1_WT1 <- intersect(ressig_HW1_anti.heat_ctrl$IDGeneInfo, ressig_WT1_anti.heat_ctrl$IDGeneInfo)
HW1_WT2 <- intersect(ressig_HW1_anti.heat_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
HW2_WT1 <- intersect(ressig_HW2_anti.heat_ctrl$IDGeneInfo, ressig_WT1_anti.heat_ctrl$IDGeneInfo)
HW2_WT2 <- intersect(ressig_HW2_anti.heat_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
WT1_WT2 <- intersect(ressig_WT1_anti.heat_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
##Triple
HW1_HW2_WT1 <- intersect(HW1_HW2, ressig_WT1_anti.heat_ctrl$IDGeneInfo)
HW1_HW2_WT2 <- intersect(HW1_HW2, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
HW1_WT1_WT2 <- intersect(WT1_WT2, ressig_HW1_anti.heat_ctrl$IDGeneInfo)
HW2_WT1_WT2 <- intersect(WT1_WT2, ressig_HW2_anti.heat_ctrl$IDGeneInfo)
##Quadruple
HW12_WT12 <- intersect(HW1_HW2, WT1_WT2)
```
```{r}
HW12_WT12
```

```{r}
setdiff(HW1_WT2, HW12_WT12)
```
```{r}
setdiff(WT1_WT2, HW12_WT12)
```

```{r}
pdf("~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/AXH_colony_Venn.pdf", height = 2.4, width = 2.4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

#HW1, WT2, HW2, WT1
anti.heat_ctrl_venn <- draw.quad.venn(area1 = length(ressig_HW1_anti.heat_ctrl$IDGeneInfo),
               area2 = length(ressig_WT2_anti.heat_ctrl$IDGeneInfo),
               area3 = length(ressig_HW2_anti.heat_ctrl$IDGeneInfo),
               area4 = length(ressig_WT1_anti.heat_ctrl$IDGeneInfo),
               n12 = length(HW1_WT2),
               n13 = length(HW1_HW2),
               n14 = length(HW1_WT1),
               n23 = length(HW2_WT2),
               n24 = length(WT1_WT2),
               n34 = length(HW2_WT1),
               n123 = length(HW1_HW2_WT2),
               n124 = length(HW1_WT1_WT2),
               n134 = length(HW1_HW2_WT1),
               n234 = length(HW2_WT1_WT2),
               n1234 = length(HW12_WT12),
               category = c("HW1", "WT2", "HW2", "WT1"),
               fill = c("#CCFFCC", "#CCCCFF", "#99FFCC", "#CCFFFF"),
               lwd = rep(1,4),
               cat.cex = rep(0.8, 4),
               cat.fontfamily = rep("Arial", 4),
               cat.fontface = rep("bold", 4),
               alpha = c(0.3,0.3,0.3,0.3),
               cex = c(rep(0.8,5), 1.2, rep(0.8, 9)),
               fontface = c(rep("plain", 5), "bold", rep("plain", 9)),
               fontfamily = c(rep("Arial", 15)),
               label.col = c(rep("black", 5), "black", rep("black", 9)),
               print.mode = "raw",
               sigdigs = 2,
               scaled = TRUE
)
grid.draw(anti.heat_ctrl_venn)
```

## Colony-specific up/down DEG Venn intersects by treatment

## Treatment-specific DEG venn intersects by colony
### HW1
```{r}
AH_HW1 <- intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_HW1_heat_ctrl$IDGeneInfo)
AX_HW1 <- intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_HW1_anti.heat_ctrl$IDGeneInfo)
HX_HW1 <- intersect(ressig_HW1_heat_ctrl$IDGeneInfo, ressig_HW1_anti.heat_ctrl$IDGeneInfo)
AXH_HW1 <- intersect(intersect(ressig_HW1_anti_ctrl$IDGeneInfo, ressig_HW1_heat_ctrl$IDGeneInfo), ressig_HW1_anti.heat_ctrl$IDGeneInfo)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_HW1.pdf", height = 4, width = 4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

OverVenn <- draw.triple.venn(area1 = length(ressig_HW1_anti_ctrl$IDGeneInfo),
                             area2 = length(ressig_HW1_heat_ctrl$IDGeneInfo),
                             area3 = length(ressig_HW1_anti.heat_ctrl$IDGeneInfo),
                             n12 = length(AH_HW1),
                             n13 = length(AX_HW1),
                             n23 = length(HX_HW1),
                             n123 = length(AXH_HW1),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             alpha = c(rep(0.5, 3)),
                             cex = c(rep(2, 4), 3, rep(2, 2)),
                             fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             fontfamily = c(rep("Arial", 7)),
                             label.col = c(rep("black", 4), "black", rep("black", 2)),
                             print.mode = "raw",
                             sigdigs = 2,
                             euler.d = TRUE,
                             overrideTriple = "override",
                             scaled = FALSE)
grid.draw(OverVenn)
```

## HW2
```{r}
AH_HW2 <- intersect(ressig_HW2_anti_ctrl$IDGeneInfo, ressig_HW2_heat_ctrl$IDGeneInfo)
AX_HW2 <- intersect(ressig_HW2_anti_ctrl$IDGeneInfo, ressig_HW2_anti.heat_ctrl$IDGeneInfo)
HX_HW2 <- intersect(ressig_HW2_heat_ctrl$IDGeneInfo, ressig_HW2_anti.heat_ctrl$IDGeneInfo)
AXH_HW2 <- intersect(intersect(ressig_HW2_anti_ctrl$IDGeneInfo, ressig_HW2_heat_ctrl$IDGeneInfo), ressig_HW2_anti.heat_ctrl$IDGeneInfo)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_HW2.pdf", height = 4, width = 4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

OverVenn <- draw.triple.venn(area1 = length(ressig_HW2_anti_ctrl$IDGeneInfo),
                             area2 = length(ressig_HW2_heat_ctrl$IDGeneInfo),
                             area3 = length(ressig_HW2_anti.heat_ctrl$IDGeneInfo),
                             n12 = length(AH_HW2),
                             n13 = length(AX_HW2),
                             n23 = length(HX_HW2),
                             n123 = length(AXH_HW2),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             alpha = c(rep(0.5, 3)),
                             cex = c(rep(2, 4), 3, rep(2, 2)),
                             fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             fontfamily = c(rep("Arial", 7)),
                             label.col = c(rep("black", 4), "black", rep("black", 2)),
                             print.mode = "raw",
                             sigdigs = 2,
                             euler.d = TRUE,
                             overrideTriple = "override",
                             scaled = FALSE)
grid.draw(OverVenn)
```
## WT1
```{r}
AH_WT1 <- intersect(ressig_WT1_anti_ctrl$IDGeneInfo, ressig_WT1_heat_ctrl$IDGeneInfo)
AX_WT1 <- intersect(ressig_WT1_anti_ctrl$IDGeneInfo, ressig_WT1_anti.heat_ctrl$IDGeneInfo)
HX_WT1 <- intersect(ressig_WT1_heat_ctrl$IDGeneInfo, ressig_WT1_anti.heat_ctrl$IDGeneInfo)
AXH_WT1 <- intersect(intersect(ressig_WT1_anti_ctrl$IDGeneInfo, ressig_WT1_heat_ctrl$IDGeneInfo), ressig_WT1_anti.heat_ctrl$IDGeneInfo)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_WT1.pdf", height = 4, width = 4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

OverVenn <- draw.triple.venn(area1 = length(ressig_WT1_anti_ctrl$IDGeneInfo),
                             area2 = length(ressig_WT1_heat_ctrl$IDGeneInfo),
                             area3 = length(ressig_WT1_anti.heat_ctrl$IDGeneInfo),
                             n12 = length(AH_WT1),
                             n13 = length(AX_WT1),
                             n23 = length(HX_WT1),
                             n123 = length(AXH_WT1),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             alpha = c(rep(0.5, 3)),
                             cex = c(rep(2, 4), 3, rep(2, 2)),
                             fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             fontfamily = c(rep("Arial", 7)),
                             label.col = c(rep("black", 4), "black", rep("black", 2)),
                             print.mode = "raw",
                             sigdigs = 2,
                             euler.d = TRUE,
                             overrideTriple = "override",
                             scaled = FALSE)
grid.draw(OverVenn)
```

## WT2
```{r}
AH_WT2 <- intersect(ressig_WT2_anti_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo)
AX_WT2 <- intersect(ressig_WT2_anti_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
HX_WT2 <- intersect(ressig_WT2_heat_ctrl$IDGeneInfo, ressig_WT2_anti.heat_ctrl$IDGeneInfo)
AXH_WT2 <- intersect(intersect(ressig_WT2_anti_ctrl$IDGeneInfo, ressig_WT2_heat_ctrl$IDGeneInfo), ressig_WT2_anti.heat_ctrl$IDGeneInfo)
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/Venn_AXH_WT2.pdf", height = 4, width = 4)
vp = viewport(height = unit(4, "in"),  width = unit(4, "in"))
grid.newpage()
grid.rect(vp = vp, gp = gpar(col = "white"))

OverVenn <- draw.triple.venn(area1 = length(ressig_WT2_anti_ctrl$IDGeneInfo),
                             area2 = length(ressig_WT2_heat_ctrl$IDGeneInfo),
                             area3 = length(ressig_WT2_anti.heat_ctrl$IDGeneInfo),
                             n12 = length(AH_WT2),
                             n13 = length(AX_WT2),
                             n23 = length(HX_WT2),
                             n123 = length(AXH_WT2),
                             #category = c("Antibiotics", "Heat", "Antibiotics + Heat"),
                             fill = c("#99FFFF", "#FFCC99", "#FFFF66"),
                             lwd = rep(1,3),
                             #cat.cex = rep(1.5, 3),
                             #cat.fontfamily = rep("Arial", 3),
                             #cat.fontface = rep("bold", 3),
                             alpha = c(rep(0.5, 3)),
                             cex = c(rep(2, 4), 3, rep(2, 2)),
                             fontface = c(rep("plain", 4), "bold", rep("plain", 2)),
                             fontfamily = c(rep("Arial", 7)),
                             label.col = c(rep("black", 4), "black", rep("black", 2)),
                             print.mode = "raw",
                             sigdigs = 2,
                             euler.d = TRUE,
                             overrideTriple = "override",
                             scaled = FALSE)
grid.draw(OverVenn)
```

## Overall and Colony-specific DEG Venn
```{r}
DEGlists <- list(Heat = ressig_heat_ctrl$IDGeneInfo, Antibiotics = ressig_anti_ctrl$IDGeneInfo, Antibiotics.Heat = ressig_anti.heat_ctrl$IDGeneInfo)
```
```{r}
upset(fromList(DEGlists))
```
```{r}
pdf(file = "~/computing/scripts/EAPSI_AntiXHeat-master/DESeqresults_Pdam/figures/UpSet_AXH.pdf", height = 6, width = 8.5)
upset(fromList(DEGlists), 
      sets = c("Antibiotics.Heat", "Antibiotics", "Heat"),
      keep.order = TRUE,
      order.by = "degree",
      decreasing = FALSE,
      empty.intersections = "on",
      mb.ratio = c(0.5, 0.5),
      point.size = 4, line.size = 2,
      text.scale = c(2, 1.5, 2, 1, 2, 1.5),
      sets.bar.color = c("#FFCC33", "#00CCCC", "#FF6600"),
      main.bar.color = c("#FF6600", "#00CCCC", "#FFCC33", rep("black", 4)))
```


```{r}
DEGlists2 <- list(heat_up = ressig_heat_ctrl_up$IDGeneInfo, anti_up = ressig_anti_ctrl_up$IDGeneInfo, anti.heat_up = ressig_anti.heat_ctrl_up$IDGeneInfo)
```
```{r}
upset(fromList(DEGlists2))
```
```{r}
DEGlists3 <- list(heat_dn = ressig_heat_ctrl_dn$IDGeneInfo, anti_dn = ressig_anti_ctrl_dn$IDGeneInfo, anti.heat_dn = ressig_anti.heat_ctrl_dn$IDGeneInfo)
```
```{r}
upset(fromList(DEGlists3))
```
# Gene Counts Boxplots
Create ggboxplot functions
```{r}
ggboxplot <- function(gene) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$genotype), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = colcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
```
```{r}
gglistplot <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$treatment), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = condfillcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```
```{r}
gglistplot_genotype <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$genotype), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + facet_grid(.~stable_samples$genotype)
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = colcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```
```{r}
gglistplot_LPS <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
gplot <- filter(gplot, treatment == "control" | treatment == "LPS")
print(ggplot(gplot, aes(x=treatment, y=count, color=gplot$treatment)) 
      + geom_violin(aes(fill=gplot$treatment), show.legend = FALSE) 
      + geom_point(size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors_LPS) 
      + scale_fill_manual(values=condfillcolors_LPS) 
      + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```

## Prepare lists of gene IDs based on interesting groups
## Differentially expressed gene lists
```{r}
DElists <- list.files(path = "./DESeqresults_Pdam/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r, message=FALSE}
anti_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti_ctrl_up_ID.txt", what = "character")
LPS_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_LPS_ctrl_up_ID.txt", what = "character")
heat_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_heat_ctrl_up_ID.txt", what = "character")
anti.heat_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_ctrl_up_ID.txt", what = "character")
anti.heat.LPS_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_ctrl_up_ID.txt", what = "character")
```
```{r, message=FALSE}
anti_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti_ctrl_dn_ID.txt", what = "character")
LPS_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_LPS_ctrl_dn_ID.txt", what = "character")
heat_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_heat_ctrl_dn_ID.txt", what = "character")
anti.heat_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_ctrl_dn_ID.txt", what = "character")
anti.heat.LPS_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_ctrl_dn_ID.txt", what = "character")
```
```{r, message=FALSE}
anti.heat_anti_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_anti_up_ID.txt", what = "character")
anti.heat_heat_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_heat_up_ID.txt", what = "character")
anti.heat.LPS_anti_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_anti_up_ID.txt", what = "character")
anti.heat.LPS_heat_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_heat_up_ID.txt", what = "character")
anti.heat.LPS_LPS_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_LPS_up_ID.txt", what = "character")
```
```{r, message=FALSE}
anti.heat_anti_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_anti_dn_ID.txt", what = "character")
anti.heat_heat_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_heat_dn_ID.txt", what = "character")
anti.heat.LPS_anti_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_anti_dn_ID.txt", what = "character")
anti.heat.LPS_heat_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_heat_dn_ID.txt", what = "character")
anti.heat.LPS_LPS_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_LPS_dn_ID.txt", what = "character")
```
## Plot gene count boxplots
### DE Genes
```{r}
gglistplot(anti_ctrl_up_IDs[1:20])
```
```{r}
gglistplot(LPS_ctrl_up_IDs[1:15])
```
```{r}
gglistplot(LPS_ctrl_up_IDs[1:20])
```
```{r}
gglistplot(heat_ctrl_up_IDs[1:10])
```
```{r}
gglistplot(anti.heat_ctrl_up_IDs[1:10])
```
```{r}
gglistplot(anti.heat.LPS_ctrl_up_IDs[1:10])
```

### Innate immune genes
Lists of gene IDs
#### Pattern recognition receptors
```{r}
TLR_IDs <- scan("./genefamilies_Pdam/TLRs_IDs.txt", what = "character")
Lectin_IDs <- scan("./genefamilies_Pdam/Lectins_IDs.txt", what = "character")
#NLR_IDs
TNFR_IDs <- scan("./genefamilies_Pdam/TNFRs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TNFR_IDs)
```
```{r}
gglistplot(Lectin_IDs)
```

#### Signaling pathways
NFKB
```{r}
NFKB_IDs <- scan("./genefamilies_Pdam/NFKB_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(NFKB_IDs)
```
JNK
MAPK
```{r}
TRAF_IDs <- scan("./genefamilies_Pdam/TRAFs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TRAF_IDs)
```

#### Effector mechanisms
MPEG1 pdam_00017055
Damicornin pdam_00023321
DELTA-stichotoxin-She4b
Phospholipases
```{r}
PL_IDs <- scan("./genefamilies_Pdam/PL_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(PL_IDs)
```
Toxins
```{r}
Tox_IDs <- scan("./genefamilies_Pdam/Toxin_IDs.txt", what = "character")
```

```{r}
gglistplot(Tox_IDs[28])
```

```{r}
effectorIDs <- c("pdam_00017055", "pdam_00004721", "pdam_00023321")
gglistplot(effectorIDs)
```

#### Misc. immune genes
DMBT1
HLF

### Apoptosis pathway genes
Caspases

### Antioxidant response
SOD
PXDN

### Transcription factors
HLF

WGCNA Hub Genes
```{r}
hubIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/hubgenes/hubs_IDs.txt", what = "character")
```
```{r}
gglistplot(hubIDs[1:20])
```

WGCNA Module Genes
## Input lists of co-expressed module genes after WGCNA analysis
```{r, warning = FALSE}
modulelists <- list.files(path = "./WGCNAmodules_Pdam_ctrl-LPS/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r}
blackIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/lists/black_ID.txt", what = "character")
```

```{r}
intblackIDs <- c(14,21,29,199,200,213,231,336,359:364,402,408,415,437,481)
gglistplot_LPS(blackIDs[intblackIDs])
```
# Double Plots
## Assemble data frame with complete log-fold changes between treatment contrasts

```{r}
doubledf <- data.frame(cbind(res_LPS_ctrl$log2FoldChange, -log10(res_LPS_ctrl$padj),
                             res_anti_ctrl$log2FoldChange, -log10(res_anti_ctrl$padj),
                             res_heat_ctrl$log2FoldChange, -log10(res_heat_ctrl$padj),
                             res_anti.heat_ctrl$log2FoldChange, -log10(res_anti.heat_ctrl$padj),
                             res_anti.heat.LPS_ctrl$log2FoldChange, -log10(res_anti.heat.LPS_ctrl$padj)))
colnames(doubledf) <- c("LFC_LPS", "padj_LPS",
                        "LFC_Anti", "padj_Anti",
                        "LFC_Heat", "padj_Heat",
                        "LFC_AntiHeat", "padj_AntiHeat",
                        "LFC_AntiHeatLPS", "padj_AntiHeatLPS")
doubledf$ID <- rownames(res)
doubledf <- doubledf[,c(11, 1:10)]
doubledf$IDGeneInfo <- res$IDGeneInfo

# Add DEG up/down columns for single treatments
doubledf$DEG_LPS <- ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1, "Upregulated", 
                           ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1, "Downregulated", "NA"))
doubledf$DEG_LPS <- factor(doubledf$DEG_LPS, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

doubledf$DEG_Anti <- ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1, "Upregulated", 
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1, "Downregulated", "NA"))
doubledf$DEG_Anti <- factor(doubledf$DEG_Anti, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

doubledf$DEG_Heat <- ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1, "Upregulated", 
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1, "Downregulated", "NA"))
doubledf$DEG_Heat <- factor(doubledf$DEG_Heat, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)
  
doubledf$DEG_AntiHeat <- ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1, "Upregulated", 
                                ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1, "Downregulated", "NA"))
doubledf$DEG_AntiHeat <- factor(doubledf$DEG_AntiHeat, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

doubledf$DEG_AntiHeatLPS <-  ifelse(doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "Upregulated", 
                                    ifelse(doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "Downregulated", "NA"))
doubledf$DEG_AntiHeatLPS <- factor(doubledf$DEG_AntiHeatLPS, levels = c("Upregulated", "Downregulated", "NA"), ordered = TRUE)

doubledf <- doubledf[,c(1,12,2:3,13,4:5,14,6:7,15,8:9,16,10:11,17)]

# Add DEG threshold columns for two treatment consensus DEGs
doubledf$cDEG_LPS.Anti <- ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1, "UP_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1, "UP_DN",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1, "DN_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1, "DN_DN", "AA"))))

doubledf$cDEG_LPS.Heat <- ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1, "UP_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1, "UP_DN",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1, "DN_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1, "DN_DN", "AA"))))

doubledf$cDEG_Anti.Heat <- ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1, "UP_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1, "UP_DN",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1, "DN_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1, "DN_DN", "AA"))))

doubledf$cDEG_Anti.AntiHeat <- ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1, "UP_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1, "UP_DN",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1, "DN_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1, "DN_DN", "AA"))))

doubledf$cDEG_Anti.AntiHeatLPS <- ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "UP_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "UP_DN",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "DN_UP",
                            ifelse(doubledf$padj_Anti > 2 & doubledf$LFC_Anti < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "DN_DN", "AA"))))

doubledf$cDEG_Heat.AntiHeat <- ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1, "UP_UP",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1, "UP_DN",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1, "DN_UP",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1 & 
                            doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1, "DN_DN", "AA"))))

doubledf$cDEG_Heat.AntiHeatLPS <- ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "UP_UP",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "UP_DN",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "DN_UP",
                            ifelse(doubledf$padj_Heat > 2 & doubledf$LFC_Heat < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "DN_DN", "AA"))))

doubledf$cDEG_LPS.AntiHeatLPS <- ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "UP_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "UP_DN",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "DN_UP",
                            ifelse(doubledf$padj_LPS > 2 & doubledf$LFC_LPS < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "DN_DN", "AA"))))

doubledf$cDEG_AntiHeat.AntiHeatLPS <- ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "UP_UP",
                            ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat > 1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "UP_DN",
                            ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS > 1, "DN_UP",
                            ifelse(doubledf$padj_AntiHeat > 2 & doubledf$LFC_AntiHeat < -1 & 
                            doubledf$padj_AntiHeatLPS > 2 & doubledf$LFC_AntiHeatLPS < -1, "DN_DN", "AA"))))

write.csv(doubledf, file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/tables/all/all_LFC-pdaj.csv")

#doubledf <- doubledf %>% filter(DEG_Anti.AntiHeat!="NA") %>% filter(DEG_Heat.AntiHeat!="NA") %>% filter(DEG_Anti.Heat!="NA")
```

## Doubleplots

### LPS vs. Antibiotics
```{r}
doubledf <- doubledf %>% arrange(cDEG_LPS.Anti)
pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/doubleplots/double_LPS.Anti.pdf", height = 8, width = 8)
dpg <- ggplot(doubledf, aes(LFC_LPS, LFC_Anti))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_LPS.Anti)) +
  scale_colour_manual(values = DEGcolors) +
  geom_text_repel(data = doubledf[doubledf$cDEG_LPS.Anti=="UP_UP",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_LPS.Anti=="UP_UP"]), size = 1.5, box.padding = unit(0.25, "lines")) +
  geom_text_repel(data = doubledf[doubledf$cDEG_LPS.Anti=="DN_DN",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_LPS.Anti=="DN_DN"]), size = 1.5, box.padding = unit(0.25, "lines"))
```

### Antibiotics vs. Antibiotics + Heat
```{r}
doubledf <- doubledf %>% arrange(cDEG_Anti.AntiHeat)
pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/double_Anti.AntiHeat.pdf", height = 8, width = 8)
dpg <- ggplot(doubledf, aes(LFC_Anti, LFC_AntiHeat))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_Anti.AntiHeat)) +
  scale_colour_manual(values = DEGcolors) +
  geom_text_repel(data = doubledf[doubledf$cDEG_Anti.AntiHeat=="UP_UP"[1:10],], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_Anti.AntiHeat=="UP_UP"[1:10]]), size = 1.5, box.padding = unit(0.25, "lines")) +
  geom_text_repel(data = doubledf[doubledf$cDEG_Anti.AntiHeat=="DN_DN"[1:10],], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_Anti.AntiHeat=="DN_DN"[1:10]]), size = 1.5, box.padding = unit(0.25, "lines"))
```

### Heat vs. Antibiotics + Heat
```{r}
doubledf <- doubledf %>% arrange(cDEG_Heat.AntiHeat)
pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/double_Heat.AntiHeat.pdf", height = 8, width = 8)
dpg <- ggplot(doubledf, aes(LFC_Heat, LFC_AntiHeat))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_Heat.AntiHeat)) +
  scale_colour_manual(values = DEGcolors) +
  geom_text_repel(data = doubledf[doubledf$cDEG_Heat.AntiHeat=="UP_UP"[1:10],], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_Heat.AntiHeat=="UP_UP"[1:10]]), size = 1.5, box.padding = unit(0.25, "lines")) +
  geom_text_repel(data = doubledf[doubledf$cDEG_Heat.AntiHeat=="DN_DN"[1:10],], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_Heat.AntiHeat=="DN_DN"[1:10]]), size = 1.5, box.padding = unit(0.25, "lines"))
```

### LPS vs. Antibiotics + Heat + LPS
```{r}
doubledf <- doubledf %>% arrange(cDEG_LPS.AntiHeatLPS)
pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/double_LPS.AntiHeatLPS.pdf", height = 8, width = 8)
dpg <- ggplot(doubledf, aes(LFC_LPS, LFC_AntiHeatLPS))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_LPS.AntiHeatLPS)) +
  scale_colour_manual(values = DEGcolors) +
  geom_text_repel(data = doubledf[doubledf$cDEG_LPS.AntiHeatLPS=="UP_UP",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_LPS.AntiHeatLPS=="UP_UP"]), size = 1.5, box.padding = unit(0.25, "lines")) +
  geom_text_repel(data = doubledf[doubledf$cDEG_LPS.AntiHeatLPS=="DN_DN",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_LPS.AntiHeatLPS=="DN_DN"]), size = 1.5, box.padding = unit(0.25, "lines"))
```

### Antibiotics + Heat vs. Antibiotics + Heat + LPS
```{r}
doubledf <- doubledf %>% arrange(cDEG_AntiHeat.AntiHeatLPS)
pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/double_AntiHeat.AntiHeatLPS.pdf", height = 8, width = 8)
dpg <- ggplot(doubledf, aes(LFC_AntiHeat, LFC_AntiHeatLPS))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_AntiHeat.AntiHeatLPS)) +
  scale_colour_manual(values = DEGcolors) 
  #geom_text_repel(data = doubledf[doubledf$cDEG_AntiHeat.AntiHeatLPS=="UP_UP",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_AntiHeat.AntiHeatLPS=="UP_UP"]), size = 1.5, box.padding = unit(0.25, "lines")) +
  #geom_text_repel(data = doubledf[doubledf$cDEG_AntiHeat.AntiHeatLPS=="DN_DN",], aes(label = doubledf$IDGeneInfo[doubledf$cDEG_AntiHeat.AntiHeatLPS=="DN_DN"]), size = 1.5, box.padding = unit(0.25, "lines"))
```

## Filter and select data frame for specific treatments

### Antibiotics, Heat, Antibiotics + Heat
```{r}
anti_heat_doubledf <- doubledf %>% select(ID, IDGeneInfo, LFC_Anti, padj_Anti, DEG_Anti, LFC_Heat, padj_Heat, DEG_Heat, LFC_AntiHeat, padj_AntiHeat, DEG_AntiHeat, cDEG_Anti.Heat, cDEG_Anti.AntiHeat, cDEG_Heat.AntiHeat) %>% filter(DEG_Anti!="NA" | DEG_Heat!="NA" | DEG_AntiHeat!="NA") %>% filter(padj_Anti!="NA") %>% filter(padj_Heat!="NA") %>% filter(padj_AntiHeat!="NA")

write.csv(anti_heat_doubledf, file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/tables/all/anti_heat_LFC-pdaj.csv")
```

### Antibiotics vs. Heat
```{r}
anti_heat_doubledf <- anti_heat_doubledf %>% arrange(cDEG_Anti.Heat)
#pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/doubleplots/double_Anti.Heat.pdf", height = 8, width = 8)
dpg <- ggplot(anti_heat_doubledf, aes(LFC_Anti, LFC_Heat))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_Anti.Heat)) +
  scale_colour_manual(values = DEGcolors) 
```

### Antibiotics vs. Antibiotics + Heat
```{r}
anti_heat_doubledf <- anti_heat_doubledf %>% arrange(cDEG_Anti.AntiHeat)
#pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/doubleplots/double_Anti.Heat.pdf", height = 8, width = 8)
dpg <- ggplot(anti_heat_doubledf, aes(LFC_Anti, LFC_AntiHeat))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_Anti.AntiHeat)) +
  scale_colour_manual(values = DEGcolors) 
```

### Heat vs. Antibiotics + Heat
```{r}
anti_heat_doubledf <- anti_heat_doubledf %>% arrange(cDEG_Heat.AntiHeat)
#pdf(file = "~/computing/scripts/EAPSI.HW-WT-master/DESeqresults_Pdam/figures/doubleplots/double_Anti.Heat.pdf", height = 8, width = 8)
dpg <- ggplot(anti_heat_doubledf, aes(LFC_Heat, LFC_AntiHeat))
dpg + 
  geom_hline(yintercept = 0, color = "black", size = 0.2) +
  geom_vline(xintercept = 0, color = "black", size = 0.2) +
  geom_abline(intercept = 0, slope = 1, color = "black", size = 0.2, linetype = "dashed") +
  scale_x_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  scale_y_continuous(limits = c(-10,10), breaks = seq(-10,10,2)) + 
  geom_point(aes(color=cDEG_Heat.AntiHeat)) +
  scale_colour_manual(values = DEGcolors) 
```

```{r}
library(rgl)
```

```{r}
plot3d(anti_heat_doubledf$LFC_Anti, anti_heat_doubledf$LFC_Heat, anti_heat_doubledf$LFC_AntiHeat, xlim = c(-10,10), xlab = "Anti", ylim = c(-10,10), ylab = "Heat", zlim = c(-10,10), zlab = "Anti + Heat")
```


# topGO Enrichment Tests
## Input required *P. damicornis* GO annotation data and construct Gene-to-GO object for custom annotation mapping
```{r, warning=FALSE}
geneID_GO <- readMappings(file = "./data/pdam_emapper_genesGO.txt")
str(head(geneID_GO))
```
## Construct GO-to-Gene object for GO term searches of interesting genes
```{r}
GO_geneID <- inverseList(geneID_GO)
str(head(GO_geneID))
```
## Generate gene universe and GO universe from Gene-to-GO and GO-to-Gene objects
```{r}
geneNames <- names(geneID_GO)
str(head(geneNames))
```
```{r}
GONames <- names(GO_geneID)
str(head(GONames))
```

## Input lists of positively and negatively differentially expressed genes after DESeq2 analysis
Anti_Control (Up/Down)
LPS_Control (Up/Down)
Heat_Control (Up/Down)
Anti+Heat_Control (Up/Down)
Anti+Heat+LPS_Control (Up/Down)
## Construct topGOdata object using *P. damicornis* gene models, differentially expressed gene lists and gene ontology annotations
## Perform Fisher's exact test to identify enriched gene ontology terms
## Perform Kolmogorov-Smirnov test to identify enriched gene ontology terms
## Analyse highly-significant MF GO terms and corresponding p-values, then produce a table of results
Remember to use DESeq_geneID.sh to generate gene ID lists
```{r}
DElists <- list.files(path = "./outputs/DESeq-results/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r}
topGOanalyzeDE <- function(filename) {
### Custom function for full topGO analysis and results table generation
DEgenes <- read.delim(filename, header = FALSE, stringsAsFactors = FALSE)
DEgenes <- as.data.frame(DEgenes)
geneList <- factor(as.integer(geneNames %in% DEgenes$V1))
names(geneList) <- geneNames
#node size is used to prune the GO hierarchy from the terms which have less than that many annotated genes 
Pdam_GOdata <- new("topGOdata",
              description = DElist,
              ontology = "MF",
              allGenes = geneList,
              nodeSize = 10,
              annot = annFUN.gene2GO,
              gene2GO = geneID_GO)
resultFisher01 <- runTest(Pdam_GOdata, algorithm = "weight01", statistic = "fisher")
resultFisher <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "fisher")
resultKS <- runTest(Pdam_GOdata, algorithm = "classic", statistic = "ks")
resultKSelim <- runTest(Pdam_GOdata, algorithm = "elim", statistic = "ks")
allRes <- GenTable(Pdam_GOdata, 
                   elimKS = resultKSelim,
                   classicKS = resultKS,
                   classicFisher = resultFisher,
                   weight01Fisher = resultFisher01,
                   orderBy = "weight01Fisher", 
                   ranksOf = "weight01Fisher", 
                   topNodes = 50)
print(allRes)
resultfile <- paste(filename, "_topGO.csv", sep = "")
write.csv(allRes, file = resultfile)
}
```

```{r message=TRUE}
for (DElist in DElists)
{
  print(DElist)
  topGOanalyzeDE(DElist)
}
```
```{bash}
mv ./outputs/DESeq-results/lists/*.csv ./outputs/DESeq-results/topGO
```
```{bash}
bash ./bash/awk_REVIGO.sh
mv ./outputs/DESeq-results/topGO/*.txt ./outputs/DESeq-results/REVIGO
sed -i -e 's/"//g' ./outputs/DESeq-results/REVIGO/*.txt
rm  ./outputs/DESeq-results/REVIGO/*.txt-e
```

# GO_MWU Enrichment Test
Generate gene ID and measure (signed log10 p-values) tables from DESeq2 lists
```{r}
ID_measure <- data.frame(row.names(res_anti), res_anti$log2FoldChange, res_anti$pvalue)
ID_measure$logP <- ifelse(ID_measure$res_anti.log2FoldChange > 0, 1*round(-log10(ID_measure$res_anti.pvalue), 3), -1*round(-log10(ID_measure$res_anti.pvalue), 3))
#generate function for this and do for all comparisons!
```

# REVIGO plots
```{r}
library(scales)
library(colorspace)
```
```{r}
REVIGOs <- list.files(path = "./DESeqresults_Pdam/REVIGO/REVIGOres", pattern = "REVIGO", full.names = TRUE)
```
```{r}
REVIGOs[1]
```
```{r}
revigo.data <- read.csv(REVIGOs[1]) %>% dplyr::select(-representative, -eliminated)
revigo.data.frame <- data.frame(revigo.data)
revigo.data.frame.clean <- revigo.data.frame [(revigo.data.frame$plot_X != "null" & revigo.data.frame$plot_Y != "null"), ]
```

```{r}
REVIGOplot <- function(revigo) {
plotTitle <- as.character(revigo)
revigo.names <- c("term_ID","description","frequency","plot_X","plot_Y","plot_size","log10_p_value","uniqueness","dispensability");
revigo.data <- read.csv(revigo) %>% dplyr::select(-representative, -eliminated)
one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$plot_size <- as.numeric( as.character(one.data$plot_size) );
one.data$log10_p_value <- as.numeric( as.character(one.data$log10_p_value) );
#one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#view(one.data)
# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns, etc. can be changed below
p1 <- ggplot( data = one.data )
p1 <- p1 + 
  geom_point( aes(plot_X, plot_Y, colour = log10_p_value, size = plot_size), alpha = I(0.6) ) + 
  geom_point( aes(plot_X, plot_Y, size = plot_size), shape = 21, fill = "transparent", colour = I (alpha ("black", 0.6) )) + 
  scale_color_gradientn(colors = c("blue", "green", "yellow", "red"), limits = c(min(one.data$log10_p_value), 0)) +
  scale_size_area() + 
  scale_size(range=c(5, 30)) + 
  scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0)) +
  ggtitle(plotTitle) +
  theme(plot.title = element_text(size=12));
ex <- one.data [ one.data$dispensability < 0.15, ]; 
p1 <- p1 + geom_text( data = ex, aes(plot_X, plot_Y, label = description), colour = I(alpha("black", 0.85)), size = 3 ) + 
  labs (y = "semantic space x", x = "semantic space y") + theme(legend.key = element_blank())
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10) +
  ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);
# --------------------------------------------------------------------------
print(p1)
}
```

```{r, message=FALSE}
#pdf(file = "./DESeqresults_Pdam/figures/REVIGObubbles", width = 11, height=8.5)
for (i in REVIGOs)
{
  REVIGOplot(i)
}
```