---
title: "Pocillopora_AXH_microbiome_QC"
author: "Mike Connelly"
date: "8/4/2020"
output: 
  html_document:
      code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_AXH")
```
```{r packages}
library("tidyverse")
# extended visualization functions
source("./R/ggrare.R")
```

## Quality control plots
```{r rd}
# Read in read depth information
readdepth <- read_csv("./outputs/qiime2/QC/per-sample-fastq-counts.csv")
colnames(readdepth) <- c("SampleID", "Count")
readdepth <- merge.data.frame(map, readdepth, by = "SampleID")
```
### Read depth QC plot
```{r rd_qc}
readdepth.AXH <- dplyr::filter(readdepth, Treatment == "control" | Treatment == "Heat" | Treatment == "Antibiotics" | Treatment == "Antibiotics.Heat" | Treatment == "positive" | Treatment == "negative") %>% dplyr::filter(Reef == "Houwan" | Reef == "Wanglitung" | Reef == "controls")
###
readdepth.AXH$SampleID <- gsub("\\.6", "_control_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.5", "_anti_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.4", "_heat_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.1", "_anti+heat_", readdepth.AXH$SampleID)
###
readdepth.AXH$Treatment <- factor(readdepth.AXH$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat", "positive", "negative"), ordered = TRUE)
readdepth.AXH <- dplyr::arrange(readdepth.AXH, Treatment, Colony)
readdepth.AXH$SampleID <- factor(readdepth.AXH$SampleID, levels = readdepth.AXH$SampleID, ordered = TRUE)
###
ggdepth <- ggplot(readdepth.AXH, aes(SampleID, Count)) 
ggdepth +
  geom_bar(stat = "identity", aes(color = Treatment, fill = Treatment)) +
  #scale_y_continuous(trans = 'log10') +
  scale_color_manual(values = c(condcolors_AxH, "black", "black")) +
  scale_fill_manual(values = c(condfillcolors_AxH, "dark gray", "light gray")) +
  ylab("16S Read Count") +
  theme(axis.text.x = element_text(angle = 90)) +
  ggtitle("Total number of 16S reads per sample")
ggsave(file = "./outputs/phyloseq-results/figures/readdepth_AXH.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
ggsave(file = "./manuscript_figures/FigS##_16S_readdepth.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
```

### Rarefaction curve
```{r rare_curve}
grare <- ggrare(ps_AXH_all, step = 10, color = "Treatment", label = "Colony", plot = FALSE, se = TRUE)
grare <- grare + 
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condfillcolors_AxH) +
  #facet_wrap(~Treatment) +
  ggtitle("Rarefaction curve of 16S amplicon sequence variants")
print(grare)
ggsave(file = "./outputs/phyloseq-results/figures/rarefaction_.AXH.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
ggsave(file = "./manuscript_figures/FigS##_rarefactioncurve.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
```

### Low-depth 8-10-20
```{r}
readdepth %>% 
  filter(Count < 50000) %>% 
  filter(Colony != "controls") %>% 
  arrange(SampleID)
```
```{r}
grare + coord_cartesian(xlim = 20000)
```

