---
title: "Pocillopora_AXH_transcriptome_analysis"
author: "Mike Connelly"
date: "4/24/2020"
output: 
  html_document:
      code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, cache = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_AxH/")
```
## Setup packages and working directories
Include all packages needed for the entire DESeq2 analysis and downstream visualizations
```{r packages, message=FALSE, include=FALSE}
library("tidyverse")
library("DESeq2")
library("apeglm")
library("variancePartition")
library("doParallel")
library("limma")
library("adegenet")
library("KOGMWU")
source("./R/GO_MWU/gomwu.functions.R")
source("./R/AXH_functions.R")
library("ggpubr")
library("ggnewscale")
library("patchwork")
library("cowplot")
library("stringr")
library("RColorBrewer")
library("wesanderson")

# Set overall theme, colors, and shapes for ggplot2
theme_set(theme_bw())
###
varcolors <- c(wes_palette(n=4, name="Zissou1"), "grey85")
###
condcolors_AxH <- c("blue", "darkorange", "cyan", "gold")
condcolors_heat <- c(condcolors_AxH[1:2])
condcolors_anti <- c(condcolors_AxH[1], condcolors_AxH[3])
condcolors_anti.heat <- c(condcolors_AxH[1], condcolors_AxH[4])
###
condfillcolors_AxH <- c("#99CCFF", "#FFCC99", "#99FFFF", "#FFFF66")
colshapes <- c(16, 17, 15, 18)
colshapes <- c(21, 24, 22, 23)
colcolors <- c("olivedrab3", "springgreen", "deepskyblue", "skyblue")
###
condcolors_null <- c(rep("black", 4))
colshapes_null <- c(rep(20, 8))
```
## Import sample metadata, counts data, and gene annotation
```{r data, echo=TRUE}
samples <- read.table("./data/EAPSI_samples_AxH.txt", header = TRUE)
samples$SampleID <- gsub('-','.', samples$SampleID)
# Import and tidy counts data
countdata <- read.delim("./outputs/STARcounts_Pdam/AxH_Pdam.counts", comment.char="#")
# Set Gene ID's as row names
row.names(countdata) <- countdata$Geneid
# Remove first six columns (Geneid, chr, start, end, strand, length)
countdata <- countdata[ ,7:ncol(countdata)]
#countdata <- countdata[ ,6:ncol(countdata)]
# Remove file prefixes and suffixes
  colnames(countdata) <- gsub("X.", "", colnames(countdata))
  colnames(countdata) <- gsub("scratch.projects.transcriptomics.mikeconnelly.projects.EAPSI_Pocillopora_AxH.outputs.STARalign_Pdam.", "", colnames(countdata))
  colnames(countdata) <- gsub("_PdamAligned.out.bam$", "", colnames(countdata))
# Sort countdata by P. damicornis gene ID
countdata.sorted <- as.data.frame(countdata[order(rownames(countdata)),])
```
```{r gene data}
gene_annotation <- read.delim(file = "./data/pdam_genome_annotations.tab", header = T) %>% arrange(ID)
rownames(gene_annotation) <- gene_annotation$ID
# Check gene feature annotation and countdata rowname order coherence
all(rownames(countdata.sorted) == gene_annotation$ID)
all(rownames(countdata.sorted) ==  rownames(gene_annotation))
# Obtain KOG annotations for P. damicornis genome
gene2kog <- gene_annotation %>% select(ID, KOG_Class) %>% filter(KOG_Class != "")
```
```{r coldata}
# Get sample data into tibble
coldata <- samples %>%
  as_tibble(.) %>%
  select(SampleID, Colony, Treatment, Batch)
coldata <- coldata %>%
  mutate(Colony.Treatment = interaction(Colony, Treatment)) %>%
  mutate_if(is.character, as.factor) %>% 
  arrange(SampleID) %>% #Order rows by sample name                         
  column_to_rownames(var = "SampleID")  
#Gather raw counts into tibble
colcounts <- countdata.sorted %>%
  rownames_to_column("gene") %>%
  as_tibble(.) %>%
  gather(key = "SampleID", value = "count", -gene)
```

## Create DESeqDataSet
```{r create_dds}
# Create full DESeqDataSet
dds <- DESeqDataSetFromMatrix(countData = countdata.sorted,
                              colData = coldata,
                              design = ~ Colony)
# Check annotation and dds object rowname order coherence
all(rownames(dds) == rownames(gene_annotation))
# Add gene feature annotation to DESeqDataSets
mcols(dds) <- cbind(mcols(dds), gene_annotation)
# Subset DESeqDataSet
### Remove genes counted less than 10 times in subset
dds <- dds[ rowSums(counts(dds)) > 10, ]
```
## Visualize global gene expression
```{r vst}
# Normalize expression data for visualization purposes using VST tranformation
vsd <- vst(dds, blind = TRUE)
```
```{r}
ggPCA(vsd, samples, condcolors_AxH, pclab = c(1,2))
```
### Principal Coordinate Analysis (PCoA)
```{r mds, eval = TRUE}
## Calculate distances among samples
sampleDists <- dist(t(assay(vsd)), method = "manhattan")
sampleDistMatrix <- as.matrix(sampleDists)

## Calculate MDS
mds <- as.data.frame(colData(vsd)) %>% 
  cbind(cmdscale(sampleDistMatrix))
```
```{r colony pcoa}
# Calculate Colony centroids for plotting
mds_col <- mds %>%
  group_by(Colony) %>%
  dplyr::summarise(c1 = mean(`1`), c2 = mean(`2`)) %>%    
  full_join(mds)
#set factor orders 
mds_col$Colony <- factor(mds$Colony, levels = c("HW1", "HW2", "WT1", "WT2"), ordered = TRUE)
mds_col$Treatment <- factor(mds$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat"), ordered = TRUE)

# Plot with spiders
pcoa1 <- ggplot(mds_col, aes(fill = Treatment)) +
  geom_segment(mapping = aes(x = `1`, y = `2`, xend = c1, yend = c2),
               lwd = 0.25, col = "dark grey") +
  #colony centroid points
  geom_point(size = 4, aes(x = c1, y = c2, color = Colony, shape = Colony), stroke = 2, fill = "black", show.legend = TRUE) +
  #sample points
  geom_point(size = 3, aes(x = `1`, y = `2`, fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = colcolors) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes) +
  labs(x = "PC1", y = "PC2") +
  #theme_custom() +
  guides(fill = guide_legend(override.aes = list(fill = condcolors_AxH, shape = 21, alpha = 1, stroke = 0.5))) +
  theme(legend.spacing.y = unit(0, "cm")) 
pcoa1title <- expression(paste("PCoA of overall ", italic("Pocillopora"), " spp. gene expression"))
pcoa1 <- pcoa1 + ggtitle(pcoa1title, subtitle = "Plotted with colony centroids")
pcoa1
ggsave(pcoa1, filename = "./outputs/DESeq-results/figures/pcoa_colony.png", width = 6, height = 4, units = "in")
ggsave(pcoa1, filename = "./manuscript_figures/Fig1_PCoA_Colony.png", width = 6, height = 4, units = "in")
```
```{r treatment pcoa}
# Calculate Treatment centroids for plotting
mds_trmt <- mds %>%
  group_by(Treatment) %>%
  dplyr::summarise(c1 = mean(`1`), c2 = mean(`2`)) %>%    
  full_join(mds)
#set factor orders 
mds_trmt$Colony <- factor(mds_trmt$Colony, levels = c("HW1", "HW2", "WT1", "WT2"), ordered = TRUE)
mds_trmt$Treatment <- factor(mds_trmt$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat"), ordered = TRUE)

# Plot with spiders
pcoa2 <-ggplot(mds_trmt, aes(fill = Treatment)) +
  #treatment ellipses
  #stat_ellipse(aes(x = `1`, y = `2`, color = Treatment, fill = Treatment), geom = "polygon", type = "norm", alpha = 0.0) + 
  #sample-centroid spiders paths
  geom_segment(mapping = aes(x = `1`, y = `2`, xend = c1, yend = c2),
               lwd = 0.25, col = "dark grey") +
  #treatment centroid points
  geom_point(size = 4, aes(x = c1, y = c2, color = Treatment), fill = "black", shape = 21, stroke = 2, show.legend = TRUE) +
  #sample points
  geom_point(size = 3, aes(x = `1`, y = `2`, fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes) +
  labs(x = "PC1", y = "PC2") +
  guides(shape = guide_legend(override.aes = list(shape = colshapes, alpha = 1, stroke = 0.5))) +
  #theme_custom() +
  theme(legend.spacing.y = unit(0, "cm"))
pcoa2title <- expression(paste("PCoA of overall ", italic("Pocillopora"), " spp. gene expression"))
pcoa2 <- pcoa2 + ggtitle(pcoa2title, subtitle = "Plotted with treatment centroids")
pcoa2
ggsave(pcoa2, filename = "./outputs/DESeq-results/figures/pcoa_treatment.png", width = 6, height = 4, units = "in")
ggsave(pcoa2, filename = "./manuscript_figures/Fig1_PCoA_Treatment.png", width = 6, height = 4, units = "in")
```

### Variance partitioning
Partition total transcriptome variance among explanatory factors including sequencing batch, colony/genotype, treatment, and their interaction.
```{r varpart_total}
varformula <- ~ (1|Batch) + (1|Colony) + (1|Treatment) + (1|Treatment:Colony)
varpart_total <- fitExtractVarPartModel(assay(vsd), varformula, coldata)
# sort variables (i.e. columns) by median fraction of variance explained
vpt <- sortCols(varpart_total)
# Violin plot of contribution of each variable to total variance
varplot <- plotVarPart(vpt, col = varcolors)
vartitle <- expression(paste(italic("Pocillopora"), " spp. transcriptome variance partition"))
varplot <- varplot + ggtitle(vartitle)
ggsave(varplot, filename = "./manuscript_figures/FigS2_variancePartition_total.png", height = 4, width = 5, units = "in", device = "png")
```
Partition transcriptome variance among explanatory factors within each colony/genotype
```{r varpart_col, eval = TRUE, results = 'hide'}
# Join count data with sample data, and nest vs count data by colony
rm(all)
all <- left_join(coldata, colcounts) 
all <- all %>% 
  group_by(Colony) %>%
  nest()

# Create DESeqDataSet for each colony, filter genes mean count > 1 within each colony, and vst transform
all <- all %>%
  mutate(sdat = map(.x = data, ~ data.frame(column_to_rownames(distinct(., SampleID, Treatment, Batch, group), "SampleID"))),
         counts = map(.x = data, ~ spread(select(., SampleID, gene, count), SampleID, count)),
         counts = map(.x = counts, ~ as.matrix(column_to_rownames(., "gene"))),
         dds = map2(.x = counts, .y = sdat, ~ DESeqDataSetFromMatrix(countData = .x, colData = .y, design = ~ group)),
         ddsf = map(dds, function(x) x[rowMedians(counts(x)) > 1, ]),  # Filter DESeqDataSet
         vsd = map(ddsf, ~ vst(., blind = FALSE)))  # Variance Stabilizing Transformation

# Run variance partition analysis for each colony
cl <- makeCluster(8); registerDoParallel(cl)
all <- all %>%
  mutate(vp = map2(vsd, sdat, ~ fitExtractVarPartModel(
    exprObj = assay(.x), data = .y, formula = ~ (1|Batch) + (1|Treatment)
    )))
stopCluster(cl)

# Calculate sum of variance explained by each source across all genes for each colony
allsumm <- all %>% mutate(vpsum = map(vp, ~ summarise_all(., sum)))

all$vp[[3]] %>%
  as.data.frame(.) %>%
  summarise_all(., median)

plotVarPart(all$vp[[3]])
rowSums(assay(all$vsd[[3]]))
mean(all$vp[[3]]$sym)
weighted.mean(all$vp[[3]]$sym, w = rowSums(assay(all$vsd[[3]])))
```
Plot sources of variance for each colony
```{r plot_varpart, eval = TRUE}
varpartfig <- allsumm %>% 
  unnest(vpsum, .drop = T) %>%
  gather(source, value, -Colony) %>%
  group_by(Colony) %>%
  mutate(relvar = value / sum(value)) %>%   # TOTAL VARIANCE NOT SAME FOR EACH COLONY -- NORMALIZE WITHIN COLONY
  filter(source != "Residuals") %>%
  #mutate(source = recode(source, trt2 = "temp", `sym:trt2` = "sym:temp")) %>%
  ggplot(aes(x = source, y = relvar*100, fill = colony)) +
  geom_bar(stat = "identity") +
  facet_wrap(~ colony, labeller = global_labeller) +
  labs(x = "", y = "% variance explained") +
  theme_custom() +
  theme(legend.position = "none",
        strip.text.x = element_text(size = 6))

varpartfig

save(varpartfig, file = "output/varpartfig.RData")

ggsave(filename = "output/varpart.png", plot = varpartfig, width = 69, height = 84.5, units = "mm")

fig2 <- cowplot::plot_grid(pcoa, varpartfig, rel_widths = c(2, 1), labels = "AUTO")
ggsave(filename = "figures/fig2.png", fig2, width = 169, height = 84.5, units = "mm")


allsumm %>% 
  unnest(vpsum, .drop = T) %>%
  gather(source, value, -colony) %>%
  group_by(colony) %>%
  mutate(relvar = value / sum(value)) %>%   # TOTAL VARIANCE NOT SAME FOR EACH COLONY -- NORMALIZE WITHIN COLONY
  filter(source != "Residuals") %>%
  summarise(total_explained = sum(relvar))
```

### Discriminant analysis of principal components (DAPC)
```{r DAPC}
# Discriminant function including all genes
vsd_col <- limma::removeBatchEffect(assay(vsd), vsd$Colony)
# use batch-colony-removed vst counts
dat <- data.frame(vsd_col) 
# use regular vst counts
#dat <- data.frame(assay(vsd))

# How many PCs should be kept?
# dapc2 <- dapc(t(dat), colData(hs.dds)$group, n.da=4, n.pca=35)
# temp <- optim.a.score(dapc2, n.sim = 5)
# my.dapc <- function(n.pca) dapc(t(dat), colData(hs.dds)$group, n.pca = n.pca, n.da = 4)

# library(furrr)
# plan(multiprocess)
# my.dapc.res <- tibble(n.pca = 5:20) %>%
#   mutate(dapc = map(n.pca, my.dapc),
#          a.score = furrr::future_map(dapc, a.score, n.sim = 1000),
#          mean = map_dbl(a.score, ~ .$mean),
#          cumvar = map_dbl(dapc, ~ .$var))
# 
# my.dapc.res %>%
#   arrange(-mean) %>%
#   head()

# Retaining 6 PC's gives highest a-score (25%), but only utilizes 44% of variance. Retaining 15 PC's reduces a-score to 17%, but retains 72% of variance. Therefore, let's use 15 PC's for the DA. 

dp1 <- dapc(t(dat), colData(dds)$Treatment,
            n.pca = 15, n.da = 2)   # Retain 15 PCs and 2 discriminant functions
scatter(dp1,bg="white",scree.da=TRUE,scree.pca=TRUE,legend=TRUE) 
# Axis one represents antibiotics treatment response, and effect of D under control moving toward this transcriptional state.  Axis two represents the heat stress response.

dapc <- tibble(SampleID = rownames(dp1$ind.coord),
               grp = dp1$grp,
               Colony = coldata$Colony,
               Treatment = coldata$Treatment,
               LD1 = dp1$ind.coord[,1],
               LD2 = dp1$ind.coord[,2])
dapc <- dapc %>%
  group_by(grp) %>%
  summarize(c1 = mean(LD1),
            c2 = mean(LD2)) %>%
  full_join(dapc) 
#set factor orders 
dapc$Colony <- factor(dapc$Colony, levels = c("HW1", "HW2", "WT1", "WT2"), ordered = TRUE)
dapc$Treatment <- factor(dapc$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat"), ordered = TRUE)
```
```{r}
# Plot with spiders
dapc.fig <- 
  ggplot(dapc, aes(fill = Treatment)) +
  #sample-centroid spiders paths
  geom_segment(mapping = aes(x = LD1, y = LD2, xend = c1, yend = c2), lwd = 0.25, col = "dark grey") +
  #treatment centroid points
  geom_point(aes(x = c1, y = c2, color = Treatment), size = 4, fill = "black", shape = 21, stroke = 2, show.legend = FALSE) +
  #sample points
  geom_point(aes(x = LD1, y = LD2, shape = Colony), size = 3, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_shape_manual(name = "Colony", values = colshapes) +
  scale_fill_manual(name = "Treatment", values = condcolors_AxH) +
  guides(shape = guide_legend(override.aes = list(shape = colshapes))) +
 # guides(shape = guide_legend(override.aes = list(fill = "black", size = 2))) +
  labs(x = "LD1", y = "LD2")# +
  #theme_custom() +
  #theme(legend.position = c(0.25, 0.15)) +
  #theme(plot.margin = margin(t = 0, r = 0.9, unit = "cm"))# +
  #theme(legend.spacing.y = unit(0, "cm"),
  #      legend.spacing.x = unit(0, "cm")) +
  #theme(legend.box = "horizontal") +
 # theme(legend.position = "none") +
 # annotate(geom = "text", x = -1.5, y = -1.6, adj = 1, label = "sym / temp", size = 3.5) +
 # annotate(geom = "text", x = -1.5, y = -2.825, adj = 1, size = 2.5,
  #         label = "C / control\nC / heated\nD / control\nD / heated") +
 # annotate(geom = "point", x = -3.2, y = seq(from = -2.15, by = -0.45, length.out = 4), 
#           size = 2, shape = c(21, 21, 24, 24), fill = c("black", "white", "black", "white"))
dapc.fig
```
```{r}
xplot <- 
  ggdensity(dapc, x = "LD1", fill = "Treatment", lwd = 0.5,
            palette = condcolors_AxH) +
  clean_theme() + rremove("legend") + 
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
#ggplot(dapc, aes(x = LD1, fill = grp)) + geom_density(alpha = 0.5, adjust = 1)
xplot

yplot <- 
  ggdensity(dapc, x = "LD2", fill = "Treatment", lwd = 0.5,
            palette = condcolors_AxH) +
  clean_theme() + rremove("legend") + 
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
yplot

dapcplot <- plot_grid(xplot, NULL, dapc.fig, ncol = 1, align = "hv", rel_heights = c(0.7, -0.28, 2))
dapcplot
ggsave(filename = "./manuscript_figures/FigS7_DAPC.png", plot = dapcplot, width = 5, height = 5, units = "in")
```
The DAPC shows...

```{r, include = FALSE, eval = FALSE}
# DAPC - axis of heat stress in C corals (Ccontrol vs. Cheat), then see where D's fall along it (sensu Kenkel and Matz)

# use batch-colony-removed vst counts
vsd2 <- vsd
#assay(vsd2) <- limma::removeBatchEffect(assay(vsd), vsd$colony)
## Subset C-control and C-heat corals
CcCh <- vsd2[, colData(vsd2)$group %in% c("Cc", "Ch")]
dat <- data.frame(assay(CcCh))

# Fit DAPC axis
dp1 <- dapc(t(dat), colData(CcCh)$group, n.pca = 12, n.da = 1)
scatter(dp1, scree.pca = T)

# Calculate values along this axis
## Subset D-control and D-heat corals
DcDh <- vsd2[, colData(vsd2)$group %in% c("Dc", "Dh")]
dat2 <- data.frame(assay(DcDh))
## Get scores for these samples along the Cc-Ch DAPC axis
pred.sup <- predict.dapc(dp1, newdata=(t(dat2)))

# Create dapc object to plot new scores
newdp <- dp1
newdp$ind.coord<-pred.sup$ind.scores
newdp$posterior<-pred.sup$posterior
newdp$assign<-pred.sup$assign
newdp$grp <- colData(DcDh)$group
scatter(newdp)

# Get scores for all groups
dpc <- tibble(group = c(as.character(dp1$grp), as.character(newdp$grp)),
              core = c(rownames(dp1$ind.coord), rownames(newdp$ind.coord)),
              colony = str_sub(core, 3, 4),
              score = c(dp1$ind.coord, newdp$ind.coord))
ggplot(dpc, aes(x = score, fill = group)) + geom_density(adjust = 1, alpha = 0.5)
ggplot(dpc, aes(x = group, y = score, shape = colony, color = colony)) + geom_point()

# Test for differences between groups
mod <- lm(score ~ group, data = dpc)
anova(mod)
lsm <- lsmeans(mod, specs = "group")
contrast(lsm, "pairwise")

# Ccontrol -- -1.94
# Cheat ---    1.94    (diff = 3.89)
# Dcontrol -- -0.16


# I deemed this approach inappropriate because we shouldn't necessarily constrain the responses of D corals to occur along an axis of heat stress for C corals... they could respond in an entirely different way along an entirely different axis. Therefore, the DAPC with all four groups is more appropriate. LD1 still ends up corresponding very strongly to the axis of heat stress for C corals, and we still C that shuffling to D moves corals toward the heat stress end of the axis. We can still do a statistical test on LD1 values for different groups. 
```


## Differential gene expression analysis (DESeq2)
```{r genotypic DESeq}
#First, genotype-specific treatement models
design(dds) <- formula(~ Colony.Treatment)
#Set control treatment as reference factor level
dds$Treatment <- factor(dds$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds$Colony.Treatment, ref = "HW1.control")
#Perform DESeq2 analysis
dsr1 <- DESeq(dds)
resultsNames(dsr1)
# Define contrasts
contrasts <- tibble(num = c("Heat", "Antibiotics", "Antibiotics.Heat", rep("Antibiotics.Heat", 2)), 
                    den = c(rep("control", 3), "Heat", "Antibiotics"))
# Create tibble with DESeq results for all treatment contrasts for each colony
DE <- crossing(Colony = c("HW1", "HW2", "WT1", "WT2"), contrasts) %>%
   mutate(dsr = pmap(list(Colony, num, den), function(x, y, z) {
   results(dsr1, contrast = c("Colony.Treatment", paste0(x, ".", c(y, z))))}))
```
```{r overall DESeq}
#Second, overall colony and treatment interaction model
design(dds) <- formula(~ Batch + Colony + Treatment + Colony:Treatment)
#Set control treatment as reference factor level
dds$Treatment <- factor(dds$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds$Treatment, ref = "control")
#Perform DESeq2 analysis
dsr2 <- DESeq(dds)
resultsNames(dsr2)
# Get DESeq results for all treatment contrasts
DE <- crossing(Colony = "all", contrasts) %>%
  mutate(dsr = map2(num, den, ~ results(dsr2, contrast = c("Treatment", .x, .y)))) %>%
  bind_rows(DE)
```
```{r DEG table}
DE <- DE %>%
  mutate(df = map(dsr, ~ rownames_to_column(data.frame(.), "ID")),
         df = map(df, ~ left_join(., gene_annotation, by = "ID")))
# Filter out significant up and down-regulated genes
DE <- DE %>%
  mutate(sig = map(df, ~ filter(., padj < 0.05)),
          up = map(sig, ~ filter(., log2FoldChange > 0)),
        down = map(sig, ~ filter(., log2FoldChange < 0))) 

# Generate signed logP values for all differential expression contrasts
DE <- DE %>% mutate(logP = map(dsr, ~ data.frame(
  gene = rownames(data.frame(.)),
  logP = -log10(data.frame(.)$pvalue) * sign(data.frame(.)$log2FoldChange))))

# Count number of differentially expressed genes within each colony, and overall, for each contrast
DEtab <- DE %>%
  mutate(nsig = map_dbl(sig, ~ nrow(.)),
         nup = map_dbl(up, ~ nrow(.)),
         ndn = map_dbl(down, ~ nrow(.)),
         `DEGs [up, down]` = paste0(nsig, " [", nup, ", ", ndn, "]")) %>%
  mutate(Colony = factor(Colony, levels = c("all", "HW1", "HW2", "WT1", "WT2"))) %>%
  select(Colony, num, den, `DEGs [up, down]`)

DE_table <- DEtab %>% filter(den == "control") %>%
  knitr::kable(format = "markdown", caption = "Number of differentially expressed genes within individual colonies and across all colonies for each specified contrast. DEGs identified using DESeq with an adjusted p-value < 0.05. In brackets are the numbers of significantly up- and down-regulated genes.")
DE_table
```

The next chunk writes out the results tables as comma-separated files for all the differential expression results contrasts, with gene ID and annotation information for preparation as a supplemental table similar to the one with the LPS paper.

```{r sig_tables}
# Write significant DEG results tables to csv files
DE %>%
  filter(den == "control") %>%
  mutate(sig_file_out = paste0(num, "_", den, "_", Colony, "_", "sig", ".csv"),
         sig_file_out_path = file.path("./outputs/DESeq-results/tables/sig/", sig_file_out),
         data = walk2(sig, sig_file_out_path, write_csv))
# Write ALL DEG results tables to csv files
DE %>%
  filter(den == "control") %>%
  mutate(file_out = paste0(num, "_", den, "_", Colony, "_", "all", ".csv"),
         file_out_path = file.path("./outputs/DESeq-results/tables/all/", file_out),
         data = walk2(df, file_out_path, write_csv))
```


Why are there only 14 differentially expressed genes between the heat and control samples? --> Missing samples

### Assess similarity of responses across colonies
```{r DEgenes_colony_comparison}
shared_DEGs_colonies <- DE %>%
  unnest(sig) %>%
  filter(Colony != "all") %>%
  group_by(num, den) %>%
  dplyr::count(gene, log2FoldChange > 0) %>%   # In how many colonies is same gene DE in same direction?
  summarise(`1 colony` = sum(n==1), `2 colonies` = sum(n==2), `3 colonies` = sum(n==3), `4 colonies` = sum(n==4))
```
### Assess similarity of responses across treatments
```{r DEgenes_treatment_comparison}

```
```{r treatent_venn}

```


## DESeq results visualizations
### Volcano plots

For each of these plot styles, create data handling and plotting functions to process essential inputs into the plot
```{r}

```
```{r}
volcanoplot(DE$dsr[[1]])
```


### Gene expression heatmaps
```{r heatmap colors}
### Create heatmap annotation data frame
anno_col <- as.data.frame(colData(vsd)[, c("Colony", "Treatment")])

### Create colony and condition annotation colors
anno_colors <- list(
  Colony = c(HW1=colcolors[1], HW2=colcolors[2], WT1=colcolors[3], WT2=colcolors[4]),
  Treatment = c(control=condcolors_AxH[1],  Heat=condcolors_AxH[3], Antibiotics=condcolors_AxH[2], Antibiotics.Heat=condcolors_AxH[4]),
  DEG = c(Upregulated="red", Downregulated="blue"))

### Create sample heatmap cell color
colonyclustercol <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
### Create gene expression cell color
paletteLength <- 400
# Red --> Blue gradient
geneexpcolors <- colorRampPalette(rev(c("red", "yellow", "white", "cyan", "blue")))(paletteLength)
# Orange --> Cyan gradient
geneexpcolors1 <- colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(paletteLength)
# Red --> Blue dark gradient
geneexpcolors2 <- colorRampPalette(rev(c("red", "yellow", "grey10", "cyan", "blue")))(paletteLength)
### Color bars to preview
color.bar(geneexpcolors, -1, 1)
color.bar(geneexpcolors1, -1, 1)
color.bar(geneexpcolors2, -1, 1)
```
```{r sampleheatmap}
## Sample Clustering Heatmap
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix( sampleDists )

sample_heatmap <- pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colonyclustercol,
         annotation_colors = anno_colors,
         annotation_col = anno_col,
         annotation_row = anno_col,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 8,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 4,
         treeheight_row = 50,
         cutree_rows = 4)
```
```{r vargeneheatmap, message = FALSE, warning = FALSE}
### Create top variable gene expression matrix 
#Create heatmap matrix and annotation groupings
nvar <- 500
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), nvar)
matvar  <- assay(vsd)[ topVarGenes, ]
rownames(matvar) <- rowData(vsd)$ID[topVarGenes]
matvar  <- matvar - rowMeans(matvar)
pheatmap(matvar, 
         col = geneexpcolors,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Top Most Variable Gene Expression")
```
```{r deg_heatmaps}
### Create differentially expressed gene matrix 
#Create heatmap matrix and annotation groupings
mat_genes  <- assay(vsd)[DE$up[[1]]$gene, ]
row.names(mat_genes) <- DE$up[[1]]$gene#[1:95]
mat_genes <- mat_genes - rowMeans(mat_genes)
pheatmap(mat_genes, 
         col = geneexpcolors2,
         #breaks = seq(-3, 3, by = 0.015),
         legend = TRUE,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         annotation_names_col = FALSE,
         annotation_names_row = FALSE,
         show_rownames = TRUE,
         fontsize_row = 10,
         show_colnames = FALSE,
         fontsize_col = 10,
         treeheight_col = 50,
         cutree_cols = 1,
         treeheight_row = 40,
         cutree_rows = 1,
         main="Heatmap of Differential Gene Expression")
```

### Gene counts boxplots

### Double-treatment plots

## DESeq Gene Ontology (GO) enrichment analysis
### Heat vs. control 
```{r heat signed logP}
# Write overall Heat vs. control logP values to file for GO_MWU analysis
DE %>%
  filter(Colony == "all", num == "Heat", den == "control") %>%
  unnest(logP) %>%
  select(gene, logP) %>%
  write.table(., file = "./R/GO_MWU/heat.control.logP.txt", row.names = FALSE, quote = FALSE, sep = ",")
```
```{r heat GO_MWU}
# Run GO_MWU for Heat vs. control for Biological Process (BP) GO terms
commandArgs <- function(...) c("heat.control.logP.txt", "BP")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Heat vs. control for Molecular Function GO terms
commandArgs <- function(...) c("heat.control.logP.txt", "MF")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Heat vs. control for Cellular Component (CC) GO terms
commandArgs <- function(...) c("heat.control.logP.txt", "CC")
source("./R/GO_MWU/GO_MWU.R")
```
```{r heat.control_GO}
heat.control.GO <- as.list(
  c(BP = "./R/GO_MWU/MWU_BP_heat.control.logP.txt",
    MF = "./R/GO_MWU/MWU_MF_heat.control.logP.txt",
    CC = "./R/GO_MWU/MWU_CC_heat.control.logP.txt")
  ) %>% map(read.table, header = T) %>%
  bind_rows(.id = "ontology") %>%
  as_tibble() %>%
  filter(p.adj < 0.01)

heat.control.GO %>%
  select(ontology, delta.rank, name, p.adj) %>%
  arrange(ontology, delta.rank) %>%
  knitr::kable()
```
```{r heat GO_MWU plot}
# Plot GO_MWU results for Heat vs. control 
commandArgs <- function(...) c("heat.control.logP.txt", "BP", "GO_MWU.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("heat.control.logP.txt", "MF", "GO_MWU.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("heat.control.logP.txt", "CC", "GO_MWU.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
```
### Antibiotics vs. control 
```{r anti signed logP}
# Write overall Antibiotics vs. control logP values to file for GO_MWU analysis
DE %>%
  filter(Colony == "all", num == "Antibiotics", den == "control") %>%
  unnest(logP) %>%
  select(gene, logP) %>%
  write.table(., file = "./R/GO_MWU/anti.control.logP.txt", row.names = FALSE, quote = FALSE, sep = ",")
```
```{r anti GO_MWU}
# Run GO_MWU for Antibiotics vs. control for Biological Process (BP) GO terms
commandArgs <- function(...) c("anti.control.logP.txt", "BP")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Antibiotics vs. control for Molecular Function GO terms
commandArgs <- function(...) c("anti.control.logP.txt", "MF")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Antibiotics vs. control for Cellular Component (CC) GO terms
commandArgs <- function(...) c("anti.control.logP.txt", "CC")
source("./R/GO_MWU/GO_MWU.R")
```
```{r anti.control_GO}
anti.control.GO <- as.list(
  c(BP = "./R/GO_MWU/MWU_BP_anti.control.logP.txt",
    MF = "./R/GO_MWU/MWU_MF_anti.control.logP.txt",
    CC = "./R/GO_MWU/MWU_CC_anti.control.logP.txt")
  ) %>% map(read.table, header = T) %>%
  bind_rows(.id = "ontology") %>%
  as_tibble() %>%
  filter(p.adj < 0.01)

anti.control.GO %>%
  select(ontology, delta.rank, name, p.adj) %>%
  arrange(ontology, delta.rank) %>%
  knitr::kable()
```
```{r anti GO_MWU plot}
# Plot GO_MWU results for Antibiotics vs. control 
commandArgs <- function(...) c("anti.control.logP.txt", "BP", "GO_MWU.anti.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("anti.control.logP.txt", "MF", "GO_MWU.anti.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("anti.control.logP.txt", "CC", "GO_MWU.anti.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
```
### Antibiotics.Heat vs. control 
```{r anti.heat signed logP}
# Write overall Antibiotics.Heat vs. control logP values to file for GO_MWU analysis
DE %>%
  filter(Colony == "all", num == "Antibiotics.Heat", den == "control") %>%
  unnest(logP) %>%
  select(gene, logP) %>%
  write.table(., file = "./R/GO_MWU/anti.heat.control.logP.txt", row.names = FALSE, quote = FALSE, sep = ",")
```
```{r anti.heat GO_MWU}
# Run GO_MWU for Antibiotics.Heat vs. control for Biological Process (BP) GO terms
commandArgs <- function(...) c("anti.heat.control.logP.txt", "BP")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Antibiotics.Heat vs. control for Molecular Function GO terms
commandArgs <- function(...) c("anti.heat.control.logP.txt", "MF")
source("./R/GO_MWU/GO_MWU.R")

# Run GO_MWU for Antibiotics.Heat vs. control for Cellular Component (CC) GO terms
commandArgs <- function(...) c("anti.heat.control.logP.txt", "CC")
source("./R/GO_MWU/GO_MWU.R")
```
```{r anti.heat.control_GO}
anti.heat.control.GO <- as.list(
  c(BP = "./R/GO_MWU/MWU_BP_anti.heat.control.logP.txt",
    MF = "./R/GO_MWU/MWU_MF_anti.heat.control.logP.txt",
    CC = "./R/GO_MWU/MWU_CC_anti.heat.control.logP.txt")
  ) %>% map(read.table, header = T) %>%
  bind_rows(.id = "ontology") %>%
  as_tibble() %>%
  filter(p.adj < 0.01)

anti.heat.control.GO %>%
  select(ontology, delta.rank, name, p.adj) %>%
  arrange(ontology, delta.rank) %>%
  knitr::kable()
```
```{r anti.heat GO_MWU plot}
# Plot GO_MWU results for Antibiotics.Heat vs. control 
commandArgs <- function(...) c("anti.heat.control.logP.txt", "BP", "GO_MWU.anti.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("anti.heat.control.logP.txt", "MF", "GO_MWU.anti.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
commandArgs <- function(...) c("anti.heat.control.logP.txt", "CC", "GO_MWU.anti.heat.control.png")
source("./R/GO_MWU/GO_MWU_plot.R")
```
```{r allGO}
#Supplemental table with all GO results
allGO <- bind_rows(.id = "contrast",
  heat.control = heat.control.GO,
  anti.control = anti.control.GO,
  anti.heat.control = anti.heat.control.GO
) %>%
  select(contrast, ontology, delta.rank, name, p.adj) %>%
  arrange(contrast, ontology, sign(delta.rank), p.adj)

write_csv(allGO, path = "./outputs/DESeq-results/GO_MWU/allGO.csv")
```

## EuKaryotic Orthologous Groups (KOG) enrichment analysis

Using the signed log(p-values) from the differential expression analysis, we can analyze which KOG classifications are up- or down-regulated within each colony, and across all four colonies together.

```{r KOG, results = 'hide'}
# Ensure gene KOG annotation is present
head(gene2kog)
# Run KOG.MWU analysis on all contrasts
KOG <- DE %>%
  mutate(KOG = map(logP, ~ kog.mwu(., gene2kog)),
         # If a KOG term has <10 genes associated with it, set delta rank to zero
         KOG = map(KOG, ~ mutate(., delta.rank = ifelse(nseqs < 10, 0, delta.rank))))
# Generate all KOG delta rank tables
kogtables <- KOG %>%
  unnest(KOG) %>%
  select(Colony, num, den, term, delta.rank) %>%
  split(list(.$num, .$den), drop = TRUE) %>%
  map(~ spread(., Colony, delta.rank)) %>%
  map(~ column_to_rownames(select(., -num, -den), "term"))

# Generate KOG p-value stars
kogpvals <- KOG %>%
  unnest(KOG) %>%
  select(Colony, num, den, term, padj) %>%
  split(list(.$num, .$den), drop = TRUE) %>%
  map(~ spread(., Colony, padj)) %>%
  map(~ column_to_rownames(select(., -num, -den), "term")) %>%
  map(~ transmute_all(., gtools::stars.pval))
```
```{r KOG_heatmaps}
KOG_heat.control <- KOGheatmap(kogtables$Heat.control, kogpvals$Heat.control, main = "Heat vs. Control", 
                      fontsize = 6)
ggsave(filename = "./outputs/DESeq-results/KOGMWU/figures/KOGheat.control.png", plot = KOG_heat.control, height = 2.5, width = 4, units = "in")
KOG_anti.control <- KOGheatmap(kogtables$Antibiotics.control, kogpvals$Antibiotics.control, main = "Antibiotics vs. Control", 
                      fontsize = 6)
ggsave(filename = "./outputs/DESeq-results/KOGMWU/figures/KOGanti.control.png", plot = KOG_anti.control, height = 2.5, width = 4, units = "in")
KOG_anti.heat.control <- KOGheatmap(kogtables$Antibiotics.Heat.control, kogpvals$Antibiotics.Heat.control, main = "Antibiotics + Heat vs. Control", 
                      fontsize = 6)
ggsave(filename = "./outputs/DESeq-results/KOGMWU/figures/KOGanti.heat.control.png", plot = KOG_anti.heat.control, height = 2.5, width = 4, units = "in")
```
```{r KOG_treatment contrasts}
kogtable_contrasts <- KOG %>%
  unnest(KOG) %>%
  filter(Colony == "all") %>%
  select(num, den, term, delta.rank) %>%
  unite("contrast", num, den, sep = ".") %>%
  spread(contrast, delta.rank) %>%
  column_to_rownames("term")
  
kogpval_constrasts <- KOG %>%
  unnest(KOG) %>%
  filter(Colony == "all") %>%
  select(num, den, term, padj) %>%
  unite("contrast", num, den, sep = ".") %>%
  spread(contrast, padj) %>%
  column_to_rownames("term") %>%
  transmute_all(gtools::stars.pval)

KOG_treatments <- KOGheatmap(
  select(kogtable_contrasts, Heat.control, Antibiotics.control, Antibiotics.Heat.control),
  select( kogpval_constrasts, Heat.control, Antibiotics.control, Antibiotics.Heat.control),
  labels_col = c(expression("Heat"~"vs."~"Control"),
                 expression("Antibiotics"~"vs."~"Control"),
                 expression("Antibiotics + Heat"~"vs."~"Control")),
  angle_col = 315,
  fontsize = 8)

ggsave(filename = "./outputs/DESeq-results/KOGMWU/figures/KOG_treatment_heatmap.png", KOG_treatments, width = 4.8, height = 3.5, units = "in")
ggsave(filename = "./manuscript_figures/Fig2B_KOGheatmap.png", KOG_treatments, width = 4.8, height = 5, units = "in")
```

## WGCNA

## WGCNA Gene Ontology (GO) enrichment analysis
## Mike's old approach
### Select which treatments to use in DESeq object construction
anti: Control-Antibiotics
heat: Control-Heat
anti.heat: Control-Antibiotics.Heat
antixheat: Control-Antibiotics-Heat-Antibiotics.Heat
```{r}
countdata_anti <- dplyr::select(countdata.sorted, matches("[HW][wt][12].[56]."))
countdata_heat <- dplyr::select(countdata.sorted, matches("[HW][wt][12].[46]."))
countdata_anti.heat <- dplyr::select(countdata.sorted, matches("[HW][wt][12].[16]."))
###
countdata_antixheat <- dplyr::select(countdata.sorted, matches("[HW][wt][12].[1456]."))
```
```{r}
countdata_HW1 <- dplyr::select(countdata_antixheat, matches("Hw1.*"))
countdata_HW2 <- dplyr::select(countdata_antixheat, matches("Hw2.*"))
countdata_WT1 <- dplyr::select(countdata_antixheat, matches("Wt1.*"))
countdata_WT2 <- dplyr::select(countdata_antixheat, matches("Wt2.*"))
```

```{r}
samples_anti <- filter(samples, Treatment == "control" | Treatment == "Antibiotics")
samples_heat <- filter(samples, Treatment == "control" | Treatment == "Heat")
###
samples_anti.heat <- filter(samples, Treatment == "control" | Treatment == "Antibiotics.Heat")
###
samples_antixheat <- filter(samples, Treatment == "control" | Treatment == "Antibiotics" | Treatment == "Heat" | Treatment == "Antibiotics.Heat")
```
```{r}
samples_HW1 <- filter(samples_antixheat, Colony == "HW1")# %>% select(-contains("W"), -contains("2"))
samples_HW2 <- filter(samples_antixheat, Colony == "HW2")# %>% select(-contains("W"), -contains("1"))
samples_WT1 <- filter(samples_antixheat, Colony == "WT1")# %>% select(-contains("H"), -contains("2"))
samples_WT2 <- filter(samples_antixheat, Colony == "WT2")# %>% select(-contains("H"), -contains("1"))
```
### Convert to matrices 
```{r, include=FALSE}
      countdata_anti <- as.matrix(countdata_anti)
      countdata_heat <- as.matrix(countdata_heat)
      countdata_anti.heat <- as.matrix(countdata_anti.heat)
      countdata_antixheat <- as.matrix(countdata_antixheat)
```
```{r, include=FALSE}
      countdata_HW1 <- as.matrix(countdata_HW1)
      countdata_HW2 <- as.matrix(countdata_HW2)
      countdata_WT1 <- as.matrix(countdata_WT1)
      countdata_WT2 <- as.matrix(countdata_WT2)
```



### Create DESeq data objects
### DESeq2 treatment model objects
```{r, message=FALSE}
      dds_anti <- DESeqDataSetFromMatrix(countData=countdata_anti, colData=samples_anti, design= ~Batch + Colony + Treatment)
      dds_anti$Treatment <- factor(dds_anti$Treatment, levels=c("control", "Antibiotics"))
relevel(dds_anti$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_heat <- DESeqDataSetFromMatrix(countData=countdata_heat, colData=samples_heat, design= ~Batch + Colony + Treatment)
      dds_heat$Treatment <- factor(dds_heat$Treatment, levels=c("control", "Heat"))
relevel(dds_heat$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_anti.heat <- DESeqDataSetFromMatrix(countData=countdata_anti.heat, colData=samples_anti.heat, design= ~Batch + Colony + Treatment)
      dds_anti.heat$Treatment <- factor(dds_anti.heat$Treatment, levels=c("control", "Antibiotics.Heat"))
relevel(dds_anti.heat$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_antixheat <- DESeqDataSetFromMatrix(countData=countdata_antixheat, colData=samples_antixheat, design= ~Batch + Colony + Treatment)
      dds_antixheat$Treatment <- factor(dds_antixheat$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_antixheat$Treatment, ref = "control")
```
### DESeq2 genotype model objects
```{r, message=FALSE}
      dds_HW1 <- DESeqDataSetFromMatrix(countData=countdata_HW1, colData=samples_HW1, design= ~Treatment)
### Batch effect encompasses Treatment
      dds_HW1$Treatment <- factor(dds_HW1$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_HW1$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_HW2 <- DESeqDataSetFromMatrix(countData=countdata_HW2, colData=samples_HW2, design= ~Batch + Treatment)
      dds_HW2$Treatment <- factor(dds_HW2$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_HW2$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_WT1 <- DESeqDataSetFromMatrix(countData=countdata_WT1, colData=samples_WT1, design= ~Batch + Treatment)
      dds_WT1$Treatment <- factor(dds_WT1$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_WT1$Treatment, ref = "control")
```
```{r, message=FALSE}
      dds_WT2 <- DESeqDataSetFromMatrix(countData=countdata_WT2, colData=samples_WT2, design= ~Batch + Treatment)
      dds_WT2$Treatment <- factor(dds_WT2$Treatment, levels=c("control", "Antibiotics", "Heat", "Antibiotics.Heat"))
relevel(dds_WT2$Treatment, ref = "control")
```
### Check annotation and dds object rowname order coherence
```{r}
all(rownames(dds_anti) == rownames(gene_annotation))
all(rownames(dds_heat) == rownames(gene_annotation))
all(rownames(dds_anti.heat) == rownames(gene_annotation))
all(rownames(dds_antixheat) == rownames(gene_annotation))
```
```{r}
all(rownames(dds_HW1) == rownames(gene_annotation))
all(rownames(dds_HW2) == rownames(gene_annotation))
all(rownames(dds_WT1) == rownames(gene_annotation))
all(rownames(dds_WT2) == rownames(gene_annotation))
```

### Add gene feature annotation to DESeqDataSets
```{r}
mcols(dds_anti) <- cbind(mcols(dds_anti), gene_annotation)
mcols(dds_heat) <- cbind(mcols(dds_heat), gene_annotation)
mcols(dds_anti.heat) <- cbind(mcols(dds_anti.heat), gene_annotation)
mcols(dds_antixheat) <- cbind(mcols(dds_antixheat), gene_annotation)
```
```{r}
mcols(dds_HW1) <- cbind(mcols(dds_HW1), gene_annotation)
mcols(dds_HW2) <- cbind(mcols(dds_HW2), gene_annotation)
mcols(dds_WT1) <- cbind(mcols(dds_WT1), gene_annotation)
mcols(dds_WT2) <- cbind(mcols(dds_WT2), gene_annotation)
```

### Perform DESeq2 analyses
```{r, message=FALSE}
      dds_anti <- DESeq(dds_anti)
      dds_heat <- DESeq(dds_heat)
      dds_anti.heat <- DESeq(dds_anti.heat)
      dds_antixheat <- DESeq(dds_antixheat)
```
```{r, message=FALSE}
      dds_HW1 <- DESeq(dds_HW1)
      dds_HW2 <- DESeq(dds_HW2)
      dds_WT1 <- DESeq(dds_WT1)
      dds_WT2 <- DESeq(dds_WT2)
```

### Filter low counts
```{r}
      keep <- rowSums(counts(dds_antixheat), na.rm = TRUE) >= 10
      dds_anti <- dds_anti[keep,]
      dds_heat <- dds_heat[keep,]
      dds_anti.heat <- dds_anti.heat[keep,]
      dds_antixheat <- dds_antixheat[keep,]
```
```{r}
      dds_HW1 <- dds_HW1[keep,]
      dds_HW2 <- dds_HW2[keep,]
      dds_WT1 <- dds_WT1[keep,]
      dds_WT2 <- dds_WT2[keep,]
```

### Obtain DESeq results
```{r}
      res_anti <- results(dds_anti)
      res_heat <- results(dds_heat)
      res_anti.heat <- results(dds_anti.heat)
      res_antixheat <- results(dds_antixheat)
```
```{r}
      res_HW1 <- results(dds_HW1)
      res_HW2 <- results(dds_HW2)
      res_WT1 <- results(dds_WT1)
      res_WT2 <- results(dds_WT2)
```

### Check output of Antibiotics DESeq2 analysis to ensure quality
```{r}
resultsNames(dds_anti)
summary(res_anti)
```
```{r}
res_anti$IDGeneInfo <- mcols(dds_anti)$IDGeneInfo
res_anti$IDGeneInfo <- as.character(res_anti$IDGeneInfo)
```
```{r}
plotDispEsts(dds_anti)
plotMA(res_anti, ylim = c(-20, 20))
```

### Check output of Heat DESeq2 analysis to ensure quality
```{r}
resultsNames(dds_heat)
summary(res_heat)
```

```{r}
res_heat$ID <- mcols(dds_heat)$ID

res_heat$ID <- as.character(res_heat$ID)
```
```{r}
plotDispEsts(dds_heat)
plotMA(res_heat, ylim = c(-20, 20))
```

### Check output of Antibiotics.Heat DESeq2 analysis to ensure quality
```{r}
resultsNames(dds_anti.heat)
summary(res_anti.heat)
```
```{r}
res_anti.heat$IDGeneInfo <- mcols(dds_anti.heat)$IDGeneInfo
res_anti.heat$IDGeneInfo <- as.character(res_anti.heat$IDGeneInfo)
```
```{r}
plotDispEsts(dds_anti.heat)
plotMA(res_anti.heat, ylim = c(-20, 20))
```

### Check output of AntibioticsXHeat DESeq2 analysis to ensure quality
```{r}
resultsNames(dds_antixheat)
summary(res_antixheat)
```
```{r}
res_antixheat$IDGeneInfo <- mcols(dds_antixheat)$IDGeneInfo
res_antixheat$IDGeneInfo <- as.character(res_antixheat$IDGeneInfo)
```
```{r}
plotDispEsts(dds_antixheat)
plotMA(res_antixheat, ylim = c(-20, 20))
```
