---
title: "Pocillopora_AXH_microbiome_analysis"
author: "Mike Connelly"
date: "7/28/2020"
output: 
  html_document:
      code_folding: hide
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/projects/EAPSI_Pocillopora_AXH")
```
## Setup packages and working directories
```{r packages, include=FALSE}
library("plyr")
library("tidyverse")
# Essential microbiome analysis packages
library("phyloseq")
library("microbiome")
library("vegan")
library("ape")
library("ALDEx2")


# Data visualization packages
library("pheatmap")
library("data.table")
library("DT")
# Graphics packages
library("cowplot")
library("patchwork")
library("gridExtra")
library("ggthemes")
library("ggpubr")
library("ggrepel")
library("ggnewscale")
library("RColorBrewer")
library("wesanderson")
library("circlize")
library("stringr")
library("extrafont")
library("extrafontdb")
# extended visualization functions
source("./R/ggrare.R")
source("./R/AXH_functions.R")
```
```{r colors}
### Set overall theme, colors, and shapes for ggplot2
# remember to set theme_mec() from transcriptomic script for proper formatting
theme_set(theme_mec())
# colors for experimental treatments
condcolors_AxH <- c("blue", "darkorange", "cyan", "gold")
condcolors_heat <- c(condcolors_AxH[1:2])
condcolors_anti <- c(condcolors_AxH[1], condcolors_AxH[3])
condcolors_anti.heat <- c(condcolors_AxH[1], condcolors_AxH[4])
condfillcolors_AxH <- c("#99CCFF", "#FFCC99", "#99FFFF", "#FFFF66") # fill colors for some figures
# shapes and colors for colonies
colshapes <- c(21, 24, 22, 23)
colcolors <- c("olivedrab3", "springgreen", "deepskyblue", "skyblue")
# null colors for presentation figures
condcolors_null <- c(rep("black", 4))
colshapes_null <- c(rep(20, 8))
```
```{r taxa colors, include=FALSE}
# Bacteria taxa
# Phyla
phyla_colors <- read_csv("./Rmd/taxa_colors/phyla_colors.csv", col_names = c("phylum", "color"))
# Classes
classes_colors <- read_csv("./Rmd/taxa_colors/classes_colors.csv", col_names = c("phylum", "class", "color"))
# Orders
orders_colors <- read_csv("./Rmd/taxa_colors/orders_colors.csv", col_names = c("phylum", "class", "order", "color"))
# Families
families_colors <- read_csv("./Rmd/taxa_colors/families_colors3.csv", col_names = c("phylum", "class", "order", "family", "color"))
# Genera
#genera_colors <- 
# Specices
#species_colors <- 
```
## Read in ASV table, taxonomy, sample data
```{r asv}
# Read in ASV table
ASV <- read.table("./outputs/qiime2/export_silva/all_silva_feature-table.tsv", row.names = 1, header = TRUE)
ASV <- t(ASV)
```
```{r taxomony}
# Read in taxononomic classification information
taxonomy <- as.matrix(read.table("./outputs/qiime2/export_silva/taxonomy4.txt", sep = "\t", row.names = 1, header = FALSE, fill = TRUE))
colnames(taxonomy) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
taxonomy[taxonomy==""] <- NA
taxdf <- as.data.frame(taxonomy)
```
```{r tree_metadata}
# Read in phylogenetic tree, sample metadata and set factor orders
tree <- read_tree("./outputs/qiime2/export_silva/tree.nwk")
map <- read.table("./data/qiime2_metadata.tsv", sep ="\t", row.names = 1, header = FALSE)
colnames(map) <- c ("Reef",	"Colony",	"Environment",	"Treatment")
map$SampleID <- rownames(map)
map$Reef <- factor(map$Reef, levels = c("Houwan", "Wanglitung", "controls"), ordered = TRUE)
map$Colony <- factor(map$Colony, levels = c("HW1", "HW2", "WT1", "WT2", "controls"), ordered = TRUE)
map$Treatment <- factor(map$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat", "positive", "negative"), ordered = TRUE)
```
## Create phyloseq object
```{r phyloseq}
ps <- phyloseq(otu_table(ASV, taxa_are_rows = FALSE),
               sample_data(map),
               tax_table(taxonomy),
               phy_tree(tree))
ps
# summarize_phyloseq(ps)
```
```{r ps_AxH_all}
# Pre-process, filter, and QC phyloseq object
# Stable site control, Heat, Antibiotics, Antibiotics.Heat sample subset
nsamples(ps)
samples <- sample_names(ps)
# regex to select Hw1, Hw2, Wt1, and Wt2 genotypes in control (6), heat stress (4), antibiotics (5), and combined (1) treatments
samples.AXH <- grep("[HW][wt][1-2].[1456][abc]", samples, value = TRUE) 
ps.AXH_allASVs <- prune_samples(samples.AXH, ps)
ps.AXH_allASVs
# 48 samples, 5 sample variables
```
```{r ps_assigned}
# Retain only assigned taxa
ps_assigned <- subset_taxa(ps, Kingdom!="Unassigned")
ps_assigned
# 532 taxa and 7 taxonomic ranks
# Filter taxa that have a mean count of 5 across all coral samples
ps_filtered <- filter_taxa(ps_assigned, function(x) mean(x) > 5, TRUE)
ps_filtered
# ?filter_taxa
```
```{r ps_AxH_assigned}
nsamples(ps_filtered) # 53
samplenames <- sample_names(ps_assigned)

# select Hw1, Hw2, Wt1, and Wt2 genotypes in control (6), heat stress (4), antibiotics (5), and combined (1) treatments
ps.AXH <- prune_samples(samples.AXH, ps_filtered)
ps.AXH # 48
summarize_phyloseq(ps.AXH)
```
```{r ps_transformations}
# Relative abundance transformation
ps.AXH.rel <- transform_sample_counts(ps.AXH, function(x) {x / sum(x)} )
# Centered log-ratio transformation, recommended by Gloor et al. 2017: https://doi.org/10.3389/fmicb.2017.02224
ps.AXH.clr <- microbiome::transform(ps.AXH, "clr")
# summarize_phyloseq(ps.AXH.clr)
```
```{r controls}
# Control sample subset
nsamples(ps_assigned)
samples <- sample_names(ps_assigned)
samplescontrol <- grep("[HW][wt][1-2].[6][abc]", samples, value = TRUE)
ps.control <- prune_samples(samplescontrol, ps_assigned)
ps.control
# summarize_phyloseq(ps.control)
ps.control.rel <- transform_sample_counts(ps.control, function(x) {x / sum(x)} )
ps.control.clr <- microbiome::transform(ps.control, "clr")
```

## Quality control plots
```{r rd}
# Read in read depth information
readdepth <- read_csv("./outputs/qiime2/QC/per-sample-fastq-counts.csv")
colnames(readdepth) <- c("SampleID", "Count")
readdepth <- merge.data.frame(map, readdepth, by = "SampleID")
```
### Read depth QC plot
```{r rd_qc}
readdepth.AXH <- dplyr::filter(readdepth, Treatment == "control" | Treatment == "Heat" | Treatment == "Antibiotics" | Treatment == "Antibiotics.Heat" | Treatment == "positive" | Treatment == "negative") %>% dplyr::filter(Reef == "Houwan" | Reef == "Wanglitung" | Reef == "controls")
###
readdepth.AXH$SampleID <- gsub("\\.6", "_control_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.5", "_anti_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.4", "_heat_", readdepth.AXH$SampleID)
readdepth.AXH$SampleID <- gsub("\\.1", "_anti+heat_", readdepth.AXH$SampleID)
###
readdepth.AXH$Treatment <- factor(readdepth.AXH$Treatment, levels = c("control", "Antibiotics", "Heat", "Antibiotics.Heat", "positive", "negative"), ordered = TRUE)
readdepth.AXH <- dplyr::arrange(readdepth.AXH, Treatment, Colony)
readdepth.AXH$SampleID <- factor(readdepth.AXH$SampleID, levels = readdepth.AXH$SampleID, ordered = TRUE)
###
ggdepth <- ggplot(readdepth.AXH, aes(SampleID, Count)) 
ggdepth +
  geom_bar(stat = "identity", aes(color = Treatment, fill = Treatment)) +
  #scale_y_continuous(trans = 'log10') +
  scale_color_manual(values = c(condcolors_AxH, "black", "black")) +
  scale_fill_manual(values = c(condfillcolors_AXH, "dark gray", "light gray")) +
  ylab("16S Read Count") +
  theme(axis.text.x = element_text(angle = 90)) +
  ggtitle("Total number of 16S reads per sample")
ggsave(file = "./outputs/phyloseq-results/figures/readdepth_AXH.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
ggsave(file = "./manuscript_figures/FigS##_16S_readdepth.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
```

### Rarefaction curve
```{r rare_curve}
grare <- ggrare(ps.AXH, step = 10, color = "Treatment", label = "Colony", plot = FALSE, se = TRUE)
grare + 
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condfillcolors_AXH) +
  #facet_wrap(~Treatment) +
  ggtitle("Rarefaction curve of 16S amplicon sequence variants")
ggsave(file = "./outputs/phyloseq-results/figures/rarefaction_.AXH.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
ggsave(file = "./manuscript_figures/FigS##_rarefactioncurve.pdf", width = 169, height = 85,  units = "mm", device = "pdf")
```


## Beta diversity ordination plots and statisitical tests
### Euclidean distances
```{r pca_rel}
ps_euc <- ordinate(ps.AXH.rel, "RDA", "euclidean")
p1 <- plot_ordination(ps.AXH.rel, ps_euc, type = "sample", color = "Treatment", shape = "Colony") +
  #stat_ellipse(aes(fill = Treatment, group = Treatment), geom = "polygon", type = "norm", alpha = 0.1) + 
#sample points
    geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
    scale_color_manual(values = condcolors_AxH, name = "Treatment") +
    scale_fill_manual(values = condcolors_AxH, name = "Treatment") +
    scale_shape_manual(values = colshapes, name = "Colony") + 
  ggtitle("PCA of 16S beta diversity", subtitle = "Relative abundance")
print(p1)
```
```{r}
# Calculate Treatment centroids for plotting
ord <- cbind(sample_data(ps.AXH),  ps.AXH.ord.PCoA.brayP$vectors)
ord_trmt <- ord %>%
  group_by(Treatment) %>%
  dplyr::summarise(c1 = mean(`Axis.1`), c2 = mean(`Axis.2`)) %>%    
  full_join(ord)
```

```{r pca_clr}
# The Aitchison distance is superior to both the widely used Jensen-Shannon divergence and the Bray-Curtis dissimilarity metrics, being more stable to subsetting and aggregating of the data, and being a true linear distance (Gloor et al. 2017)
ps_euc <- ordinate(ps.AXH.clr, "RDA", "euclidean")
pcaclr <- plot_ordination(ps.AXH.rel, ps_euc, type = "sample", color = "Treatment", shape = "Colony") +
    #sample points
    geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
    scale_color_manual(values = condcolors_AxH, name = "Treatment") +
    scale_fill_manual(values = condcolors_AxH, name = "Treatment") +
    scale_shape_manual(values = colshapes, name = "Colony") + 
  ggtitle("PCA of bacteria community beta diversity", subtitle = "Centered log ratio transformation")
print(pcaclr)
ggsave(pcaclr, filename = "./outputs/phyloseq-results/figures/pca_clr.pdf", width = 112, height = 80, units = "mm", device = "pdf")
#ggsave(pcaclr, filename = "./manuscript_figures/Fig3A_PCAclr_16S_Treatment.pdf", width = 112, height = 80, units = "mm", device = "pdf")
```

### Sample ordinations
Perform sample ordinations on unprocessed data, relative abundance-transformed data, and centered-log ratio-transformed data
Euclidean
Bray-Curtis
Unweighted UniFrac
Weighted UniFrac

```{r ordinations comparisons}
physeq <- ps.AXH.rel
dist <- "bray"
ord_meths <- c("DCA", "CCA", "RDA", "DPCoA", "NMDS", "MDS", "PCoA")
psord <- function(i, physeq, dist){
        ordi <- ordinate(physeq, method = i, distance = dist)
        plot_ordination(physeq, ordi, type = "sample", color = "Treatment", shape = "Colony", axes = c(1,2))
        }
plist <- llply(as.list(ord_meths), psord, physeq, dist)
names(plist) <- ord_meths

pdataframe = ldply(plist, function(x){
    df = x$data[, 1:2]
    colnames(df) = c("Axis_1", "Axis_2")
    return(cbind(df, x$data))
})
names(pdataframe)[1] = "method"

p <- ggplot(pdataframe, aes(Axis_1, Axis_2, color = Treatment, shape = Colony)) + 
#sample points
    geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
    scale_color_manual(values = condcolors_AxH, name = "Treatment") +
    scale_fill_manual(values = condcolors_AxH, name = "Treatment") +
    scale_shape_manual(values = colshapes, name = "Colony") + 
  facet_wrap(~method, scales="free")
p
ggsave(filename = "./outputs/phyloseq-results/figures/ordinations_bray.pdf", height = 8, width = 10, units = "in")
```
```{r ordinations}
#principal coordinates analysis, bray-curtis
ps.AXH.ord.PCoA.brayP <- ordinate(ps.AXH, method="PCoA", distance="bray")
#principal coordinates analysis, manhattan
ps.AXH.ord.PCoA.manhattan <- ordinate(ps.AXH, method="PCoA", distance="manhattan")
#principal coordinates analysis, unifrac
ps.AXH.ord.PCoA.unifrac <- ordinate(ps.AXH, method="PCoA", distance="unifrac")
#principal coordinates analysis,weighted unifrac
ps.AXH.ord.PCoA.wunifrac <- ordinate(ps.AXH, method="PCoA", distance="wunifrac")
#NMDS, bray-curtis
ps.AXH.ord.NMDS.brayP <- ordinate(ps.AXH, method="NMDS", distance="bray")
#NMDS, unifrac
ps.AXH.ord.NMDS.unifrac <- ordinate(ps.AXH, method="NMDS", distance="unifrac")
#NMDS, weighted unifrac
ps.AXH.ord.NMDS.wunifrac <- ordinate(ps.AXH, method="NMDS", distance="wunifrac")
```
```{r}
#principal coordinates analysis, bray-curtis
ps.control.ord.PCoA.brayP <- ordinate(ps.control.rel, method="PCoA", distance="bray")
```

It might be helpful to re-write the plot_ordination function to include centroid points and spiders
Also make a central ggplot theme and expected output file sizes
```{r pcoa_spider_treatment}
# Calculate Treatment centroids for plotting
ord <- cbind(sample_data(ps.AXH),  ps.AXH.ord.PCoA.brayP$vectors)
ord_trmt <- ord %>%
  group_by(Treatment) %>%
  dplyr::summarise(c1 = mean(`Axis.1`), c2 = mean(`Axis.2`)) %>%    
  full_join(ord)
# Plot with spiders
pcoa3 <- pcoa_spider(ps.AXH, ps.AXH.ord.PCoA.brayP, ord_trmt)
pcoa3title <- expression(paste("PCoA of overall ", italic("Pocillopora"), " spp. bacteria communities"))
#pcoa3 <- pcoa3 + ggtitle(pcoa3title, subtitle = "Plotted with treatment centroids")
pcoa3
ggsave(pcoa3, filename = "./outputs/phyloseq-results/figures/pcoa_treatment.pdf", width = 112, height = 80, units = "mm", device = "pdf")
ggsave(pcoa3, filename = "./manuscript_figures/Fig3A_PCoA_16S_Treatment.pdf", width = 112, height = 80, units = "mm", device = "pdf")
```

NMDS
I want to try to calculate treatment centroids for plotting as with the PCoA above
https://stackoverflow.com/questions/47516448/how-to-get-ordispider-like-clusters-in-ggplot-with-nmds

```{r nmds_ellipse_treatment}
nmds1 <- plot_ordination(ps.AXH.rel, ps.AXH.ord.NMDS.brayP, type = "sample", axes = c(1,2)) +
  stat_ellipse(aes(fill = Treatment), geom = "polygon", type = "norm", alpha = 0.15) + 
#sample points
    geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
    scale_color_manual(values = condcolors_AxH, name = "Treatment") +
    scale_fill_manual(values = condcolors_AxH, name = "Treatment") +
    scale_shape_manual(values = colshapes, name = "Colony") + 
  coord_fixed(1) +
  ggtitle("Bacteria Community NMDS") +
  theme(plot.title = element_text(size = 18),
        axis.title = element_text(size = 16),
        axis.text = element_text(size = 14),
        legend.title = element_text(size = 16),
        legend.text = element_text(size = 12))
print(nmds1)
ggsave(nmds1, filename = "./outputs/phyloseq-results/figures/nmds1_treatment.pdf",  width = 112, height = 80, units = "mm", device = "pdf")
```

### Constrained ordinations
```{r}
#http://deneflab.github.io/MicrobeMiseq/demos/mothur_2_phyloseq.html
# CAP ordinate
cap_ord <- ordinate(
    physeq = ps.AXH, 
    method = "CAP",
    distance = "bray",
    formula = ~ Colony
)
```
```{r}
cap_plot <- plot_ordination(ps.AXH, cap_ord, type = "sample", axes = c(1,2)) +
  stat_ellipse(aes(fill = Treatment), geom = "polygon", type = "norm", alpha = 0.6) + 
  geom_point(size = 6, aes(color = Treatment, shape = Colony)) +
  scale_color_manual(values = condcolors_AxH, name="Treatment") +
  scale_fill_manual(values = condfillcolors_AxH, name = "Treatment") +
  scale_shape_manual(values = colshapes, name="Colony") + 
  coord_fixed(1) +
  theme(plot.title = element_text(size = 18),
        axis.title = element_text(size = 16),
        axis.text = element_text(size = 14),
        legend.title = element_text(size = 16),
        legend.text = element_text(size = 12))
```
```{r cap_arrows}
# Now add the environmental variables as arrows
arrowmat <- vegan::scores(cap_ord, display = "bp")
# Add labels, make a data.frame
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)

# Define the arrow aesthetic mapping
arrow_map <- aes(xend = CAP1, 
    yend = CAP2, 
    x = 0, 
    y = 0, 
    shape = NULL, 
    color = NULL, 
    label = labels)

label_map <- aes(x = 1.3 * CAP1, 
    y = 1.3 * CAP2, 
    shape = NULL, 
    color = NULL, 
    label = labels)

arrowhead = arrow(length = unit(0.02, "npc"))

# Make a new graphic
cap_plot + 
  geom_segment(
    mapping = arrow_map, 
    size = .5, 
    data = arrowdf, 
    color = "gray", 
    arrow = arrowhead
  ) + 
  geom_text(
    mapping = label_map, 
    size = 4,  
    data = arrowdf, 
    show.legend = FALSE
  )
```
```{r}
# Do a permutational ANOVA on constrained axes used in ordination
anova(cap_ord)
```


### Beta diversity statistical testing
```{r bray-curtis_dist}
# Calculate Bray-Curtis distance matrix
psdist_bray <- phyloseq::distance(ps.AXH.rel, method = "bray")
# make a data frame from the sample_data
sampledf <- data.frame(sample_data(ps.AXH)) %>% select(-Environment)
```
```{r permdisp}
# PERMDISP tests
beta <- betadisper(psdist_bray, sampledf$Treatment)
permutest(beta)
beta <- betadisper(psdist_bray, sampledf$Reef)
permutest(beta)
beta <- betadisper(psdist_bray, sampledf$Colony)
permutest(beta)
```
```{r permanova}
# Adonis/PERMANOVA tests
adonis(psdist_bray ~ Treatment, data = sampledf, method = "bray")
# 
adonis(psdist_bray ~ Reef, data = sampledf, method = "bray")
# 
adonis(psdist_bray ~ Colony, data = sampledf, method = "bray")
```

### Distances to centroids boxplots
```{r}
dis <- vegdist(otu_table(ps.AXH), method ="bray")
# PERMDISP2 procedure for the analysis of multivariatehomogeneity of group dispersions (variances).
mod <- betadisper(dis, sampledf$Treatment, type = "centroid")
mod
anova(mod)
## Permutation test for F
permutest(mod, pairwise = TRUE)
(mod.HSD <- TukeyHSD(mod))
```
```{r}
## Draw a boxplot of the distances to centroid for each group
boxplot(mod)
boxplot(mod$distances ~ mod$group)
# extract the distances to centroids
mod_distances <- data.frame(mod$distances)
mod_distances <- mod_distances %>% rownames_to_column(., "SampleID")
mod_distances <- mod_distances %>% left_join(., sampledf, by = "SampleID", .keep = TRUE)
```
```{r}
beta_boxplot <- mod_distances %>% ggplot(aes(Treatment, mod.distances)) + 
  geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony") + 
  scale_x_discrete(labels = c("C", "H", "A", "A+H")) +
  #facet_grid(~Colony) +
  #ylim(c(0,100)) +
  xlab(NULL) +
  ylab("Beta diversity distance to centroid") +
  ggtitle("Beta diversity distance to centroid boxplot") +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 0),
        strip.background = element_blank(),
        strip.text = element_blank())
beta_boxplot
ggsave(beta_boxplot, filename = "./outputs/phyloseq-results/figures/distance_centroid_boxplot.pdf", width = 4, height = 3, units = "in")
```

```{r}
# centroids, segments, and group hulls in ordination plost
###https://chrischizinski.github.io/rstats/adonis/
```

### ANOSIM


## Taxa composition plots
### Phylum-level barplots
```{r phylum_barplot_frame, echo=FALSE, message=FALSE, error=FALSE}
glom_phylum <- tax_glom(ps.AXH.rel, taxrank = "Phylum")
ps.AXH_phylum <- psmelt(glom_phylum)
#ps.AXH_phylum <- filter(ps.AXH_phylum, Abundance > 0)
ps.AXH_phylum$Phylum <- as.character(ps.AXH_phylum$Phylum)
ps.AXH_phylum$Kingdom <- gsub("D_0__", "", ps.AXH_phylum$Kingdom)
ps.AXH_phylum$Phylum <- gsub("D_1__", "", ps.AXH_phylum$Phylum)
ps.AXH_phylum$Phylum[ps.AXH_phylum$Abundance < 0.01] <- "<1% abundance"
ps.AXH_phylum$Phylum <- factor(ps.AXH_phylum$Phylum, levels = c("<1% abundance", "Actinobacteria", "Planctomycetes", "Dependentiae", "Firmicutes", "Spirochaetes", "Nanoarchaeaeota", "Cyanobacteria", "Bacteroidetes", "Proteobacteria"))
ps.AXH_phylum$Sample<- gsub("6", "control.", ps.AXH_phylum$Sample)
ps.AXH_phylum$Sample<- gsub("5", "anti.", ps.AXH_phylum$Sample)
ps.AXH_phylum$Sample<- gsub("4", "heat.", ps.AXH_phylum$Sample)
ps.AXH_phylum$Sample<- gsub("\\.1", ".anti+heat.", ps.AXH_phylum$Sample)
ps.AXH_phylum$Sample <- factor(ps.AXH_phylum$Sample, levels = c("Hw1.control.a", "Hw1.control.b", "Hw1.control.c", "Hw2.control.a", "Hw2.control.b", "Hw2.control.c", "Wt1.control.a", "Wt1.control.b", "Wt1.control.c", "Wt2.control.a", "Wt2.control.b", "Wt2.control.c", "Hw1.heat.a", "Hw1.heat.b", "Hw1.heat.c", "Hw2.heat.a", "Hw2.heat.b", "Hw2.heat.c", "Wt1.heat.a", "Wt1.heat.b", "Wt1.heat.c", "Wt2.heat.a", "Wt2.heat.b", "Hw1.anti.a", "Hw1.anti.b", "Hw1.anti.c", "Hw2.anti.a", "Hw2.anti.b", "Hw2.anti.c", "Wt1.anti.a", "Wt1.anti.b", "Wt1.anti.c", "Wt2.anti.a", "Wt2.anti.b", "Wt2.anti.c", "Hw1.anti+heat.a", "Hw1.anti+heat.b", "Hw1.anti+heat.c", "Hw2.anti+heat.a", "Hw2.anti+heat.b", "Hw2.anti+heat.c", "Wt1.anti+heat.a", "Wt1.anti+heat.b", "Wt1.anti+heat.c", "Wt2.anti+heat.a", "Wt2.anti+heat.b", "Wt2.anti+heat.c"), ordered = TRUE)
```
```{r phylum_barplot_inspect}
#pdf(file = "./outputs/phyloseq-results/figures/barplot_phylum_.AXH.pdf", height = 8, width = 8.5)
### all samples
p_phylum1 <- ggplot(data = ps.AXH_phylum, aes(x = Sample, y = Abundance, fill = Phylum))
p_phylum1 <- p_phylum1 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = phyla_colors$color) +
  guides(fill=guide_legend(nrow = 3))
p_phylum1 <- p_phylum1 +
  ggtitle("Bacteria Phyla Relative Abundance") + 
  theme(axis.text.x = element_text(angle = 300, size = 8), 
        legend.text = element_text(size = 10), legend.position = "bottom")
print(p_phylum1)

### samples collapsed into colonies
p_phylum2 <- ggplot(data = ps.AXH_phylum, aes(x = Colony, y = Abundance, fill = Phylum))
p_phylum2 <- p_phylum2 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Treatment) +
  scale_fill_manual(values = phyla_colors$color) +
  guides(fill=guide_legend(nrow = 3))
p_phylum2 <- p_phylum2 +
  ggtitle("Bacteria Phyla Relative Abundance") + 
  theme(axis.text.x = element_text(angle = 0, size = 12), 
        legend.text = element_text(size = 10), legend.position = "bottom",
        strip.text = element_text(size = 12),
        strip.background = element_rect(fill = "light grey"))
print(p_phylum2)
g_phylum2 <- ggplot_gtable(ggplot_build(p_phylum2))
strip_both <- which(grepl('strip-', g_phylum2$layout$name))
k <- 1
for (i in strip_both) {
j <- which(grepl('rect', g_phylum2$grobs[[i]]$grobs[[1]]$childrenOrder))
g_phylum2$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- condfillcolors_AXH[k]
k <- k+1
}
gp2 <- grid.draw(g_phylum2)
print(gp2)

### samples collapsed into treatments
p_phylum3 <- ggplot(data = ps.AXH_phylum, aes(x = Treatment, y = Abundance, fill = Phylum))
p_phylum3 <- p_phylum3 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = phyla_colors$color) +
  guides(fill=guide_legend(nrow = 3)) +
  theme(axis.text.x = element_text(angle = 270),
        legend.text = element_text(size = 10), legend.position = "bottom")
p_phylum3
```
```{r phylum_piechart}
### Pie chart of overall relative abundance
slices <- c(sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "<1% abundance"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Actinobacteria"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Dependentiae"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Firmicutes"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Planctomycetes"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Nanoarcheaeota"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Spirochaetes"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Cyanobacteria"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Bacteroidetes"]),
            sum(ps.AXH_phylum$Abundance[ps.AXH_phylum$Phylum == "Proteobacteria"])
            )
pct <- round((slices/sum(slices)*100), digits = 2)
lbls <- c("<1% abundance", "Actinobacteria", "Dependentiae", "Firmicutes", "Planctomycetes", "Nanoarchaeaeota", "Spirochaetes", "Cyanobacteria", "Bacteroidetes", "Proteobacteria")
pie <- data.frame(slices, lbls, pct)
pie$lblspct <- paste(pie$lbls, pie$pct)
pie$lblspct <- paste(pie$lblspct, "%", sep="")
pie(pie$slices, labels = pie$lblspct, col = phyla_colors$color)

#Want to further sum low-abundance taxa and group into "Other" category to reduce occlusion on pie chart
#pie$lbls[pie$pct < 1] <- "Other"
#sum(pie$pct[pie$pct < 1])
```

### Class-level barplots
```{r}
glom_class <- tax_glom(ps.AXH.rel, taxrank = "Class")
ps.AXH_class <- psmelt(glom_class)
ps.AXH_class <- filter(ps.AXH_class, Abundance > 0)
ps.AXH_class$Class <- as.character(ps.AXH_class$Class)
ps.AXH_class$Kingdom <- gsub("D_0__", "", ps.AXH_class$Kingdom)
ps.AXH_class$Phylum <- gsub("D_1__", "", ps.AXH_class$Phylum)
ps.AXH_class$Class <- gsub("D_2__", "", ps.AXH_class$Class)
ps.AXH_class$Phylum[ps.AXH_class$Abundance < 0.01] <- "<1% abundance"
ps.AXH_class$Class[ps.AXH_class$Abundance < 0.01] <- "<1% abundance"
ps.AXH_class$Phylum <- factor(ps.AXH_class$Phylum, levels = c("<1% abundance", "Actinobacteria", "Planctomycetes", "Dependentiae", "Firmicutes", "Spirochaetes", "Nanoarchaeaeota", "Cyanobacteria", "Bacteroidetes", "Proteobacteria"))
ps.AXH_class$Class <- factor(ps.AXH_class$Class, levels = c("<1% abundance", "Actinobacteria", "Planctomycetacia", "Babeliae", "Bacilli", "Spirochaetia", "Woesearchaeia", "Oxyphotobacteria", "Bacteroidia", "Deltaproteobacteria", "Alphaproteobacteria", "Gammaproteobacteria"))
ps.AXH_class$Sample<- gsub("6", "control.", ps.AXH_class$Sample)
ps.AXH_class$Sample<- gsub("5", "anti.", ps.AXH_class$Sample)
ps.AXH_class$Sample<- gsub("4", "heat.", ps.AXH_class$Sample)
ps.AXH_class$Sample<- gsub("\\.1", ".anti+heat.", ps.AXH_class$Sample)
ps.AXH_class$Sample <- factor(ps.AXH_class$Sample, levels = c("Hw1.control.a", "Hw1.control.b", "Hw1.control.c", "Hw2.control.a", "Hw2.control.b", "Hw2.control.c", "Wt1.control.a", "Wt1.control.b", "Wt1.control.c", "Wt2.control.a", "Wt2.control.b", "Wt2.control.c", "Hw1.heat.a", "Hw1.heat.b", "Hw1.heat.c", "Hw2.heat.a", "Hw2.heat.b", "Hw2.heat.c", "Wt1.heat.a", "Wt1.heat.b", "Wt1.heat.c", "Wt2.heat.a", "Wt2.heat.b", "Hw1.anti.a", "Hw1.anti.b", "Hw1.anti.c", "Hw2.anti.a", "Hw2.anti.b", "Hw2.anti.c", "Wt1.anti.a", "Wt1.anti.b", "Wt1.anti.c", "Wt2.anti.a", "Wt2.anti.b", "Wt2.anti.c", "Hw1.anti+heat.a", "Hw1.anti+heat.b", "Hw1.anti+heat.c", "Hw2.anti+heat.a", "Hw2.anti+heat.b", "Hw2.anti+heat.c", "Wt1.anti+heat.a", "Wt1.anti+heat.b", "Wt1.anti+heat.c", "Wt2.anti+heat.a", "Wt2.anti+heat.b", "Wt2.anti+heat.c"), ordered = TRUE)
```
```{r}
#pdf(file = "./outputs/phyloseq-results/figures/barplot_class_.AXH.pdf", height = 8, width = 8.5)
### all samples
p_class1 <- ggplot(data = ps.AXH_class, aes(x = Sample, y = Abundance, fill = Class))
p_class1 <- p_class1 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = classes_colors$color) +
  guides(fill=guide_legend(nrow = 4))
p_class1 <- p_class1 +
  ggtitle("Bacteria Class Relative Abundance") +
  theme(axis.text.x = element_text(angle = 300, size = 8), 
        legend.text = element_text(size = 10), legend.position = "bottom")
print(p_class1)

### samples collapsed into colonies
p_class2 <- ggplot(data = ps.AXH_class, aes(x = Colony, y = Abundance, fill = Class))
p_class2 <- p_class2 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Treatment) +
  scale_fill_manual(values = classes_colors$color) +
  guides(fill=guide_legend(nrow = 3))
p_class2 <- p_class2 +
  ggtitle("Bacteria Class Relative Abundance") + 
  theme(axis.text.x = element_text(angle = 0, size = 12), 
        legend.text = element_text(size = 10), legend.position = "bottom",
        strip.text = element_text(size = 12),
        strip.background = element_rect(fill = "light grey"))
print(p_class2)
g_class2 <- ggplot_gtable(ggplot_build(p_class2))
strip_both <- which(grepl('strip-', g_class2$layout$name))
k <- 1
for (i in strip_both) {
j <- which(grepl('rect', g_class2$grobs[[i]]$grobs[[1]]$childrenOrder))
g_class2$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- condfillcolors_AXH[k]
k <- k+1
}
gc2 <- grid.draw(g_class2)
print(gc2)

### samples collapsed into treatments
p_class3 <- ggplot(data = ps.AXH_class, aes(x = Treatment, y = Abundance, fill = Class))
p_class3 <- p_class3 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = classes_colors$color) +
  guides(fill=guide_legend(nrow = 15)) +
  theme(axis.text.x = element_text(angle = 270))
print(p_class3)
```
```{r}
### Pie chart of overall relative abundance
slices <- c(sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "<1% abundance"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Verrucomicrobiae"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Campylobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Actinobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Babeliae"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Anaerolineae"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Woesearchaeia"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Spirochaetia"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Oxyphotobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Bacteroidia"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Deltaproteobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Alphaproteobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "Gammaproteobacteria"])
            )
            
pct <- round((slices/sum(slices)*100), digits = 2)
lbls <- c("<1% abundance", "Verrucomicrobiae", "Campylobacteria", "Actinobacteria", "Babeliae", "Anaerolineae", "Woesearchaeia", "Spirochaetia", "Oxyphotobacteria", "Bacteroidia", "Deltaproteobacteria", "Alphaproteobacteria", "Gammaproteobacteria")
pie <- data.frame(slices, lbls, pct)
pie$lblspct <- paste(pie$lbls, pie$pct)
pie$lblspct <- paste(pie$lblspct, "%", sep="")
pie(pie$slices, labels = pie$lblspct, col = classes_colors$color)

#Want to further sum low-abundance taxa and group into "Other" category to reduce occlusion on pie chart
#pie$lbls[pie$pct < 1] <- "Other"
#sum(pie$pct[pie$pct < 1])
```

### Order-level barplots
```{r}
glom_order <- tax_glom(ps.AXH.rel, taxrank = "Order")
ps.AXH_order <- psmelt(glom_order)
ps.AXH_order <- filter(ps.AXH_order, Abundance > 0)
ps.AXH_order$Order <- as.character(ps.AXH_order$Order)
ps.AXH_order$Kingdom <- gsub("D_0__", "", ps.AXH_order$Kingdom)
ps.AXH_order$Phylum <- gsub("D_1__", "", ps.AXH_order$Phylum)
ps.AXH_order$Class <- gsub("D_2__", "", ps.AXH_order$Class)
ps.AXH_order$Order <- gsub("D_3__", "", ps.AXH_order$Order)
ps.AXH_order$Phylum[ps.AXH_order$Abundance < 0.01] <- "<1% abundance"
ps.AXH_order$Class[ps.AXH_order$Abundance < 0.01] <- "<1% abundance"
ps.AXH_order$Order[ps.AXH_order$Abundance < 0.05] <- "<1% abundance"
ps.AXH_order$Phylum <- factor(ps.AXH_order$Phylum, levels = c("<1% abundance", "Actinobacteria", "Planctomycetes", "Dependentiae", "Firmicutes", "Spirochaetes", "Nanoarchaeaeota", "Cyanobacteria", "Bacteroidetes", "Proteobacteria"))
ps.AXH_order$Class <- factor(ps.AXH_order$Class, levels = c("<1% abundance", "Actinobacteria", "Planctomycetacia", "Babeliae", "Bacilli", "Spirochaetia", "Woesearchaeia", "Oxyphotobacteria", "Bacteroidia", "Deltaproteobacteria", "Alphaproteobacteria", "Gammaproteobacteria"))
ps.AXH_order$Order <- gsub("Euryarchaeota archaeon SCGC AAA286-E23", "Archaeon SCGC", ps.AXH_order$Order)
ps.AXH_order$Order <- factor(ps.AXH_order$Order, levels = c("<1% abundance", "Spirochaetales", "Archaeon SCGC", "Nostocales", "Cytophagales", "Myxococcales", "Rhodobacterales", "Rhodovibrionales", "Francisellales", "Betaproteobacteriales", "Cellvibrionales", "Xanthomonadales", "Vibrionales", "Alteromonadales", "Oceanospirillales"))
ps.AXH_order$Sample<- gsub("6", "control.", ps.AXH_order$Sample)
ps.AXH_order$Sample<- gsub("5", "anti.", ps.AXH_order$Sample)
ps.AXH_order$Sample<- gsub("4", "heat.", ps.AXH_order$Sample)
ps.AXH_order$Sample<- gsub("\\.1", ".anti+heat.", ps.AXH_order$Sample)
ps.AXH_order$Sample <- factor(ps.AXH_order$Sample, levels = c("Hw1.control.a", "Hw1.control.b", "Hw1.control.c", "Hw2.control.a", "Hw2.control.b", "Hw2.control.c", "Wt1.control.a", "Wt1.control.b", "Wt1.control.c", "Wt2.control.a", "Wt2.control.b", "Wt2.control.c", "Hw1.heat.a", "Hw1.heat.b", "Hw1.heat.c", "Hw2.heat.a", "Hw2.heat.b", "Hw2.heat.c", "Wt1.heat.a", "Wt1.heat.b", "Wt1.heat.c", "Wt2.heat.a", "Wt2.heat.b", "Hw1.anti.a", "Hw1.anti.b", "Hw1.anti.c", "Hw2.anti.a", "Hw2.anti.b", "Hw2.anti.c", "Wt1.anti.a", "Wt1.anti.b", "Wt1.anti.c", "Wt2.anti.a", "Wt2.anti.b", "Wt2.anti.c", "Hw1.anti+heat.a", "Hw1.anti+heat.b", "Hw1.anti+heat.c", "Hw2.anti+heat.a", "Hw2.anti+heat.b", "Hw2.anti+heat.c", "Wt1.anti+heat.a", "Wt1.anti+heat.b", "Wt1.anti+heat.c", "Wt2.anti+heat.a", "Wt2.anti+heat.b", "Wt2.anti+heat.c"), ordered = TRUE)
```

## Corrections needed
It might be worthwhile to edit the functions to include the aesthetics (ie. facet label colors) and streamline the code
```{r}
pdf(file = "./outputs/phyloseq-results/figures/barplot_order_AXH.pdf", height = 8.5, width = 11)
length(unique(ps.AXH_order$Order))

### all samples
p_order1 <- ggplot(data = ps.AXH_order, aes(x = Sample, y = Abundance, fill = Order))
p_order1 <- p_order1 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = orders_colors$color) +
  guides(fill=guide_legend(nrow = 4))
p_order1 <- p_order1 +
  ggtitle("Bacteria Order Relative Abundance") +
  theme(axis.text.x = element_text(angle = 300, size = 8), 
        legend.text = element_text(size = 10), legend.position = "bottom")
print(p_order1)

### samples collapsed into colonies
p_order2 <- ggplot(data = ps.AXH_order, aes(x = Colony, y = Abundance, fill = Order))
p_order2 <- p_order2 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Treatment) +
  scale_fill_manual(values = orders_colors$color) +
  guides(fill=guide_legend(nrow = 3))
p_order2 <- p_order2 +
  ggtitle("Bacteria Order Relative Abundance") + 
  theme(axis.text.x = element_text(angle = 0, size = 12), 
        legend.text = element_text(size = 10), legend.position = "bottom",
        strip.text = element_text(size = 12),
        strip.background = element_rect(fill = "light grey"))
print(p_order2)
g_order2 <- ggplot_gtable(ggplot_build(p_order2))
strip_both <- which(grepl('strip-', g_order2$layout$name))
k <- 1
for (i in strip_both) {
j <- which(grepl('rect', g_order2$grobs[[i]]$grobs[[1]]$childrenOrder))
g_order2$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- condfillcolors_AXH[k]
k <- k+1
}
gc2 <- grid.draw(g_order2)
print(gc2)

### samples collapsed into treatments
p_order3 <- ggplot(data = ps.AXH_order, aes(x = Treatment, y = Abundance, fill = Order))
p_order3 <- p_order3 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = orders_colors$color) +
  guides(fill=guide_legend(nrow = 15)) +
  theme(axis.text.x = element_text(angle = 270))
print(p_order3)
```

### Family-level barplots
```{r}
glom_family <- tax_glom(ps.AXH.rel, taxrank = "Family")
ps.AXH_family <- psmelt(glom_family)
ps.AXH_family$Family <- as.character(ps.AXH_family$Family)
ps.AXH_family$Kingdom <- gsub("D_0__", "", ps.AXH_family$Kingdom)
ps.AXH_family$Phylum <- gsub("D_1__", "", ps.AXH_family$Phylum)
ps.AXH_family$Class <- gsub("D_2__", "", ps.AXH_family$Class)
ps.AXH_family$Order <- gsub("D_3__", "", ps.AXH_family$Order)
ps.AXH_family$Family <- gsub("D_4__", "", ps.AXH_family$Family)
ps.AXH_family <- filter(ps.AXH_family, Abundance > 0.00)
ps.AXH_family$Family[ps.AXH_family$Abundance < 0.03] <- "<3% abundance"
ps.AXH_family$Phylum <- factor(ps.AXH_family$Phylum, levels = c("<1% abundance", "Actinobacteria", "Planctomycetes", "Dependentiae", "Firmicutes", "Spirochaetes", "Nanoarchaeaeota", "Cyanobacteria", "Bacteroidetes", "Proteobacteria"))
ps.AXH_family$Class <- factor(ps.AXH_family$Class, levels = c("<1% abundance", "Actinobacteria", "Planctomycetacia", "Babeliae", "Bacilli", "Spirochaetia", "Woesearchaeia", "Oxyphotobacteria", "Bacteroidia", "Deltaproteobacteria", "Alphaproteobacteria", "Gammaproteobacteria"))
ps.AXH_family$Order <- gsub("Euryarchaeota archaeon SCGC AAA286-E23", "Archaeon SCGC", ps.AXH_family$Order)
ps.AXH_family$Order <- factor(ps.AXH_family$Order, levels = c("<1% abundance", "Spirochaetales", "Archaeon SCGC", "Nostocales", "Cytophagales", "Myxococcales", "Rhodobacterales", "Rhodovibrionales", "Francisellales", "Betaproteobacteriales", "Cellvibrionales", "Xanthomonadales", "Vibrionales", "Alteromonadales", "Oceanospirillales"))
ps.AXH_family$Family <- gsub("uncultured", "unc.", ps.AXH_family$Family)
ps.AXH_family$Family <- gsub("Euryarchaeota archaeon SCGC AAA286-E23", "Archaeon SCGC", ps.AXH_family$Family)

#ps.AXH_family$Family <- factor(ps.AXH_family$Family, levels = c("<1% abundance", "Corynebacteriaceae", "Flavobacteriaceae", "Saprospiraceae", "Nodosilineaceae", "unc. bacterium", "Staphylococcaceae", "Streptococcaceae", "Pirellulaceae", "Spirochaetaceae", "Archaeon SCGC", "Nostocaceae", "Xenococcaceae", "Amoebophilaceae", "Cyclobacteriaceae", "P3OB-42", "Rhodobacteraceae", "Kiloniellaceae", "Micavibrionaceae", "Midichloriaceae", "Stappiaceae", "Hyphomonadaceae", "Parvibaculaceae", "Methylophilaceae", "Burkholderiaceae", "Cellvibrionaceae", "Spongiibacteraceae", "Coxiellaceae", "Francisellaceae", "Pasteurellaceae", "Salinisphaeraceae", "Xanthomonadaceae", "unc. Vibrio sp.", "Vibrionaceae", "Alteromonadaceae", "Colwelliaceae", "Oleiphilaceae", "Saccharospirillaceae", "Nitrincolaceae", "Endozoicomonadaceae"))
ps.AXH_family$Family <- factor(ps.AXH_family$Family, levels = c("<3% abundance", "Saprospiraceae", "Spirochaetaceae", "Archaeon SCGC", "Nostocaceae", "Xenococcaceae", "Amoebophilaceae", "Cyclobacteriaceae", "P3OB-42", "Rhodobacteraceae", "Kiloniellaceae", "Hyphomonadaceae", "Methylophilaceae", "Burkholderiaceae", "Cellvibrionaceae", "Spongiibacteraceae", "Francisellaceae", "Salinisphaeraceae", "Xanthomonadaceae", "Vibrionaceae", "Alteromonadaceae", "Oleiphilaceae", "Saccharospirillaceae", "Nitrincolaceae", "Endozoicomonadaceae"))

ps.AXH_family$Sample<- gsub("6", "control.", ps.AXH_family$Sample)
ps.AXH_family$Sample<- gsub("5", "anti.", ps.AXH_family$Sample)
ps.AXH_family$Sample<- gsub("4", "heat.", ps.AXH_family$Sample)
ps.AXH_family$Sample<- gsub("\\.1", ".anti+heat.", ps.AXH_family$Sample)
ps.AXH_family$Sample <- factor(ps.AXH_family$Sample, levels = c("Hw1.control.a", "Hw1.control.b", "Hw1.control.c", "Hw2.control.a", "Hw2.control.b", "Hw2.control.c", "Wt1.control.a", "Wt1.control.b", "Wt1.control.c", "Wt2.control.a", "Wt2.control.b", "Wt2.control.c", "Hw1.heat.a", "Hw1.heat.b", "Hw1.heat.c", "Hw2.heat.a", "Hw2.heat.b", "Hw2.heat.c", "Wt1.heat.a", "Wt1.heat.b", "Wt1.heat.c", "Wt2.heat.a", "Wt2.heat.b", "Hw1.anti.a", "Hw1.anti.b", "Hw1.anti.c", "Hw2.anti.a", "Hw2.anti.b", "Hw2.anti.c", "Wt1.anti.a", "Wt1.anti.b", "Wt1.anti.c", "Wt2.anti.a", "Wt2.anti.b", "Wt2.anti.c", "Hw1.anti+heat.a", "Hw1.anti+heat.b", "Hw1.anti+heat.c", "Hw2.anti+heat.a", "Hw2.anti+heat.b", "Hw2.anti+heat.c", "Wt1.anti+heat.a", "Wt1.anti+heat.b", "Wt1.anti+heat.c", "Wt2.anti+heat.a", "Wt2.anti+heat.b", "Wt2.anti+heat.c"), ordered = TRUE)
```
```{r family_barplot}
### all samples
p_family1 <- ggplot(data = ps.AXH_family, aes(x = Sample, y = Abundance, fill = Family))
p_family1 <- p_family1 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = families_colors$color) +
  guides(fill=guide_legend(nrow = 20)) +
  theme(axis.text.x = element_text(angle = 270), legend.text = element_text(size = 4))
p_family1

### samples collapsed into colonies
pdf(file = "./manuscript_figures/Fig3C_RelAbundBarplot.pdf", width = 6.5, height = 4.1)#  width = 62, height = 80, units = "mm", device = "pdf")
p_family2 <- ggplot(data = ps.AXH_family, aes(x = Colony, y = Abundance, fill = Family))
p_family2 <- p_family2 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Treatment) +
### create custom color palette for 39 bacteria families
  scale_fill_manual(values = families_colors$color, name = "Bacteria Family") +
  guides(fill = guide_legend(nrow = 6)) +
  #ggtitle("Bacteria Family Relative Abundance") +
  ylab("Relative Abundance") +
  theme(legend.position = "bottom",
        legend.spacing.x = unit(1, "mm"),
        legend.key.width = unit(8, "mm"),
        legend.spacing.y = unit(1, "mm"),
        axis.title = element_text(margin = margin(c(3,1,3,1), unit = "mm")))
p_family2
g_family2 <- ggplot_gtable(ggplot_build(p_family2))
strip_both <- which(grepl('strip-', g_family2$layout$name))
k <- 1
for (i in strip_both) {
j <- which(grepl('rect', g_family2$grobs[[i]]$grobs[[1]]$childrenOrder))
g_family2$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- condfillcolors_AXH[k]
k <- k+1
}
gf2 <- grid.draw(g_family2)
print(gf2)
dev.off()

### samples collapsed into treatments
p_family3 <- ggplot(data = ps.AXH_family, aes(x = Treatment, y = Abundance, fill = Family))
p_family3 <- p_family3 +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = families_colors$color) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270), legend.text = element_text(size = 4))
p_family3
```

### Genus-level barplots
```{r}
glom_genus <- tax_glom(ps.AXH.rel, taxrank = "Genus")
ps.AXH_genus <- psmelt(glom_genus)
ps.AXH_genus$Family <- as.character(ps.AXH_genus$Family)
ps.AXH_genus$Kingdom <- gsub("D_0__", "", ps.AXH_genus$Kingdom)
ps.AXH_genus$Phylum <- gsub("D_1__", "", ps.AXH_genus$Phylum)
ps.AXH_genus$Class <- gsub("D_2__", "", ps.AXH_genus$Class)
ps.AXH_genus$Order <- gsub("D_3__", "", ps.AXH_genus$Order)
ps.AXH_genus$Family <- gsub("D_4__", "", ps.AXH_genus$Family)
ps.AXH_genus$Genus <- gsub("D_5__", "", ps.AXH_genus$Genus)
ps.AXH_genus <- filter(ps.AXH_genus, Abundance > 0.00)
ps.AXH_genus$Family[ps.AXH_genus$Abundance < 0.01] <- "<1% abundance"
ps.AXH_genus$Family <- gsub("uncultured", "unc.", ps.AXH_genus$Family)
ps.AXH_genus$Family <- gsub("Euryarchaeota archaeon SCGC AAA286-E23", "Archaeon SCGC", ps.AXH_genus$Family)
ps.AXH_genus$Sample<- gsub("6", "control.", ps.AXH_genus$Sample)
ps.AXH_genus$Sample<- gsub("5", "anti.", ps.AXH_genus$Sample)
ps.AXH_genus$Sample<- gsub("4", "heat.", ps.AXH_genus$Sample)
ps.AXH_genus$Sample<- gsub("\\.1", ".anti+heat.", ps.AXH_genus$Sample)
ps.AXH_genus$Sample <- factor(ps.AXH_genus$Sample, levels = c("Hw1.control.a", "Hw1.control.b", "Hw1.control.c", "Hw2.control.a", "Hw2.control.b", "Hw2.control.c", "Wt1.control.a", "Wt1.control.b", "Wt1.control.c", "Wt2.control.a", "Wt2.control.b", "Wt2.control.c", "Hw1.heat.a", "Hw1.heat.b", "Hw1.heat.c", "Hw2.heat.a", "Hw2.heat.b", "Hw2.heat.c", "Wt1.heat.a", "Wt1.heat.b", "Wt1.heat.c", "Wt2.heat.a", "Wt2.heat.b", "Hw1.anti.a", "Hw1.anti.b", "Hw1.anti.c", "Hw2.anti.a", "Hw2.anti.b", "Hw2.anti.c", "Wt1.anti.a", "Wt1.anti.b", "Wt1.anti.c", "Wt2.anti.a", "Wt2.anti.b", "Wt2.anti.c", "Hw1.anti+heat.a", "Hw1.anti+heat.b", "Hw1.anti+heat.c", "Hw2.anti+heat.a", "Hw2.anti+heat.b", "Hw2.anti+heat.c", "Wt1.anti+heat.a", "Wt1.anti+heat.b", "Wt1.anti+heat.c", "Wt2.anti+heat.a", "Wt2.anti+heat.b", "Wt2.anti+heat.c"), ordered = TRUE)

colCount <- length(unique(ps.AXH_genus$Genus))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
```
```{r}
### all samples
pdf(file = "./outputs/phyloseq-results/figures/barplot_genus_.AXH.pdf", height = 8.5, width = 11)
p_genus <- ggplot(data = ps.AXH_genus, aes(x = Sample, y = Abundance, fill = Genus))
p_genus <- p_genus +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270), legend.text = element_text(size = 4))
p_genus
###samples collapsed into colonies
p_genus <- ggplot(data = ps.AXH_genus, aes(x = Treatment, y = Abundance, fill = Genus))
p_genus <- p_genus +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Colony) +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 300), legend.text = element_text(size = 10))
p_genus
### samples collapsed into treatments
p_genus <- ggplot(data = ps.AXH_genus, aes(x = Treatment, y = Abundance, fill = Genus))
p_genus <- p_genus +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270), legend.text = element_text(size = 4))
p_genus
```

### Species-level barplots
```{r}
glom_species <- tax_glom(ps.AXH, taxrank = "Species")
ps.AXH_species <- psmelt(glom_species)
ps.AXH_species$Species <- as.character(ps.AXH_species$Species)
ps.AXH_species <- filter(ps.AXH_species, Abundance > 0.001)
ps.AXH_species$Species[ps.AXH_species$Abundance < 0.01] <- "<1% abundance"
```
```{r}
colCount <- length(unique(psctrl.LPS_species$Species))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
```
```{r}
pdf(file = "./phyloseq_results/figures/barplot_species_ctrl.LPS.pdf", height = 8.5, width = 11)
### all samples
p_species <- ggplot(data = psctrl.LPS_species, aes(x = Sample, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
### samples collapsed into colonies
p_species <- ggplot(data = psctrl.LPS_species, aes(x = Treatment, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Colony) +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
### samples collapsed into treatments
p_species <- ggplot(data = psctrl.LPS_species, aes(x = Treatment, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
```

Â¬
```{r}
taxdf <- as.data.frame(taxonomy)
length(unique(taxdf$Family))
unique(taxdf$Family)
```

### Heatmaps
```{r}
plot_heatmap(ps.AXH.rel, taxa.label = "Class", sample.order = c("Hw1.6a", "Hw1.6b", "Hw1.6c", "Hw2.6a", "Hw2.6b", "Hw2.6c", "Wt1.6a", "Wt1.6b", "Wt1.6c", "Wt2.6a", "Wt2.6b", "Wt2.6c", "Hw1.4a", "Hw1.4b", "Hw1.4c", "Hw2.4a", "Hw2.4b", "Hw2.4c", "Wt1.4a", "Wt1.4b", "Wt1.4c", "Wt2.4a", "Wt2.4b", "Hw1.5a", "Hw1.5b", "Hw1.5c", "Hw2.5a", "Hw2.5b", "Hw2.5c", "Wt1.5a", "Wt1.5b", "Wt1.5c", "Wt2.5a", "Wt2.5b", "Wt2.5c", "Hw1.1a", "Hw1.1b", "Hw1.1c", "Hw2.1a", "Hw2.1b", "Hw2.1c", "Wt1.1a", "Wt1.1b", "Wt1.1c", "Wt2.1a", "Wt2.1b", "Wt2.1c"))
```
## Alpha diversity plots and statistics
```{r estimate_richness}
ps.AXH_richness <- estimate_richness(ps.AXH) %>% rownames_to_column("ID")
alphameasures <- c("Observed", "Chao1", "ACE", "Shannon", "Simpson", "InvSimpson", "Fisher")
```
### Overall alpha diversity boxplots
```{r}
plot_richness(ps.AXH, x="Treatment", measures = alphameasures) +
  geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony")
```
```{r}
#pdf(file = "./outputs/phyloseq-results/figures/richness_ctrl.LPS.pdf", height = 2.95, width = 4.5)
plot_richness(ps.AXH, x="Treatment", measures=c("Chao1", "Simpson", "Shannon")) +
 geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony") + 
  ylim(c(0,100)) +
  xlab(NULL) +
  ylab(NULL) +
  ggtitle("Bacteria Community Diversity") +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 0),
        strip.background = element_blank(),
        strip.text = element_blank())
```
```{r chao1_boxplot}
chao1 <- plot_richness(ps.AXH, x="Treatment", measures=c("Chao1")) +
  geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony") + 
  scale_x_discrete(labels = c("C", "H", "A", "A.H")) +
  ylim(c(0,100)) +
  xlab(NULL) +
  ylab("Chao1 ASV richness") +
  #ggtitle("Alpha diversity boxplot: Chao1 estimator") +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 0),
        strip.background = element_blank(),
        strip.text = element_blank())
chao1
ggsave(chao1, filename = "./outputs/phyloseq-results/figures/chao1boxplot.pdf",  width = 50, height = 75, units = "mm", device = "pdf")
ggsave(chao1, filename = "./manuscript_figures/Fig3B_chao1boxplot.pdf",  width = 50, height = 75, units = "mm", device = "pdf")
```
```{r shannon_boxplot}
shannon <- plot_richness(ps.AXH, x="Treatment", measures=c("Shannon")) +
 geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony") + 
  scale_x_discrete(labels = c("C", "H", "A", "A+H")) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.01), limits = c(0,4)) +
  xlab(NULL) +
  ylab("Shannnon diversity index") +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 0),
        strip.background = element_blank(),
        strip.text = element_blank())
shannon
```
```{r simpson_boxplot}
simpson <- plot_richness(ps.AXH, x="Treatment", measures=c("Simpson")) +
 geom_boxplot(aes(fill = Treatment)) +
  scale_fill_manual(values = condfillcolors_AXH) +
  new_scale_fill() + 
  geom_point(aes(color = Treatment, shape = Colony), size = 3, alpha = 0.8) +
  geom_point(size = 3, aes(fill = Treatment, shape = Colony), color = "black", stroke = 0.5, show.legend = FALSE) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = condcolors_AxH) +
  scale_shape_manual(values = colshapes, name="Colony") + 
  scale_x_discrete(labels = c("C", "H", "A", "A+H")) +
  ylim(c(0,1)) +
  xlab(NULL) +
  ylab("Simpson eveness index") +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 0),
        strip.background = element_blank(),
        strip.text = element_blank())
simpson
```
```{r}
richness_boxplots <- (chao1 | shannon | simpson)
richness_boxplots 
```

### Genotype-specific alpha diversity boxplots
```{r, message=FALSE, warning=FALSE}
alpha_geno <- function(ps = ps.AXH, alphameasure) {
alphabox <- plot_richness(ps, x = "Colony", measures = alphameasure) +
  geom_boxplot(aes(fill = Colony)) +
  geom_point(size = 3, aes(shape = Colony)) +
  scale_fill_manual(values = colcolors) +
  scale_shape_manual(values = colshapes)
print(alphabox)
}
for (alphameasure in alphameasures) {
  alpha_geno(ps.AXH, alphameasure)
}
```
Incorporate treatment separation
```{r, message=FALSE, warning=FALSE}
alpha_geno <- function(ps = ps.AXH, alphameasure) {
alphabox <- plot_richness(ps, x = "Colony", color = "Treatment", measures = alphameasure) +
  geom_boxplot(aes(fill = Colony)) +
  geom_point(size = 3, aes(shape = Colony)) +
  scale_color_manual(values = condcolors_AxH) +
  scale_fill_manual(values = colcolors) +
  scale_shape_manual(values = colshapes)
print(alphabox)
}
for (alphameasure in alphameasures) {
  alpha_geno(ps.AXH, alphameasure)
}
```

## Corrections needed 

### Statistical testing of alpha diversity differences
```{r}
# make a data frame from the sample_data
sampledf <- data.frame(sample_data(ps.AXH))
```
```{r}
ps_richness_meta <- left_join(ps.AXH_richness, sampledf, by = "SampleID")
```
Kruskal-Wallis testing for treatment effect
```{r}
kruskal.test(data=ps_richness_meta, Observed ~ Treatment)
kruskal.test(data=ps_richness_meta, Chao1 ~ Treatment)
kruskal.test(data=ps_richness_meta, ACE ~ Treatment)
kruskal.test(data=ps_richness_meta, Shannon ~ Treatment)
kruskal.test(data=ps_richness_meta, Simpson ~ Treatment)
kruskal.test(data=ps_richness_meta, InvSimpson ~ Treatment)
kruskal.test(data=ps_richness_meta, Fisher ~ Treatment)
```
Kruskal-Wallis testing for reef effect
```{r}
kruskal.test(data=ps_richness_meta, Observed ~ Reef)
kruskal.test(data=ps_richness_meta, Chao1 ~ Reef)
kruskal.test(data=ps_richness_meta, ACE ~ Reef)
kruskal.test(data=ps_richness_meta, Shannon ~ Reef)
kruskal.test(data=ps_richness_meta, Simpson ~ Reef)
kruskal.test(data=ps_richness_meta, InvSimpson ~ Reef)
kruskal.test(data=ps_richness_meta, Fisher ~ Reef)
```

Kruskal-Wallis testing for colony effect
```{r}
kruskal.test(data=ps_richness_meta, Observed ~ Colony)
kruskal.test(data=ps_richness_meta, Chao1 ~ Colony)
kruskal.test(data=ps_richness_meta, ACE ~ Colony)
kruskal.test(data=ps_richness_meta, Shannon ~ Colony)
kruskal.test(data=ps_richness_meta, Simpson ~ Colony)
kruskal.test(data=ps_richness_meta, InvSimpson ~ Colony)
kruskal.test(data=ps_richness_meta, Fisher ~ Colony)
```


## Core microbiome
Coral core microbiome cutoffs: 50%, 80%, 90%, 100%

```{r}
ps.AXH.present <- prune_taxa(taxa_sums(ps.AXH) > 0, ps.AXH.rel)
ps.AXH.present
summarize_phyloseq(ps.AXH.present)
```
```{r}
datatable(otu_table(ps.AXH.present))
```
```{r}
core.taxa <- core_members(ps.AXH.present, detection = 0.001, prevalence = 80/100)
core.taxa
# Extract the taxonomy table
taxonomy.present <- as.data.frame(tax_table(ps.AXH.present))
# Subset this taxonomy table to include only core OTUs  
core_taxa_id <- subset(taxonomy.present, rownames(taxonomy.present) %in% core.taxa)
DT::datatable(core_taxa_id)
```

## Differential abundance testing
### ALDeX2
#### Heat
```{r}
# Treatment sample subsets of phyloseq objects
nsamples(ps_assigned)
samples <- sample_names(ps_assigned)
samples.HxC <- grep("[HW][wt][1-2].[46][abc]", samples, value = TRUE)
ps.HxC <- prune_samples(samples.HxC, ps_assigned)
ps.HxC
summarize_phyloseq(ps.AxC)
# Read out ASV tables from phyloseq objects
aldex.asv <- as.data.frame(t(otu_table(ps.HxC)))
```
```{r}
conds <- as.character(sample_data(ps.HxC)$Treatment)
conds <- factor(conds, levels = c("control", "Heat"), ordered = TRUE)
```
```{r}
x.ps <- aldex(reads = aldex.asv, conditions = conds, mc.samples = 128, test = "t", effect = TRUE, include.sample.summary = FALSE, denom = "all", verbose = TRUE)
```
```{r}
write.csv(x.ps, file = "./outputs/phyloseq-results/aldex2/heat-aldex2.csv")
```
```{r}
#pdf("./outputs/aldex2/anti-aldex2.pdf", height = 4, width = 8)
par(mfrow=c(1,2))
aldex.plot(x.ps, type="MW", test="welch", xlab="Dispersion", ylab="Difference")
aldex.plot(x.ps, type = "MA", test = "welch", xlab = "Log-ratio abundance", ylab = "Difference")

#dev.off()
```
```{r}
aldexmw <- aldex.plot_gg(x.ps, type = "MW", test = "welch")
aldexmw_heat <- aldexmw + 
  ggtitle("Heat vs. Control Treatment")
aldexma <- aldex.plot_gg(x.ps, type = "MA", test = "welch")
#aldex2 + ggtitle(subtitle = "Antibiotics + Heat vs. Control Treatment", "Bacteria ASV ALDeX2 Differential Abundance MA Plot") 
 
aldexplots <- (aldexmw | aldexma) 
aldexplots
ggsave(aldexplots, filename = "./outputs/phyloseq-results/aldex2/anti-asv-aldex2.pdf", width = 8, height = 4, units = "in", device = "pdf")
```
```{r}
cutoff <- 0.1
called <- x.ps[x.ps$we.eBH <= cutoff, ]
called.ASVs <- taxonomy[rownames(called), ]
called.ASVs1 <- cbind(called, called.ASVs)
```
```{r}
rare_cutoff <- 0
rare <- x.ps[x.ps$rab.all < rare_cutoff, ]
rare.ASVs <- taxonomy[rownames(rare), ]
rare.ASVs1 <- cbind(called, rare.ASVs)
```
#### Antibiotics
```{r}
# Treatment sample subsets of phyloseq objects
nsamples(ps_assigned)
samples <- sample_names(ps_assigned)
samples.AxC <- grep("[HW][wt][1-2].[16][abc]", samples, value = TRUE)
ps.AxC <- prune_samples(samples.AxC, ps_assigned)
ps.AxC
summarize_phyloseq(ps.AxC)
# Read out ASV tables from phyloseq objects
aldex.asv <- as.data.frame(t(otu_table(ps.AxC)))
```
```{r}
conds <- as.character(sample_data(ps.AxC)$Treatment)
conds <- factor(conds, levels = c("control", "Antibiotics.Heat"), ordered = TRUE)
```
```{r}
x.ps <- aldex(reads = aldex.asv, conditions = conds, mc.samples = 128, test = "t", effect = TRUE, include.sample.summary = FALSE, denom = "all", verbose = TRUE)
```
```{r}
write.csv(x.ps, file = "./outputs/phyloseq-results/aldex2/anti-aldex2.csv")
```
```{r}
#pdf("./outputs/aldex2/anti-aldex2.pdf", height = 4, width = 8)
par(mfrow=c(1,2))
aldex.plot(x.ps, type="MW", test="welch", xlab="Dispersion", ylab="Difference")
aldex.plot(x.ps, type = "MA", test = "welch", xlab = "Log-ratio abundance", ylab = "Difference")

#dev.off()
```
```{r}
aldexmw <- aldex.plot_gg(x.ps, type = "MW", test = "welch")
aldexmw_anti <- aldexmw + 
  ggtitle("Antibiotics vs. Control Treatment")
aldexma <- aldex.plot_gg(x.ps, type = "MA", test = "welch")
#aldex2 + ggtitle(subtitle = "Antibiotics + Heat vs. Control Treatment", "Bacteria ASV ALDeX2 Differential Abundance MA Plot") 
 
aldexplots <- (aldexmw | aldexma) 
aldexplots
ggsave(aldexplots, filename = "./outputs/phyloseq-results/figures/anti-asv-aldex2.pdf", width = 8, height = 4, units = "in", device = "pdf")
ggsave(aldexma, filename = "./outputs/phyloseq-results/figures/anti-asv-aldexma.pdf", width = 4, height = 3, units = "in", device = "pdf")
```
```{r}
cutoff <- 0.1
called <- x.ps[x.ps$we.eBH <= cutoff, ]
called.ASVs <- taxonomy[rownames(called), ]
called.ASVs1 <- cbind(called, called.ASVs)
```
```{r}
rare_cutoff <- 0
rare <- x.ps[x.ps$rab.all < rare_cutoff, ]
rare.ASVs <- taxonomy[rownames(rare), ]
rare.ASVs1 <- cbind(called, rare.ASVs)
```

## Examine specific taxa and within-taxa compositions
      filter_taxa
      prune_taxa
      subset_taxa
      
### Phylum-level subsets
```{r}
ps.AXH_proteobacteria <- subset_taxa(ps.AXH, Phylum=="D_1__Proteobacteria")
ps.AXH_cyanobacteria <- subset_taxa(ps.AXH, Phylum=="D_1__Cyanobacteria")
ps.AXH_bacteroidetes <- subset_taxa(ps.AXH, Phylum=="D_1__Bacteroidetes")
ps.AXH_actinobacteria <- subset_taxa(ps.AXH, Phylum=="D_1__Actinobacteria")
#datatable(tax_table(ps.AXH_cyanobacteria))
```
```{r}
glom_class <- tax_glom(ps.AXH_proteobacteria, taxrank = "Class")
ps.AXH_class <- psmelt(glom_class)
ps.AXH_class$Class <- as.character(ps.AXH_class$Class)
#ps.AXH_class <- filter(ps.AXH_class, Abundance > 0.005)
#ps.AXH_class$Class[ps.AXH_class$Abundance < 0.01] <- "<1% abundance"
ps.AXH_class$Sample <- factor(ps.AXH_class$Sample, levels = c("Hw1.6a", "Hw1.6b", "Hw1.6c", "Hw2.6a", "Hw2.6b", "Hw2.6c", "Wt1.6a", "Wt1.6b", "Wt1.6c", "Wt2.6a", "Wt2.6b", "Wt2.6c", "Hw1.3a", "Hw1.3b", "Hw1.3c", "Hw2.3a", "Hw2.3b", "Hw2.3c", "Wt1.3a", "Wt1.3b", "Wt1.3c", "Wt2.3a", "Wt2.3b", "Wt2.3c"))

colCount <- length(unique(ps.AXH_class$Class))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
### all samples
p_class <- ggplot(data = ps.AXH_class, aes(x = Sample, y = Abundance, fill = Class))
p_class <- p_class +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  #guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_class + ggtitle("Relative Abundance of Bacteria Classes (Phylum Proteobacteria)")

slices <- c(sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "D_2__Alphaproteobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "D_2__Deltaproteobacteria"]),
            sum(ps.AXH_class$Abundance[ps.AXH_class$Class == "D_2__Gammaproteobacteria"]))
pct <- round((slices/sum(slices)*100), digits = 1)
lbls <- c("Alphaproteobacteria", "Deltaproteobacteria", "Gammaproteobacteria")
lbls <- paste(lbls, pct)
lbls <- paste(lbls, "%", sep="")
pie(slices, labels = lbls, col = getPalette(colCount))
```
Cyanobacteria dominated by Class Oxyphotobacteria
Bacteroidetes dominated by Class Bacteroidea

### Class-level subsets
```{r}
ps.AXH_alphaproteobacteria <- subset_taxa(ps.AXH_proteobacteria, Class=="D_2__Alphaproteobacteria")
ps.AXH_gammaproteobacteria <- subset_taxa(ps.AXH_proteobacteria, Class=="D_2__Gammaproteobacteria")
```

```{r}
ps.AXH_alphaproteobacteria <- transform_sample_counts(ps.AXH_alphaproteobacteria, function(x) {x / sum(x)} )
glom_order <- tax_glom(ps.AXH_alphaproteobacteria, taxrank = "Order")
ps.AXH_order <- psmelt(glom_order)
ps.AXH_order$Order <- as.character(ps.AXH_order$Order)
#ps.AXH_order <- filter(ps.AXH_order, Abundance > 0.005)
ps.AXH_order$Order[ps.AXH_order$Abundance < 0.05] <- "<5% abundance"
ps.AXH_order$Sample <- factor(ps.AXH_order$Sample, levels = c("Hw1.6a", "Hw1.6b", "Hw1.6c", "Hw2.6a", "Hw2.6b", "Hw2.6c", "Wt1.6a", "Wt1.6b", "Wt1.6c", "Wt2.6a", "Wt2.6b", "Wt2.6c", "Hw1.4a", "Hw1.4b", "Hw1.4c", "Hw2.4a", "Hw2.4b", "Hw2.4c", "Wt1.4a", "Wt1.4b", "Wt1.4c", "Wt2.4a", "Wt2.4b", "Hw1.5a", "Hw1.5b", "Hw1.5c", "Hw2.5a", "Hw2.5b", "Hw2.5c", "Wt1.5a", "Wt1.5b", "Wt1.5c", "Wt2.5a", "Wt2.5b", "Wt2.5c", "Hw1.1a", "Hw1.1b", "Hw1.1c", "Hw2.1a", "Hw2.1b", "Hw2.1c", "Wt1.1a", "Wt1.1b", "Wt1.1c", "Wt2.1a", "Wt2.1b", "Wt2.1c"), ordered = TRUE)

colCount <- length(unique(ps.AXH_order$Order))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
### all samples
p_order <- ggplot(data = ps.AXH_order, aes(x = Sample, y = Abundance, fill = Order))
p_order <- p_order +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = c("dark grey", getPalette(colCount))) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_order
```

```{r}
slices <- c(sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "<5% abundance"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Caulobacterales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Rhizobiales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Rhodobacterales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Rhodospirillales"]))
pct <- round((slices/sum(slices)*100), digits = 1)
lbls <- c("<5% abundance", "Caulobacterales", "Rhizobiales", "Rhodobacterales", "Rhodospirillales")
lbls <- paste(lbls, pct)
lbls <- paste(lbls, "%", sep="")
pie(slices, labels = lbls, col = c("dark grey", getPalette(colCount)))
```
```{r}
ps.AXH_gammaproteobacteria <- transform_sample_counts(ps.AXH_gammaproteobacteria, function(x) {x / sum(x)} )
glom_order <- tax_glom(ps.AXH_gammaproteobacteria, taxrank = "Order")
ps.AXH_order <- psmelt(glom_order)
ps.AXH_order$Order <- as.character(ps.AXH_order$Order)
#ps.AXH_order <- filter(ps.AXH_order, Abundance > 0.005)
ps.AXH_order$Order[ps.AXH_order$Abundance < 0.01] <- "<1% abundance"
ps.AXH_order$Sample <- factor(ps.AXH_order$Sample <- factor(ps.AXH_order$Sample, levels = c("Hw1.6a", "Hw1.6b", "Hw1.6c", "Hw2.6a", "Hw2.6b", "Hw2.6c", "Wt1.6a", "Wt1.6b", "Wt1.6c", "Wt2.6a", "Wt2.6b", "Wt2.6c", "Hw1.4a", "Hw1.4b", "Hw1.4c", "Hw2.4a", "Hw2.4b", "Hw2.4c", "Wt1.4a", "Wt1.4b", "Wt1.4c", "Wt2.4a", "Wt2.4b", "Hw1.5a", "Hw1.5b", "Hw1.5c", "Hw2.5a", "Hw2.5b", "Hw2.5c", "Wt1.5a", "Wt1.5b", "Wt1.5c", "Wt2.5a", "Wt2.5b", "Wt2.5c", "Hw1.1a", "Hw1.1b", "Hw1.1c", "Hw2.1a", "Hw2.1b", "Hw2.1c", "Wt1.1a", "Wt1.1b", "Wt1.1c", "Wt2.1a", "Wt2.1b", "Wt2.1c"), ordered = TRUE))

colCount <- length(unique(ps.AXH_order$Order))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
### all samples
p_order <- ggplot(data = ps.AXH_order, aes(x = Sample, y = Abundance, fill = Order))
p_order <- p_order +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = c("dark grey", getPalette(colCount))) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_order

```

```{r}
slices <- c(sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "<5% abundance"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Alteromonadales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Betaproteobacteriales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Cellvibrionales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Coxiellales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Francisellales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__JTB23"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__KI89A clade"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Oceanospirillales"]),
            sum(ps.AXH_order$Abundance[ps.AXH_order$Order == "D_3__Salinisphaerales"]))
pct <- round((slices/sum(slices)*100), digits = 1)
lbls <- c("<5% abundance", "Alteromonadales", "Betaproteobacteriales", "Cellvibrionales", "Coxiellales", "Francisellales", "JTB23", "KI89A clade", "Oceanospirillales", "Salinisphaerales")
lbls <- paste(lbls, pct)
lbls <- paste(lbls, "%", sep="")
pie(slices, labels = lbls, col = c("dark grey", getPalette(colCount)))
```

### Order-level subsets
Rhodobacterales
```{r}
ps.AXH_rhodobacterales <- subset_taxa(ps.AXH, Order=="D_3__Rhodobacterales")
ps.AXH_rhodobacterales
```
```{r}
ps.AXH_oceanospirillales <- subset_taxa(ps.AXH, Order=="D_3__Oceanospirillales")
ps.AXH_oceanospirillales
```

### Family-level subsets and data exploration
Endozoicomondaceae
```{r}
ps.AXH_endozoicomonadaceae <- subset_taxa(ps.AXH.rel, Family=="D_4__Endozoicomonadaceae")
ps.AXH_endozoicomonadaceae
summarize_phyloseq(ps.AXH_endozoicomonadaceae)
```

```{r}
#ps.AXH_endozoicomonadaceae <- transform_sample_counts(ps.AXH_endozoicomonadaceae, function(x) {x / sum(x)} )
glom_species <- tax_glom(ps.AXH_endozoicomonadaceae, taxrank = "Species")
ps.AXH_species <- psmelt(glom_species)
ps.AXH_species$Species <- as.character(ps.AXH_species$Species)
#ps.AXH_species <- filter(ps.AXH_species, Abundance > 0.001)
#ps.AXH_species$Species[ps.AXH_species$Abundance < 0.001] <- "<0.1% abundance"
###
colCount <- length(unique(ps.AXH_species$Species))
getPalette <- colorRampPalette(brewer.pal(9, "Set1"))
### all samples
p_species <- ggplot(data = ps.AXH_species, aes(x = Sample, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
### samples collapsed into colonies
p_species <- ggplot(data = ps.AXH_species, aes(x = Treatment, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  facet_grid(~Colony) +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
### samples collapsed into treatments
p_species <- ggplot(data = ps.AXH_species, aes(x = Treatment, y = Abundance, fill = Species))
p_species <- p_species +
  geom_bar(aes(), stat = "identity", position = "fill") +
  scale_fill_manual(values = getPalette(colCount)) +
  guides(fill=guide_legend(nrow = 30)) +
  theme(axis.text.x = element_text(angle = 270))
p_species
###
tree <- plot_tree(ps.AXH_endozoicomonadaceae, ladderize = TRUE, color = "Treatment", size = "Abundance") 
tree + scale_color_manual(values = condcolors)
```

Alteromonadaceae
```{r}
ps.AXH_alteromonadaceae <- subset_taxa(ps.AXH, Family=="D_4__Alteromonadaceae")
ps.AXH_alteromonadaceae
summarize_phyloseq(ps.AXH_alteromonadaceae)
```
```{r}
tree <- plot_tree(ps.AXH_alteromonadaceae, ladderize = TRUE, color = "Treatment", size = "Abundance") 
tree + scale_color_manual(values = condcolors)
```

### Genus-level subsets (Endozoicomonas)
Prevalence and relative abundance of Endozoicomonas across colonies, then treatments
```{r}
ps.AXH_endozoicomonas <- subset_taxa(ps.AXH, Genus=="D_5__Endozoicomonas")
ps.AXH_endozoicomonas
summarize_phyloseq(ps.AXH_endozoicomonas)
ps.AXH_endozoicomonas <- transform_sample_counts(ps.AXH_endozoicomonas, function(x) {x / sum(x)} )
```
```{r}
glom_endo <- tax_glom(ps.AXH_endozoicomonas, taxrank = "Genus")
ps.AXH_endo <- psmelt(glom_endo)
ps.AXH_endo$Genus <- as.character(ps.AXH_endo$Genus)
view(ps.AXH_endo)
```

```{r}
sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas"])/sum(ps.AXH_genus$Abundance)
```
```{r}
sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas" & ps.AXH_genus$Colony=="HW1"])/sum(ps.AXH_genus$Abundance[ps.AXH_genus$Colony=="HW1"])

sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas" & ps.AXH_genus$Colony=="HW2"])/sum(ps.AXH_genus$Abundance[ps.AXH_genus$Colony=="HW2"])

sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas" & ps.AXH_genus$Colony=="WT1"])/sum(ps.AXH_genus$Abundance[ps.AXH_genus$Colony=="WT1"])

sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas" & ps.AXH_genus$Colony=="WT2"])/sum(ps.AXH_genus$Abundance[ps.AXH_genus$Colony=="WT2"])
```

```{r}
relative_endo <- function(samp) {
rel <- sum(ps.AXH_genus$Abundance[ps.AXH_genus$Genus=="D_5__Endozoicomonas" & ps.AXH_genus$Sample==samp])/sum(ps.AXH_genus$Abundance[ps.AXH_genus$Sample==samp])
print(rel)
}
```
```{r}
samps <- c("Hw1.6a", "Hw1.6b", "Hw1.6c", "Hw2.6a", "Hw2.6b", "Hw2.6c", "Wt1.6a", "Wt1.6b", "Wt1.6c", "Wt2.6a", "Wt2.6b", "Wt2.6c", "Hw1.4a", "Hw1.4b", "Hw1.4c", "Hw2.4a", "Hw2.4b", "Hw2.4c", "Wt1.4a", "Wt1.4b", "Wt1.4c", "Wt2.4a", "Wt2.4b", "Hw1.5a", "Hw1.5b", "Hw1.5c", "Hw2.5a", "Hw2.5b", "Hw2.5c", "Wt1.5a", "Wt1.5b", "Wt1.5c", "Wt2.5a", "Wt2.5b", "Wt2.5c", "Hw1.1a", "Hw1.1b", "Hw1.1c", "Hw2.1a", "Hw2.1b", "Hw2.1c", "Wt1.1a", "Wt1.1b", "Wt1.1c", "Wt2.1a", "Wt2.1b", "Wt2.1c")
for (samp in samps){
relative_endo(samp)
}
```


## Phylogenetic trees
```{r}
actinobacteria_tree <- plot_tree(ps.AXH_actinobacteria,
                                  nodelabf = nodeplotboot(95, 0, 3),
                                  label.tips = "Genus",
                                  text.size = 3,
                                  size = "Abundance",
                                  color ="Treatment",
                                  ladderize = "left",
                                  base.spacing = 0.01,
                                  plot.margin = 0.4) 
actinobacteria_tree + 
  #coord_polar(theta = "y") +
  scale_color_manual(values = condcolors_AxH) +
  ggtitle("Phylum Actinobacteria")
```

```{r}
rhodobacterales_tree <- plot_tree(ps.AXH_rhodobacterales,
                                  nodelabf = nodeplotboot(95, 0, 3),
                                  label.tips = "Genus",
                                  text.size = 3,
                                  size = "Abundance",
                                  color ="Treatment",
                                  ladderize = "left",
                                  base.spacing = 0.01,
                                  plot.margin = 0.4) 
pdf("./outputs/phyloseq-results/figures/rhodobacterales_tree.pdf", width = 8, height = 2.5)
rhodobacterales_tree + 
  #coord_polar(theta = "y") +
  scale_color_manual(values = condcolors_AxH) +
  ggtitle("Order Rhodobacterales")
```
```{r}
oceanospirillales_tree <- plot_tree(ps.AXH_oceanospirillales,
                                  nodelabf = nodeplotboot(95, 0, 3),
                                  label.tips = "Genus",
                                  text.size = 3,
                                  size = "Abundance",
                                  color ="Treatment",
                                  ladderize = "left",
                                  base.spacing = 0.01,
                                  plot.margin = 0.4) 
pdf("./outputs/phyloseq-results/figures/oceanospirillales_tree.pdf", width = 8, height = 5)
oceanospirillales_tree + 
  #coord_polar(theta = "y") +
  scale_color_manual(values = condcolors_AxH) +
  ggtitle("Order Oceanospirillales")
```

```{r}
altermonadaceae_tree <- plot_tree(ps.AXH_alteromonadaceae,
                                  nodelabf = nodeplotboot(95, 0, 3),
                                  label.tips = "Genus",
                                  text.size = 3,
                                  size = "Abundance",
                                  color ="Treatment",
                                  ladderize = "left",
                                  base.spacing = 0.01,
                                  plot.margin = 0.4) 
pdf("./outputs/phyloseq-results/figures/alteromonadaceae_tree.pdf", width = 8, height = 4)
altermonadaceae_tree + 
  #coord_polar(theta = "y") +
  scale_color_manual(values = condcolors_AxH) +
  ggtitle("Family Alteromonadaceae")
```

## Microbial networks

## Summary
```{r}
sessionInfo()
```

