---
title: "Stable_DESeq-boxplots_Pdam"
author: "Mike Connelly"
date: "10/24/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI.HW-WT-master/")
```
## Setup packages and working directories
```{r packages, error=FALSE, warning=FALSE, message=FALSE}
library("DESeq2")
library("tidyverse")
library("ggplot2")
library("ggrepel")
library("RColorBrewer")
library("scales")
```
Set colors and shapes for ggplot options
```{r}
condcolors <- c("#00CCCC", "#FFCC33", "#CC0033", "#0000FF", "#FF6600", "#FF66FF")
condfillcolors <- c("#99FFFF", "#FFFF66", "#FF6666", "#99CCFF", "#FFCC99", "#FFCCFF")
colshapes <- c(18, 9, 16, 10)
colcolors <- c("#CCFFCC", "#99FFCC", "#CCFFFF", "#CCCCFF")
```
```{r}
condcolors_LPS <- c("#0000FF", "#FF66FF")
condfillcolors_LPS <- c("#99CCFF", "#FFCCFF")
```

Create ggboxplot functions
```{r}
ggboxplot <- function(gene) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$genotype), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = colcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
```
```{r}
gglistplot <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$treatment), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = condfillcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```
```{r}
gglistplot_genotype <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
print(ggplot(gplot, aes(x=treatment, y=count, color=stable_samples$treatment)) 
      + geom_boxplot(aes(fill=stable_samples$genotype), show.legend = FALSE) 
      + geom_point(aes(shape = stable_samples$genotype), size = 2, show.legend = FALSE) 
      + facet_grid(.~stable_samples$genotype)
      + scale_color_manual(values=condcolors) + scale_fill_manual(values = colcolors) + scale_shape_manual(values = colshapes) + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```
```{r}
gglistplot_LPS <- function(genelist) {
for (gene in genelist) {
titlegene <- grep(gene, res$IDGeneInfo)
plotTitle <- res$IDGeneInfo[titlegene]
gplot <- plotCounts(dds, gene = gene, intgroup = "treatment", returnData = TRUE)
gplot <- filter(gplot, treatment == "control" | treatment == "LPS")
print(ggplot(gplot, aes(x=treatment, y=count, color=gplot$treatment)) 
      + geom_violin(aes(fill=gplot$treatment), show.legend = FALSE) 
      + geom_point(size = 2, show.legend = FALSE) 
      + scale_color_manual(values=condcolors_LPS) 
      + scale_fill_manual(values=condfillcolors_LPS) 
      + scale_y_continuous(trans = log2_trans(), breaks = trans_breaks("log2", function(x) 2^x),
    labels = trans_format("log2", math_format(2^.x)))
      + ggtitle(plotTitle))
}
}
```

## Prepare lists of gene IDs based on interesting groups
## Differentially expressed gene lists
```{r}
DElists <- list.files(path = "./DESeqresults_Pdam/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r, message=FALSE}
anti_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti_ctrl_up_ID.txt", what = "character")
LPS_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_LPS_ctrl_up_ID.txt", what = "character")
heat_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_heat_ctrl_up_ID.txt", what = "character")
anti.heat_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_ctrl_up_ID.txt", what = "character")
anti.heat.LPS_ctrl_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_ctrl_up_ID.txt", what = "character")
```
```{r, message=FALSE}
anti_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti_ctrl_dn_ID.txt", what = "character")
LPS_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_LPS_ctrl_dn_ID.txt", what = "character")
heat_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_heat_ctrl_dn_ID.txt", what = "character")
anti.heat_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_ctrl_dn_ID.txt", what = "character")
anti.heat.LPS_ctrl_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_ctrl_dn_ID.txt", what = "character")
```
```{r, message=FALSE}
anti.heat_anti_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_anti_up_ID.txt", what = "character")
anti.heat_heat_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_heat_up_ID.txt", what = "character")
anti.heat.LPS_anti_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_anti_up_ID.txt", what = "character")
anti.heat.LPS_heat_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_heat_up_ID.txt", what = "character")
anti.heat.LPS_LPS_up_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_LPS_up_ID.txt", what = "character")
```
```{r, message=FALSE}
anti.heat_anti_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_anti_dn_ID.txt", what = "character")
anti.heat_heat_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat_heat_dn_ID.txt", what = "character")
anti.heat.LPS_anti_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_anti_dn_ID.txt", what = "character")
anti.heat.LPS_heat_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_heat_dn_ID.txt", what = "character")
anti.heat.LPS_LPS_dn_IDs <- scan("./DESeqresults_Pdam/lists/res_anti.heat.LPS_LPS_dn_ID.txt", what = "character")
```
## Plot gene count boxplots
### DE Genes
```{r}
gglistplot(anti_ctrl_up_IDs[1:20])
```
```{r}
gglistplot(LPS_ctrl_up_IDs[1:15])
```
```{r}
gglistplot(LPS_ctrl_up_IDs[1:20])
```
```{r}
gglistplot(heat_ctrl_up_IDs[1:10])
```
```{r}
gglistplot(anti.heat_ctrl_up_IDs[1:10])
```
```{r}
gglistplot(anti.heat.LPS_ctrl_up_IDs[1:10])
```

### Innate immune genes
Lists of gene IDs
#### Pattern recognition receptors
```{r}
TLR_IDs <- scan("./genefamilies_Pdam/TLRs_IDs.txt", what = "character")
Lectin_IDs <- scan("./genefamilies_Pdam/Lectins_IDs.txt", what = "character")
#NLR_IDs
TNFR_IDs <- scan("./genefamilies_Pdam/TNFRs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TNFR_IDs)
```
```{r}
gglistplot(Lectin_IDs)
```

#### Signaling pathways
NFKB
```{r}
NFKB_IDs <- scan("./genefamilies_Pdam/NFKB_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(NFKB_IDs)
```
JNK
MAPK
```{r}
TRAF_IDs <- scan("./genefamilies_Pdam/TRAFs_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(TRAF_IDs)
```

#### Effector mechanisms
MPEG1 pdam_00017055
Damicornin pdam_00023321
DELTA-stichotoxin-She4b
Phospholipases
```{r}
PL_IDs <- scan("./genefamilies_Pdam/PL_IDs.txt", what = "character")
```
```{r}
gglistplot_genotype(PL_IDs)
```
Toxins
```{r}
Tox_IDs <- scan("./genefamilies_Pdam/Toxin_IDs.txt", what = "character")
```

```{r}
gglistplot(Tox_IDs[28])
```

```{r}
effectorIDs <- c("pdam_00017055", "pdam_00004721", "pdam_00023321")
gglistplot(effectorIDs)
```

#### Misc. immune genes
DMBT1
HLF

### Apoptosis pathway genes
Caspases

### Antioxidant response
SOD
PXDN

### Transcription factors
HLF

WGCNA Hub Genes
```{r}
hubIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/hubgenes/hubs_IDs.txt", what = "character")
```
```{r}
gglistplot(hubIDs[1:20])
```

WGCNA Module Genes
## Input lists of co-expressed module genes after WGCNA analysis
```{r, warning = FALSE}
modulelists <- list.files(path = "./WGCNAmodules_Pdam_ctrl-LPS/lists", pattern = "_ID.txt", full.names = TRUE)
```
```{r}
blackIDs <- scan("./WGCNAmodules_Pdam_ctrl-LPS/lists/black_ID.txt", what = "character")
```

```{r}
intblackIDs <- c(14,21,29,199,200,213,231,336,359:364,402,408,415,437,481)
gglistplot_LPS(blackIDs[intblackIDs])
```