---
title: "stable_AXH-aldex2"
author: "Mike Connelly"
date: "10/9/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI-bacteria/all_wd/")
```
## Setup packages and working directories
```{r, message=FALSE}
library(tidyverse)
library(ggplot2)
library(phyloseq)
library(vegan)
library(ape)
library(DESeq2)
library(edgeR)
library(microbiome)
library(RColorBrewer)
library(ggpubr)
library(grid)
library(plyr)
library(data.table)
library(DT)
library(ranacapa)
library(ALDEx2)
```

Set theme, colors and shapes for ggplot2 plotting
```{r, warning=FALSE, message=FALSE}
theme_set(theme_bw())
#Treatment color and fill scheme
condcolors <- c("#0000FF", "#00CCCC", "#FF6600", "#FFCC33", "#CC0033", "#FF66FF")
condcolors_anti <- c("#00CCCC", "#0000FF")
condcolors_heat <- c("#0000FF", "#FF6600")
condcolors_anti.heat <- c("#FFCC33", "#0000FF")
condcolors_antixheat <- c("#0000FF", "#FF6600", "#00CCCC", "#FFCC33")
###
condfillcolors_antixheat <- c("#99CCFF", "#FFCC99", "#99FFFF", "#FFFF66")
###
#Colony shape, color and fill scheme
colshapes <- c(18, 9, 16, 10, 15, 12, 17, 2)
colcolors <- c("#CCFFCC", "#99FFCC", "#CCFFFF", "#99CCFF", "#FFCCFF", "#FF99FF", "#CCCCFF", "#CC99FF")
```
```{r}
### Bacteria taxa
## Phyla
phyla_colors <- read_csv("~/computing/scripts/EAPSI_AntiXHeat-master/taxa_colors/phyla_colors.csv", col_names = c("phylum", "color"))
## Classes
classes_colors <- read_csv("~/computing/scripts/EAPSI_AntiXHeat-master/taxa_colors/classes_colors.csv", col_names = c("phylum", "class", "color"))
## Orders
orders_colors <- read_csv("~/computing/scripts/EAPSI_AntiXHeat-master/taxa_colors/orders_colors.csv", col_names = c("phylum", "class", "order", "color"))
## Families
families_colors <- read_csv("~/computing/scripts/EAPSI_AntiXHeat-master/taxa_colors/families_colors3.csv", col_names = c("phylum", "class", "order", "family", "color"))
## Genera
#genera_colors <- 
## Specices
#species_colors <- 
```

Read in unrareified 99% OTU (ASV) table
```{r}
ASV <- read.table("./export_silva/all_silva_feature-table.tsv", row.names = 1, header = TRUE)
ASV <- t(ASV)
```
Read in rareified 99% OTU (ASV) table
```{r}
ASV2600 <- read.table("./export_silva/all_silva_rarefied_feature-table.tsv", row.names = 1, header = TRUE)
ASV2600 <- t(ASV2600)
```
Read in taxononomic classification information
```{r}
taxonomy <- as.matrix(read.table("./export_silva/taxonomy4.txt", sep = "\t", row.names = 1, header = FALSE, fill = TRUE))
colnames(taxonomy) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
taxonomy[taxonomy==""] <- NA
taxdf <- as.data.frame(taxonomy)
```
Read in phylogenetic tree, sample metadata and set factor orders
```{r}
tree <- read_tree("./export_silva/tree.nwk")
map <- read.table("./metadata/metadata_all.tsv", sep ="\t", row.names = 1, header = FALSE)
colnames(map) <- c ("Reef",	"Colony",	"Environment",	"Treatment")
map$SampleID <- rownames(map)
map$Reef <- factor(map$Reef, levels = c("Houwan", "Wanglitung", "Outlet", "Houbihu", "controls"), ordered = TRUE)
map$Colony <- factor(map$Colony, levels = c("HW1", "HW2", "WT1", "WT2", "OT1", "OT2", "HB1", "HB2", "controls"), ordered = TRUE)
map$Treatment <- factor(map$Treatment, levels = c("control", "Heat", "Antibiotics", "Antibiotics.Heat", "Antibiotics.Heat.LPS", "LPS", "positive", "negative"), ordered = TRUE)
```

```{r}
data("selex")
```

```{r}
conds <- c(rep("control", 4))
```

```{r}
ps.stable.AXH
summarize_phyloseq(ps.stable.AXH)
```
```{r}
datatable(tax_table(ps.stable.AXH))
```
```{r}
ps.stable.AXH.rel <- transform_sample_counts(ps.stable.AXH, function(x) {x / sum(x)} )
ps.stable.AXH.clr <- microbiome::transform(ps.stable.AXH, "clr")
```

