---
title: "Stable_DEseq-heatmaps_Pdam"
author: "Mike Connelly"
date: "10/30/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI.HW-WT-master")
```
## Setup packages and working directories
```{r packages, error=FALSE, warning=FALSE, message=FALSE}
library("tidyverse")
library("ggplot2")
library("ggrepel")
library("calibrate")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
```
## Set general ggplot2 colors 
```{r}
condcolors <- c("#00CCCC", "#FFCC33", "#CC0033", "#0000FF", "#FF6600", "#FF66FF")
condfillcolors <- c("#99FFFF", "#FFFF66", "#FF6666", "#99CCFF", "#FFCC99", "#FFCCFF")
colshapes <- c(18, 9, 16, 10)
colcolors <- c("#CCFFCC", "#99FFCC", "#CCFFFF", "#CCCCFF")
```
## Regularized-log transformation
```{r, include=FALSE}
      rld <- rlogTransformation(dds)
```
### Create heatmap annotation data frame
```{r}
anno_col <- as.data.frame(colData(rld)[, c("treatment","genotype")])
```
### Create colony and condition annotation colors
```{r}
anno_colors <- list(
  treatment = c(control=condcolors[4], Antibiotics.Heat=condcolors[2], Antibiotics.Heat.LPS=condcolors[3],  Antibiotics=condcolors[1], Heat=condcolors[5], LPS=condcolors[6]),
  genotype = c(Houwan1=colcolors[1], Houwan2=colcolors[2], Wanglitung1=colcolors[3], Wanglitung2=colcolors[4]))
```

## Sample Clustering Heatmap
```{r}
sampleDists <- dist(t(assay(rld)))
sampleDistMatrix <- as.matrix( sampleDists )
```
### Create sample heatmap cell color
```{r}
colonyclustercol <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
```
```{r}
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colonyclustercol,
         annotation_colors = anno_colors,
         annotation_col = anno_col,
         annotation_legend = FALSE,
         treeheight_col = 30,
         treeheight_row = 30,
         main = "Stable Site Samples Heatmap")
```

## Gene Clustering Heatmaps
### Create gene expression cell color
```{r}
geneexpcolors <- colorRampPalette( rev(brewer.pal(9, "Spectral")) )(300)
```
### Create top 20 variable gene expression matrix 
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes20 <- head(order(rowVars(assay(rld)), decreasing = TRUE), 20)
mat20  <- assay(rld)[ topVarGenes20, ]
rownames(mat20) <- rowData(rld)$IDGeneInfo[topVarGenes20]
```
```{r}
mat20  <- mat20 - rowMeans(mat20)
```
## 20 Gene Heatmap
```{r}
png(filename = "./DESeqresults_Pdam/figures/Heatmap_top20gene", width = 1500, height = 400)
pheatmap(mat20, col=geneexpcolors,
         annotation_col = anno_col,
         annotation_colors = anno_colors,
         annotation_legend = FALSE,
         legend = TRUE,
         treeheight_col = 50,
         treeheight_row = 40,
         main="Stable SampleXGene Heatmap")
dev.off()
```

### Create top 50 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes50 <- head(order(rowVars(assay(rld)), decreasing = TRUE), 50)
mat50  <- assay(rld)[ topVarGenes50, ]
rownames(mat50) <- rowData(rld)$IDGeneInfo[topVarGenes50]
```
## 50 Gene Heatmap
```{r}
png(filename = "./DESeqresults_Pdam/figures/Heatmap_top50gene", width = 1500, height = 1000)
pheatmap(mat50, col=geneexpcolors,
         annotation_col = anno_col,
         annotation_row = NULL,
         annotation_colors = anno_colors,
         annotation_legend = TRUE,
         legend = TRUE,
         treeheight_col = 50,
         treeheight_row = 80,
         main="Stable SampleXGene Heatmap")
dev.off()
```

### Create top 500 variable gene expression matrix
```{r geneheatmap, message = FALSE, warning = FALSE}
#Create heatmap matrix and annotation groupings
topVarGenes500 <- head(order(rowVars(assay(rld)), decreasing = TRUE), 500)
mat500  <- assay(rld)[ topVarGenes500, ]
rownames(mat500) <- rowData(rld)$IDGeneInfo[topVarGenes500]
```
## 500 Gene Heatmap
```{r}
png(filename = "./DESeqresults_Pdam/figures/Heatmap_top500gene", width = 1500, height = 3000)
pheatmap(mat500, col=geneexpcolors,
         annotation_col = anno_col,
         annotation_row = NULL,
         annotation_colors = anno_colors,
         annotation_legend = TRUE,
         legend = TRUE,
         treeheight_col = 50,
         treeheight_row = 80,
         main="Stable SampleXGene Heatmap")
dev.off()
```