---
title: "Stable_WGCNA-correlations_Pdam"
author: "Mike Connelly"
date: "10/25/2018"
output: html_document
---

RMarkdown document containing scripts for WGCNA modelling of differential gene expression between fragments of two *Pocillopora damicornis* genotypes collected at Wanglitung reef and placed in five experimental treatment groups and once control group during EAPSI Taiwan, summer 2017.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "~/computing/scripts/EAPSI.HW-WT-master/")
```
## Load packages and setup working directory
```{r, error=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(RColorBrewer)
library(genefilter)
library(WGCNA)
library(dynamicTreeCut)
library(flashClust)
options(stringsAsFactors = FALSE)
allowWGCNAThreads(nThreads = 4)
```
## Quantify module-trait associations
```{r}
datTraits <- select(stable_samples, -treatment, -genotype, -heat, -anti, -anti.heat, -anti.heat.lps, - allheat, -allanti) 
#-allheat, -allanti, -alllps, -W1, -W2)
rownames(datTraits) <- rownames(stable_samples)
#Define numbers of genes and samples
nGenes <- ncol(datExpr);
nSamples <- nrow(datExpr);
```
## Correlate module eigengene-trait associations
```{r}
moduleTraitCor <- cor(MEs1, datTraits, use = "p")
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples)
moduleCors <- as.data.frame(moduleTraitCor, moduleTraitPvalue)
```
## Create ME-trait heatmap
```{r}
setwd("./WGCNAmodules_Pdam_ctrl-LPS/figures")
pdf("ModuleCorrelations_ctrl-LPS", width = 11, height = 16)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 2), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor,
xLabels = names(datTraits),
yLabels = names(MEs1),
yLabelsPosition = "left",
ySymbols = names(MEs1),
yColorWidth = 0.005,
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
plotLegend = FALSE,
cex.text = 0.5,
zlim = c(-1,1),
main = paste("Module-treatment correlations"))
dev.off()
```
## Create ME-trait heatmaps for interesting modules only
```{r}
for (module in intModules)
{
which.module <- module
intMEs <- MEs1[, paste("ME", which.module, sep="")]
intmoduleTraitCor <- cor(intMEs, datTraits, use = "p")
intmoduleTraitPvalue <- corPvalueStudent(intmoduleTraitCor, nSamples)
# Will display correlations and their p-values
textMatrix = paste(signif(intmoduleTraitCor, 2), "\n(",
signif(intmoduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(intmoduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = intmoduleTraitCor,
xLabels = names(datTraits),
yLabels = names(intMEs),
yLabelsPosition = "left",
ySymbols = names(intMEs),
yColorWidth = 0.005,
colorLabels = FALSE,
colors = blueWhiteRed(50),
textMatrix = textMatrix,
setStdMargins = FALSE,
plotLegend = FALSE,
cex.text = 0.6,
zlim = c(-1,1),
main = paste(which.module, "Module-trait relationships"))
}
```
## Barplots of interesting module eigengene expression
### Set treatment colors
```{r}

gendensity <- c(rep(200, 6), rep(100, 5), rep(30, 6), rep(10, 6))
gencolors <- c(rep("#CCFFCC",6), rep("#99FFCC",5), rep("#CCFFFF",6), rep("#CCCCFF",6))

condcolors <- c(rep("#0000FF",3), rep("#FF66FF",3), rep("#0000FF",3), rep("#FF66FF",2), rep("#0000FF",3), rep("#FF66FF",3), rep("#0000FF",3), rep("#FF66FF",3))
```
```{r}
par(mfrow = c(1,1)  
    #mar = c(8,2,2,2)
    )
for (module in intModules)
{
which.module <- module
ME <- MEs1[, paste("ME", which.module, sep="")]
barplot(ME, col = gencolors, 
        border =  condcolors,
        #density = gendensity,
        main = which.module, ylim = c(-0.6,0.6), 
        ylab = "eigengene expression", 
        axes = TRUE, 
        #names.arg = WTsamples$condition, 
        cex.lab = 0.01, las = 3)
}
```

ME <- as.data.frame(ME)
g <- ggplot(ME, aes(y = ME))
g + geom_point()

## Heatmaps of interesting module gene expression
```{r}
for (module in immModules)
{
which.module <- module
ME <- MEs1[, paste("ME", which.module, sep="")]
par(mfrow = c(1,1), mar = c(1,2,4,1))
plotMat(t(scale(datExpr[,moduleColors==which.module ]) ), nrgcols = 30, rlabels = T, clabels = rownames(WTsamples), rcols = which.module, title = which.module)
}
```

## Module connectivity and gene significance scatterplots
```{r}
which.module <- module
column <- match(which.module, modNames);
moduleGenes <- moduleColors==which.module;

verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]), abs(geneTraitSignificance[moduleGenes, 1]), xlab = "Module membership", ylab = "Gene Significance", main = "Module Membership vs. Gene Significance")
```

